---
title: "Zooplankton Microbiomes Across Time and Space"
author: "MG Perreault, CB Wall"
date: "6/15/2025"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 4
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '4'
editor_options: 
  chunk_output_type: console
---

```{r global options, results="hide", warning=FALSE, message=FALSE}
if (!require('knitr')) install.packages('knitr'); library('knitr')
knitr::opts_chunk$set(warning=FALSE, message=FALSE, fig.align='center', collapse=TRUE, cache=TRUE)

# load packages
if (!require("pacman")) install.packages("pacman") # for rapid install if not in library


devtools::install_github("benjjneb/dada2", ref="v1.20") # update to most recent dada2
remotes::install_github("microbiome/microbiome")

if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

if (!require("phyloseq", quietly = TRUE))
    BiocManager::install("phyloseq",  quietly = TRUE)

if (!require("dada2", quietly = TRUE))
    BiocManager::install("dada2", quietly = TRUE)

# use pacman to load CRAN packages missing
pacman::p_load('knitr', 'microbiome', 'phyloseq', 'tidyr', 'tidyverse', 'knitr', 'magrittr', 'effects', 'devtools', 'stringi', 'dplyr', "ggplot2", "gridExtra", "dada2", "phyloseq", "vegan", "cowplot", 'doBy', 'ecodist', 'phyloseqCompanion', 'pairwiseAdonis', 'glue', 'geosphere', 'data.table', 'patchwork', 'car', 'ggcorrplot', 'FactoMineR', 'devtools', 'reshape', 'lattice',  'plyr', 'magrittr', 'factoextra', 'multcompView', 'decontam', 'factoextra', 'car', 'mia', 'ade4', 'fossil', 'picante', 'reshape', 'readr', 'corrr', 'Hmisc', "MASS", "FSA", "sciplot", "decontam","BiocManager", 'ggpubr', 'ggmap', "ggordiplots", "fossil", "SpiecEasi", "igraph", "huge", "leaflet", "ggtree", "colorBlindness", "RColorBrewer","MicrobiotaProcess", "predict3d", "ggiraphExtra", "data.table", "betapart", "naniar", "rstatix")
```

## Overview

### Map
We samples 6 locations in the Eastern Sierra Nevada Mountains throughout the summer season. The map of lakes and their proximity can be seen below.

<center>  
  
![**Image**. Sierra Nevada Lakes sampled for zooplankton microbiomes and environmental conditons across a summer season.](output/maps/Fig1.map.3 panel2.jpg){width=80%}
 
</center>  

### Envrironmental Data
Load in the environmental field data and create plots for environmental conditions and water nutrients/chemistry across all lakes and sampling points. 
- first remove outliers, check data structure, and add some carpintery to the metadata data frame.
```{r depth data, results='hide'}
field.df<-read.csv("data/field_metadata.csv")
field.df$date.simple<-field.df$time_point
field.df$date.simple<-recode_factor(field.df$date.simple, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")


str(field.df)
field.df$TDP..ug.L<-as.numeric(field.df$TDP..ug.L)
field.df$TDN..ug.L<-as.numeric(field.df$TDN..ug.L)

## replace outliers in nutrient analysis
field.df<-field.df %>% 
  naniar::replace_with_na(replace = list(TDP..ug.L = 4102.48))

# order the lakes by North to South or upstream in phyloseq 
field.df$lake <- factor(field.df$lake, levels = 
                                    c("Cooney", "Blue", "Virginia", "Convict", "Serene", "Eastern Brook"))
# make meters
field.df$depth..m<-field.df$depth..ft*0.3048

#fix str
field.df$lake<-as.factor(field.df$lake)
field.df$time_point<-as.factor(field.df$time_point)

# depth mean and SE
mean.depth.m<-aggregate(depth..m~lake, FUN=mean, data=field.df)
mean.depth.SE<-aggregate(depth..m~lake, function(x) sd(x) / sqrt(length(x)), data=field.df)

colnames(mean.depth.m)<-c("lake", "depth..m")
colnames(mean.depth.SE)<-c("lake", "depth..m.SE")
depth.df<-cbind(mean.depth.m[1:2], mean.depth.SE[2])
```

- nutrients and water chemistry
```{r nutrients, fig.show='hold', fig.cap="*Fig 2.* Environmental and water chemistry metrics measured in each of the six lakes across five sampling time points."}
######## TDP

TP.plot<-ggplot(field.df, aes(x=date.simple, y=TDP..ug.L, group=lake, color=lake)) + 
  geom_point(size=2)+
  geom_line() + 
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") + 
  ylab(expression(paste("TDP", ~(mu*g~L^-1), sep = "")))+ theme_classic() + 
   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


######## TDN

TN.plot<-ggplot(field.df, aes(x=date.simple, y=TDN..ug.L, group=lake, color=lake)) + 
  geom_point(size=2)+
  geom_line() + 
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") + 
  ylab(expression(paste("TDN", ~(mu*g~L^-1), sep = ""))) + theme_classic() + 
   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


####### DOC

DOC.plot<-ggplot(field.df, aes(x=date.simple, y=DOC_values, group=lake, color=lake)) + 
  geom_line() +  
  geom_point(size = 2)+  
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") + 
  ylab(expression(paste("DOC", ~(mg~L^-1), sep = ""))) + theme_classic() + 
   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


####### chlorophyll

chla.plot<- ggplot(field.df, aes(x= date.simple, y=chla..ug.L, group=lake, color=lake)) + 
  geom_line() +  
  geom_point(size = 2)+  
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") + 
  coord_cartesian(ylim=c(0, 2.5))+
  theme(axis.text = element_text(size=7)) +  
  ylab(expression(paste("Chl-a", ~(mu*g~ L^-1), sep = "")))  + theme_classic() + 
   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


######## Dissolved oxygen

DO.percent.plot<-ggplot(field.df, aes(x= date.simple, y=DO..perc, group=lake, color=lake)) + 
  geom_line() +  
  geom_point(size = 2)+ 
  coord_cartesian(ylim=c(60, 90))+
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") + ylab("DO (%)") +
  theme(text = element_text(size = 10)) + theme_classic() + 
   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


######### Conductivity

conductivity.plot<-ggplot(field.df, aes(x= date.simple,  y=cond, group=lake, color=lake)) +
  geom_line() +  
  geom_point(size = 2)+ 
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") +
  coord_cartesian(ylim=c(0, 180))+
  ylab(expression(paste("Conductivity", ~(mu*S~ cm^-1), sep = ""))) + theme_classic() + 
   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


###### Specific conductivity

spc.plot <- ggplot(field.df, aes(x= date.simple,  y=spc, group=lake, color=lake)) + 
  geom_line() +  
  geom_point(size = 2)+ 
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") +
  coord_cartesian(ylim=c(0, 200))+
  ylab(expression(paste("SPC", ~(mu*S~ cm^-1)))) + theme_classic() +  
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


######## pH

pH.plot<-ggplot(field.df, aes(x= date.simple,  y=pH, group=lake, color=lake)) + 
  geom_line() +  
  geom_point(size = 2)+ 
  coord_cartesian(ylim=c(5, 10))+
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  xlab("Sampling Time Point") + theme_classic() +  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

######## Surface temp

laketemp.plot<-ggplot(field.df, aes(x= date.simple,  y=temp..C, group=lake, color=lake)) + 
  geom_line() +  
  geom_point(size = 2)+ 
  coord_cartesian(ylim=c(12, 22))+
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4")) +
  xlab("Sampling Time Point") + 
  ylab("Temperature (°C)") + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


## get legend
extract.legend.env <- get_legend(
  # create some space to the left of the legend
  laketemp.plot + theme(legend.box.margin = margin(0, 0, 0, 10)))


#### combine plot
env.plots<-plot_grid(laketemp.plot + guides(color="none"), DO.percent.plot + guides(color="none"), 
          pH.plot + guides(color="none"), conductivity.plot + guides(color="none"), 
          chla.plot + guides(color="none"), DOC.plot + guides(color="none"), 
          TN.plot + guides(color="none"), TP.plot + guides(color="none"),
          ncol=4, rel_widths = c(7,7,7,7,7,7,7,1), labels=c("A", "B", "C", "D", "E", "F", "G", "H"))

env.plots
dev.copy(pdf, "figures/Fig2.env.plots.pdf", height=7, width = 12)
dev.off()
```

- air temperatures and precipitation
```{r weather air temp and precipitation, fig.cap="*Fig S1.*  Average air temperature and total daily precipitation across the summer season in which lake sampling occurred." }
# air temp and rainfall

weather <- read.csv('data/weather.SNARL.csv', header = TRUE)
weather$date <- as.Date(weather$date, format="%m/%d/%y")
coeff<-0.5

weather.plot <- ggplot(weather, aes(x = date)) +
  geom_line(aes(y = avg.temp.air.C, color = "avg.temp"), linewidth = 1, linetype = "solid") +
  geom_line(aes(y = rain.total.mm*coeff, color = "rain.total"), linewidth = 1, linetype = "solid") +
  labs(x = "Date", y = "Average Air Temperature (°C)", title = NULL) + 
  scale_y_continuous(sec.axis= sec_axis(~./coeff, name = "Total Daily Precipitation (mm)")) +
  scale_x_date(breaks = "2 week", date_labels = "%b %d") +
  scale_color_manual( values = c("rain.total" = "steelblue", "avg.temp" = "orangered"), breaks = c("avg.temp", "rain.total")
  ) + theme_classic() +
theme(axis.text.x = element_text(angle = 90))

weather.plot
dev.print(pdf, "figures/FigS1.Air.Rain.snarl.plot.pdf", width = 6, height = 6)
dev.off()

```


### Zooplankton Community
Using the zooplankton community data, we will make plots for the density of plankton at each lake and at the different sampling points, creating a dot-line plot for each taxa. We will then make a relative abundance stacked-density plot to show the change in zooplankton community composition.  

#### Plankton density
- first, the dot-line plot of the common taxa
```{r plot of zooplankton community dot-line, results='hide'}
# load in the density data
# density is the number of zoops per liter
counts <- read.csv('data/Zoop.community.csv')
counts$time_point <- as.factor(counts$time_point)
counts$lake <- as.factor(counts$lake)
str(counts)

counts$date.simple<-recode_factor(counts$time_point, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")

# order the lakes by North to South or upstream in phyloseq 
counts$lake <- factor(counts$lake, levels = 
                                    c("Cooney", "Blue", "Virginia", "Convict", "Serene", "Eastern Brook"))

# make sample number (each sample is a different row)
counts$sample <- row.names(counts)


# make a new column at the end for total added density of all zoops at each time/lake combo?
counts_mod <- counts[,c(23:34)]%>%
  mutate(sum_of_rows = rowSums(.))
counts_mod

# add this new column to the orginal "counts" df 
counts$dens.row.sums <- counts_mod$sum_of_rows

# make a new column for region category
counts$region <- as.factor(ifelse(counts$lake == "Eastern Brook" | 
                                    counts$lake == "Serene" |
                                  counts$lake == "Convict", "South", "North"))

#### Total densities
overall.dens.plot <- ggplot(counts, aes(x = date.simple, y = dens.row.sums, color = lake, group = lake)) +
  geom_point(size=2) + 
  geom_line() + 
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+ 
  labs( x = "Sampling Time Point", y = "Total Zooplankton Density (#/L)") +theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
```

Test for differences in zooplankton densities across lakes and time points.  
*Note* Interactions aren't approproate for these models since there is only n=1 in each lake/time combination. We therefore use Type II SS of main effects.
```{r testing effects on densities and plotting, results='hide', collapse=TRUE}
### Lake and time effects
tot.dens.mod<-lm(dens.row.sums ~ time_point + lake, data = counts)
tot.dens.site.time.aov<-Anova(tot.dens.mod, type=2) #time point trend
#plot(allEffects(tot.dens.mod))

# posthoc just looking at time point, a trend
tot.dens.ph<-emmeans(tot.dens.mod, pairwise~time_point)
multcomp::cld(tot.dens.ph, Letters=letters)

### Env. effects
stepAIC(lm(dens.row.sums ~ latitude + elevation_m + DO..perc + log(DOC) + chla_ug.L + pH + cond + temp..C, data = counts))
tot.zoop.dens.env.mod<-lm(dens.row.sums ~ latitude + elevation_m + DO..perc + cond + temp..C, data = counts)
tot.zoop.dens.env.aov<-Anova(tot.zoop.dens.env.mod, type=2)
#plot(allEffects(lm(dens.row.sums ~ temp..C, data = counts)))
```

- Best models are below for total zooplankton densities by Site x Time, and by environmental conditons. 
- These are in *(BOTTOM) Table S1 and S2*.
```{r best model zoop densities}
tot.dens.site.time.aov
tot.zoop.dens.env.aov
```

- Now, test for differences in density of individual taxa
- We will look at Daphnia, calanoid copepods, cyclopoid copepods, nauplii, asplanchna,and other less common groups
```{r testing effects for individual taxa, results='hide'}
##### Daphnia
daphnia.dens.plot <- ggplot(counts, aes(x = date.simple, y = daphnia.dens, color = lake, group = lake)) +
  geom_point(size = 2) + 
  geom_line() + 
  labs( x = "Sampling Time Point", y = "Daphnia Density (#/L)") +
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# model
daph.dens.mod<-lm(daphnia.dens ~ time_point + lake, data = counts)
daph.dens.aov<-Anova(daph.dens.mod, type=2) #time point trend
hist(residuals(daph.dens.mod))
plot(allEffects(daph.dens.mod))

# posthoc just looking at time point, a trend
daph.dens.ph<-emmeans(daph.dens.mod, pairwise~time_point)
multcomp::cld(daph.dens.ph, Letters=letters)


##### Calanoid
calanoid.dens.plot <- ggplot(counts, aes(x = date.simple, y = cal.dens, color = lake, group = lake)) +
  geom_point(size = 2) + 
  geom_line() + 
  labs( x = "Sampling Time Point", y = "Calanoid Density (#/L)") +
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# model
calan.dens.mod<-lm(cal.dens ~ time_point + lake, data = counts)
calan.dens.aov<-Anova(calan.dens.mod, type=2) #lake effect
hist(residuals(calan.dens.mod))
plot(allEffects(calan.dens.mod))

# posthoc just looking at time point, a trend
calan.dens.ph<-emmeans(calan.dens.mod, pairwise~lake)
multcomp::cld(calan.dens.ph, Letters=letters)


#### Cyclopoid
cyclopoid.dens.plot <- ggplot(counts, aes(x = date.simple, y = cyc.dens, color = lake, group = lake)) +
  geom_point(size = 2) + 
  geom_line() + 
  labs( x = "Sampling Point", y = "Cyclopoid Density (#/L)") +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# model
cyclo.dens.mod<-lm(cyc.dens ~ time_point + lake, data = counts) # time_point
cyclo.dens.aov<-Anova(cyclo.dens.mod, type=2)
hist(residuals(cyclo.dens.mod))
plot(allEffects(cyclo.dens.mod))

# posthoc just looking at time point
cyclo.dens.ph<-emmeans(cyclo.dens.mod, pairwise~time_point)
multcomp::cld(cyclo.dens.ph, Letters=letters)


#### nauplii
nauplii.dens.plot <- ggplot(counts, aes(x = date.simple, y = nau.dens, color = lake, group = lake)) +
  geom_point(size = 2) + 
  geom_line() + 
  labs( x = "Sampling Point", y = "Nauplii Density (#/L)") +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# model
nau.dens.mod<-lm(nau.dens ~ time_point + lake, data = counts)
nau.dens.aov<-Anova(nau.dens.mod, type=2) #time point signif
hist(residuals(nau.dens.mod))
plot(allEffects(nau.dens.mod))

# posthoc just looking at time point
nau.dens.ph<-emmeans(nau.dens.mod, pairwise~time_point)
multcomp::cld(nau.dens.ph, Letters=letters)


##### asplanchna
asplanchna.dens.plot <- ggplot(counts, aes(x = date.simple, y = asp.dens, color = lake, group = lake)) +
  geom_point(size = 2) + 
  geom_line() + 
  labs( x = "Sampling Time Point", y = "Asplanchna Density (#/L)") +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))


# model
asp.dens.mod<-lm(log(asp.dens+0.001) ~ time_point + lake, data = counts)
asp.dens.aov<-Anova(asp.dens.mod, type=2) #time point trend, lake signif
hist(residuals(asp.dens.mod))
plot(allEffects(asp.dens.mod))

# posthoc just looking at time point, a trend
asp.dens.ph<-emmeans(asp.dens.mod, pairwise~lake)
multcomp::cld(asp.dens.ph, Letters=letters)


####### ####### ####### 
####### less common groups

#### Ceriodaphnia
ceriodaphnia <- ggplot(counts, aes(x = date.simple, y = cerio.dens, color = lake, group = lake)) +
  geom_point(size=2) + 
  geom_line()+
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ceriodaphnia

# model
ceri.dens.mod<-lm(cerio.dens ~ time_point + lake, data = counts)
ceri.dens.aov<-Anova(ceri.dens.mod, type=2) #lake signif
plot(allEffects(ceri.dens.mod))

# posthoc just looking at time point
ceri.dens.ph<-emmeans(ceri.dens.mod, pairwise~lake)
multcomp::cld(ceri.dens.ph, Letters=letters)


###### bosmina
bosmina.dens.plot <- ggplot(counts, aes(x = date.simple, y = bos.dens, color = lake, group = lake)) +
  geom_point(size=2) + 
  geom_line() +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

Anova(lm(bos.dens ~ time_point + lake, data = counts)) # no effect


###### kellicotia
kellicotia.dens.plot <- ggplot(counts, aes(x = date.simple, y = kell.dens, color = lake, group = lake)) +
  geom_point(size=2) + 
  geom_line() +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

Anova(lm(kell.dens ~ time_point + lake, data = counts)) # no effect


##### keratella
keratella.dens.plot <- ggplot(counts, aes(x = date.simple, y = ker.dens, color = lake, group = lake)) +
  geom_point(size=2) + 
  geom_line() +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
keratella

Anova(lm(ker.dens ~ time_point + lake, data = counts)) # no effect


#### holopedium
holopedium.dens.plot <- ggplot(counts, aes(x = date.simple, y = holo.dens, color = lake, group = lake)) +
  geom_point(size=2) + 
  geom_line() +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

Anova(lm(holo.dens ~ time_point + lake, data = counts)) # not enough species frequency


#### polyphemus
polyphemus.dens.plot <- ggplot(counts, aes(x = date.simple, y = poly.dens, color = lake, group = lake)) +
  geom_point(size=2) + geom_line() +
   scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

Anova(lm(poly.dens ~ time_point + lake, data = counts)) # no effect


#### chydorus
chydorus.dens.plot <- ggplot(counts, aes(x = date.simple, y = chy.dens, color = lake, group = lake)) +
  geom_point(size=2) + geom_line() +
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  theme_classic()

Anova(lm(chy.dens ~ time_point + lake, data = counts)) # no effect
```

- Compile the dot-line plot for all common zooplankton taxa
```{r combine zooplankton plot, results='hide', fig.cap="*Fig S3.* Changes in zooplankton densities across six mountain lakes across five sampling time points"}
####### make aggregated plot
combined.dens.plot <- ggarrange(
                           overall.dens.plot + theme(axis.title.y =element_text(size=8))+ guides(color="none"),
                           daphnia.dens.plot + theme(axis.title.y =element_text(size=8))+ guides(color="none"),
                           calanoid.dens.plot + theme(axis.title.y=element_text(size=8))+ guides(color="none"),
                           cyclopoid.dens.plot + theme(axis.title.y=element_text(size=8))+ guides(color="none"),
                           nauplii.dens.plot + theme(axis.title.y=element_text(size=8))+ guides(color="none"),
                           asplanchna.dens.plot + theme(axis.title.y=element_text(size=8))+ guides(color="none"),
                           labels = c("A", "B", "C", "D", "E", "F"), nrow = 2, ncol = 3)
                           #common.legend = TRUE, legend="right",
                           #align="hv", font.label = list(size=10, color="black", family=NULL, position = "top"))
combined.dens.plot
dev.print(pdf, "figures/FigS3.combined.dens.plot.pdf", height=7, width=8)
dev.off()

```

#### Plankton relative abundance
- Create a stacked density plot for the zooplankton relative abundance
```{r stacked density plot, results="hide", fig.cap="*Fig 3.* Relative abundances of zooplankton taxa (as proportion of total density per liter) in communities of six mountain lakes in California’s Eastern Sierra Nevada across five sampling time points."}
long_counts <- tidyr::pivot_longer(data = counts, cols = c(23:34), 
                                   names_to = "Taxa", values_to= "Density")
long_counts <- as.data.frame(long_counts)
str(long_counts)
long_counts$Taxa<-as.factor(long_counts$Taxa)
long_counts$Rel.abund <- long_counts$Density/ long_counts$dens.row.sums

is.na(long_counts)<-sapply(long_counts, is.infinite)
long_counts[is.na(long_counts)]<-0

# try with just logging the densities ("value"), still an issue of zeros but let's just look
#mod2<-lm(log(Density+0.001) ~Taxa+time_point*lake, data = long_counts)
#Anova(mod2, type=3)

long_counts$Time.cont<-as.numeric(long_counts$time_point)

# area plot
zoop_abundances_area <- ggplot(long_counts, aes(x=Time.cont, y=Rel.abund, fill=Taxa)) +
  geom_area() +
  facet_grid(cols = vars(lake)) +
  xlab("Sampling Time Point") +ylab("Relative Abundance")  +
  scale_fill_manual(values = c("firebrick3", "lightsalmon", "gold", "darkorange", "darkseagreen2", "mediumseagreen","lightskyblue","plum", "royalblue3", "palegoldenrod", "tan", "gray54")) + theme_bw()
 
zoop_abundances_area
dev.copy(pdf, "figures/Fig3.zoop_abundances_areaplot.pdf", height=6, width=15)
dev.off()

```

Run models to test for differences in zooplankton community, using Shannon diversity.  First look for the model that best explains the data using StepAIC, then run ANOVA on the best model.  
*Note* Interactions aren't approproate for these models since there is only n=1 in each lake/time combination. We therefore use Type II SS of main effects.
```{r stats on zoops, results='hide', collapse=TRUE}
# Shannon diversity
zoop.comm.div <- as.data.frame(vegan::diversity(counts[,c(23:34)], index="shannon"))
zoop.comm.div <- cbind(counts, zoop.comm.div)
names(zoop.comm.div)[47] <- "Shannon"

zoop.shan.lake.time.mod<-lm(Shannon ~ time_point + lake, data = zoop.comm.div)
zoop.shan.lake.time.aov<-Anova(zoop.shan.lake.time.mod, type=2)
# no effect of time or lake on Shannon diversity of zoop community samples (p> 0.425)

stepAIC(lm(Shannon ~ latitude + elevation_m + DO..perc + log(DOC) + chla_ug.L + pH + cond + temp..C, data = zoop.comm.div))

shan.zoop.env.mod<-lm(Shannon ~ latitude + cond + temp..C, data = zoop.comm.div)
shan.zoop.env.aov<-Anova(shan.zoop.env.mod, type=2)
plot(allEffects(lm(Shannon ~ temp..C, data = zoop.comm.div)))
```

- Best models are below for Shannon Diversity by Site x Time, and Shannon Diveristy x environmental conditons. 
- These are in *(TOP) Table S1 and S2*.
```{r best model zoopl RA}
zoop.shan.lake.time.aov
shan.zoop.env.aov
```


### Microbiome Data 
Import Phyloseq data from DADA pipeline. The data will need to be cleaned and organized so that it can be processed for downstram analysis.
- Modify the metadata to meet downstream needs
- Once data has been cleaned up, export new phyloseq objects and their parts
- Ultimate product is the non-rarified phyloseq object that can be processed in subsequent chunks
```{r loading in data and modifying metadata, results='hide', message='hide'}
##########################################################
detach("package:MicrobiotaProcess", TRUE) # reload later, has conflicts with phyloseq

# ps.prune is the final phyloseq object from pipeline, can load it in here...
### or load in the raw data and re-assemble

ps.prune.LOCAL<-readRDS("output/dada proc/ps.prune_local.RDS") #8056 taxa of 328 samples

# idiot check: make sure to remove all mitochondria and chloroplast taxonomic IDs
ps.prune.LOCAL <- subset_taxa(ps.prune.LOCAL, Family!= "Mitochondria" | 
                        is.na(Family) & Class!="Chloroplast" | is.na(Class))

# chloroplasts sneaking in under Order in some cyanobacteria, looks good @ 8056 ASVs
ps.prune.LOCAL <- subset_taxa(ps.prune.LOCAL, Order!="Chloroplast"| is.na(Class))

# make sure any ASVs in the data that not in > 1 sample, removed
ps.prune.LOCAL <- prune_taxa(taxa_sums(ps.prune.LOCAL) > 1, ps.prune.LOCAL) 
# few sneaked in, @ 8040 ASVs in 328 samples
# summarize_phyloseq(ps.prune.LOCAL) <<< use this for reporting

# order the lakes by North to South or upstream in phyloseq 
sample_data(ps.prune.LOCAL)$Lake <- factor(sample_data(ps.prune.LOCAL)$Lake, levels = 
                                           c("Cooney", "Blue", "Virginia", "Convict", "Serene", "Eastern Brook"))


# export metadata and play
metaD<-microbiome::meta(ps.prune.LOCAL)

# adjust caps
metaD<-metaD %>% 
  dplyr::rename(
    sequencing_ID = Sequencing_ID,
    sample_type = Sample_Type,
    organism = Organism,
    time_point= Time.Point,
    lake = Lake,
    sample_ID = Original_ID
  )

# correct NAs to be "water"
metaD$organism<-as.character(metaD$organism) # need to do this to change the level
metaD$organism[is.na(metaD$organism)] = "Water"
metaD$organism<-as.factor(metaD$organism)
metaD$organism<-factor(metaD$organism, levels =c("Bosmina", "Ceriodaphnia", "Daphnia", "Holopedium", 
                                             "Cyclopoid", "Calanoid", "Large Calanoid",
                                             "Water")) # back to factor

metaD$organism<-recode_factor(metaD$organism, "Large Calanoid" = "Calanoid")

# order
metaD$organism<-factor(metaD$organism, levels =c("Daphnia", "Holopedium", 
                                   "Calanoid", "Cyclopoid", "Water", "Bosmina", "Ceriodaphnia"))

# format metadata (have to redo this if loading back in)
make.fac<-c("sample_ID", "sequencing_ID", "sample_type", "time_point", "lake", "Plate", "Plate_name", "Well", "sample_control")

metaD[make.fac]<-lapply(metaD[make.fac], factor) # make all these factors

# create phy group (Cladocera vs Copepoda vs water) column
metaD$phy_group <- as.factor(ifelse(metaD$organism == "Calanoid" | 
                            metaD$organism == "Cyclopoid", "copepoda", 
                   ifelse(metaD$organism == "Daphnia" | 
                            metaD$organism == "Ceriodaphnia" | 
                            metaD$organism =="Bosmina" | 
                            metaD$organism == "Holopedium", "cladocera", 
                            "water")))

# create basin column
metaD$basin <- as.factor(ifelse(metaD$lake == "Eastern Brook" | 
                                metaD$lake == "Serene" | 
                                metaD$lake == "Convict", "South", 
                                "North"))

# create latitude column
metaD$latitude <- as.numeric(ifelse(metaD$lake == "Blue", "38.050952", 
                            (ifelse(metaD$lake == "Convict", "37.590094",
                            (ifelse(metaD$lake == "Cooney", "38.04806",
                            (ifelse(metaD$lake == "Eastern Brook", "37.431526", 
                            (ifelse(metaD$lake == "Serene", "37.438392" ,
                            (ifelse(metaD$lake == "Virginia", "38.046975",
                                  "0"))))))))))))

# create longitude column
metaD$longitude <- as.numeric(ifelse(metaD$lake == "Blue", "-119.270331", 
                            (ifelse(metaD$lake == "Convict", "-118.857422",
                            (ifelse(metaD$lake == "Cooney", "-119.27702",
                            (ifelse(metaD$lake == "Eastern Brook", "-118.742615", 
                            (ifelse(metaD$lake == "Serene", "-118.744386" ,
                            (ifelse(metaD$lake == "Virginia", "-119.265076",
                                  "0"))))))))))))
                            

##### Read counts and rarefaction curves
# Make a data frame with a column for the read counts of each sample
sample_sum_df<-as.data.frame(sample_sums(ps.prune.LOCAL))
colnames(sample_sum_df)<-"read.sum"
sample_sum_df$sampleNames<-rownames(sample_sum_df)

# are row names the same? if so, add them in
identical(rownames(sample_sum_df), rownames(metaD))
metaD$read.sum<-sample_sum_df$read.sum

# add in dates, as per metadata
metaD$date<-ifelse(metaD$time_point =="1" | metaD$lake =="Blue", "6/29/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Blue", "7/09/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Blue", "7/21/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Blue", "8/04/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Blue", "8/22/2022",
                    
                     ifelse(metaD$time_point =="1" | metaD$lake =="Convict", "6/25/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Convict", "7/07/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Convict", "7/22/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Convict", "8/05/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Convict", "8/23/2022",
                          
                     ifelse(metaD$time_point =="1" | metaD$lake =="Cooney", "6/29/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Cooney", "7/09/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Cooney", "7/21/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Cooney", "8/04/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Cooney", "8/22/2022", 
                                
                     ifelse(metaD$time_point =="1" | metaD$lake =="Eastern Brook", "6/26/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Eastern Brook", "7/06/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Eastern Brook", "7/22/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Eastern Brook", "8/09/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Eastern Brook", "8/23/2022",
                                   
                    ifelse(metaD$time_point =="1" | metaD$lake =="Serene", "6/26/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Serene", "7/06/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Serene", "7/22/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Serene", "8/09/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Serene", "8/23/2022",
                                  
                    ifelse(metaD$time_point =="1" | metaD$lake =="Virginia", "6/29/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Virginia", "7/09/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Virginia", "7/21/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Virginia", "8/04/2022",
                            "8/22/2022"
                            )))))))))))))))))))))))))))))

# adjust formatting to be Date (from)
metaD$date<-as.Date(mdy(metaD$date))

metaD<- metaD %>%
  dplyr::select(sequencing_ID, sampleNames, sample_ID, read.sum, time_point, date, lake, latitude, longitude, basin, sample_type, organism, Number.of.Organism, phy_group)

   
# the metadata above is now up to date with info necessary for downstream analysis
# now merge in the environmental data 

################################## ##################################
# read in the environmental data
env.metad<-read.csv("data/full.library.env.metaD.csv")

# make a column for 'Samplenames' and use this as rownames
env.metad$sampleNames<-env.metad$sequencing_ID
env.metad$sampleNames <- gsub("^.{0,3}", "", env.metad$sampleNames)
rownames(env.metad)<- env.metad$sampleNames

# remove the rows not found in the meta-meta data -- these are samples lost in dada2/controls
env.metad<-env.metad[(env.metad$sampleNames %in% metaD$sampleNames),]

env.metad<- env.metad %>%
  dplyr::select(sample_ID, sequencing_ID, sampleNames, sample_type, organism, Number.of.Organism, 
                time_point, lake, elevation_m, temp_C, chla_ug.L, pH, DO_perc, spc, cond, DOC, TDN, TDP)

## replace outliers in nutrient analysis
env.metad<-env.metad %>% 
  naniar::replace_with_na(replace = list(TDP = 4102.48))


################################## ##################################
# no merge, subset to make it easier to merge...
env.metad.reduce<-env.metad %>%
  dplyr::select(sampleNames, elevation_m, temp_C, chla_ug.L, pH, DO_perc, spc, cond, DOC, TDN, TDP)


# merge the two = final metadata, add in rownames for phyloseq merge
MetaD.SQ.Env<-merge(metaD, env.metad.reduce, by="sampleNames", all.x=TRUE)
rownames(MetaD.SQ.Env)<-MetaD.SQ.Env$sampleNames

# make new column for above or below treeline
MetaD.SQ.Env$elev.cat <- as.factor(ifelse(MetaD.SQ.Env$elevation_m < 2900, "Below", "Above"))

# are row names the same? if so, let's boogie
identical(rownames(sample_data(ps.prune.LOCAL)), rownames(MetaD.SQ.Env))
write.csv(MetaD.SQ.Env, "output/MetaD.SQ.Env.csv")

#updated metadata back into phyloseq
sample_data(ps.prune.LOCAL)<-MetaD.SQ.Env
PS.fin<- ps.prune.LOCAL

table(sample_data(PS.fin)$sample_type) #328 samples: 30 water, 298 zooplankton

# change to this order for later plot tests-- ultimately Bosmina and Ceriodaphnia are dropped nad large calanoids grouped into calanoids...
sample_data(PS.fin)$organism<-factor(sample_data(PS.fin)$organism, levels =c("Daphnia", "Holopedium", 
                                   "Calanoid", "Cyclopoid", "Water", "Bosmina", "Ceriodaphnia")) # back to factor

table(tax_table(PS.fin)[, "Kingdom"], exclude = NULL) # 289 Archaea, 7751 Bacteria
summarize_phyloseq(PS.fin)

########### ########### ########### ########### 
saveRDS(PS.fin, "output/dada proc/PS.fin.RDS") # save phyloseq
########### ########### ########### ########### 

#### export each element
OTU = as(otu_table(PS.fin), "matrix")
if(taxa_are_rows(PS.fin)){OTU <- t(OTU)}
# Coerce to data.frame
OTUdf = as.data.frame(OTU)

Tax.tab = as(tax_table(PS.fin), "matrix")
if(taxa_are_rows(PS.fin)){Tax.tab <- t(Tax.tab)}
# Coerce to data.frame
TAXdf = as.data.frame(Tax.tab)

Sam.tab = as(sample_data(PS.fin), "matrix")
if(taxa_are_rows(PS.fin)){Sam.tab <- t(Sam.tab)}
# Coerce to data.frame
SAMdf = as.data.frame(Sam.tab)

write.csv(OTUdf, "output/phyloseq output/OTUdf.nonrar.csv")
write.csv(TAXdf, "output/phyloseq output/TAXdf.nonrar.csv")
write.csv(SAMdf, "output/phyloseq output/SAMdf.nonrar.csv")

## with bosmina and ceriodaphnia still in dataset, and pre-rarefaction...
microbiome::summarize_phyloseq(PS.fin)
```


```{r total reads by species boxplots raw data, resuls='hide'}
#### Inspect the run: 
# what is the # of reads/sample? 
# A bit over inflated due to water being so deeply sequenced, but log-read sum gives us a good picture

# reads per sample
read.sample<-ggplot(MetaD.SQ.Env, aes(x=sample_type, y=read.sum, color=organism)) + geom_boxplot() + theme_bw() +
  scale_color_manual(values=c("coral2", "gold3", "mediumaquamarine", "springgreen4", 
                              "dodgerblue", "violet", "wheat3"))

### log reads per sample
log.read.sample<-ggplot(MetaD.SQ.Env, aes(x=organism, y=log(read.sum), color=organism)) + geom_boxplot() + theme_bw() +
  scale_color_manual(values=c("coral2", "gold3", "mediumaquamarine", "springgreen4", 
                              "dodgerblue", "violet", "wheat3"))

# read depth a bit low for zooplankton vs water, average depth is less than 2000 reads
ggplot(MetaD.SQ.Env[(MetaD.SQ.Env$sample_type=="Zooplankton"),], aes(x=sample_type, y=read.sum, color=organism)) +
  geom_boxplot() +  theme_bw() +
  scale_color_manual(values=c("coral2", "gold3", "mediumaquamarine", "springgreen4", 
                              "dodgerblue", "violet", "wheat3"))
```

- Now look at read depth and rarefaction curves to determine how to standardize sampling depth
```{r Read depth plots and rarefaction curves, eval=FALSE}

# Histogram of sample read counts
hist.depth <- ggplot(MetaD.SQ.Env, aes(read.sum)) + 
  geom_histogram(color="gold4", fill= "goldenrod2", binwidth = 10000) + 
  xlab("Read counts") +
  ggtitle("Sequencing Depth") + theme_classic()

hist.depth

# a different view
read_depth_violin <- ggplot(MetaD.SQ.Env, aes(x=1, y=read.sum)) +
  geom_violin() + ggtitle("Distribution of sample sequencing depth") + scale_y_log10() +
  xlab("Samples") + ylab("Read Depth") + geom_boxplot(width=0.1)

# Rarefaction curve
count_table_filt <- as.data.frame(otu_table(PS.fin))
rarecurve((count_table_filt), step=50, cex=0.5, xlim=c(0,100000), ylab = "ASVs", label=FALSE, col="gold4")
#abline(v = 500, lty = "dotted", col="red", lwd=2)

dev.copy(pdf, "figures/FigS2.rare.raw.pdf", height=4.5, width=7)
dev.off() 
```

Now we will subset the phyloseq objects and compare the influence of rarefaction for the phyloseq data.
- `PS.fin` is the final phyloseq object with all the data.  
- `PS.500` is rarefied to 500 read-depth (with low abundance samples removed)
- `PS.zoop` is rarefied zooplankton data (500 read depth)
- `PS.water` is rarefied water only (500 read depth)
```{r exploring what low read count samples do and what to rarefy to}
###### subset the PS objects for needs...
# final and NOT rarefied data: 
PS.fin


# rarified to 500 reads: PS.rar, then reclassify and reduce some taxa and name "PS.500'
# 44 samples removedbecause they contained fewer reads than `sample.size`
# 5460 OTUs were removed because they are no longer present
PS.rar = rarefy_even_depth(PS.fin, rngseed=111, sample.size=500, replace=F) 
PS.rar <- prune_taxa(taxa_sums(PS.rar) >0, PS.rar) # remove those asvs not in at least 1 sample again

table(sample_data(PS.rar)$sample_type) #284 samples: 28 water, 256 zooplankton, 1031 ASVs
summarize_phyloseq(PS.rar)


###### ###### ###### ###### ###### ###### ###### ###### ###### ###### ###### ###### 
# Compare full data, ordinations, and rarefied beta diversity
###### ###### ###### ###### ###### ###### ###### ###### ###### ###### ###### ###### 


###################### full data, non-rarified DCA ###################### 
# check beta diversity heterogeneity using DETRENDED CORRESPONDANCE ANALYSIS (DCA)
# can be measured by DCA (the length of the first DCA axis is a measure of heterogeneity
# DCA1 > 4 = the dataset is considered as heterogeneous and suitable for unimodal methods (RDA, CCA)
# DCA1 < 3 = it can be considered as homogeneous, suitable for linear methods (PCA, RDA)
# DCA1 =  3-4, both linear or unimodal methods are ok. After this decision, use either linear (PCA, RDA)
DCA.full<-ordinate(PS.fin, method='DCA') # DCA-axis1 length = 6
plot_ordination(PS.fin, DCA.full, color = "sample_type",  type='split')


###################### full data, non-rarified PCoA ###################### 
# distance based (db) PCoA (principal coordinate analysis)
# PCoA is preferred if wanting to interpret the absolute distances between communities in different samples.
PCoA.full <- ordinate(PS.fin, method = 'PCoA', distance = 'bray')
plot_ordination(PS.fin, PCoA.full, color = "sample_type")


###################### transformation based ordination (tb), ###################### 
# bypass the hetero/homogeneity issue, transform  species composition data by Hellinger transformation
# use either unconstrained (tb-PCA) or constrained (tb-RDA) approach
# Hellinger transformation in un-even community datasets better linearizes the distances among sites 
# ... and reduce the effect of dominant species in the site dissimilarity

####  tb-PCA, 
# Principal Components Analysis is an unconstrained method that does not use a distance matrix. 
# PCA directly uses the (transformed) microbial variables
# package will hellinger transform

library("MicrobiotaProcess") # note this has issues with phyloseq so need to unload it after
PCA.hell.nonrar <- get_pca(PS.fin, method="hellinger")
PCA.hell.nonrar.plot <- ggordpoint(PCA.hell.nonrar, biplot=TRUE, 
                      speciesannot=TRUE,
                      factorNames=c("organism"), ellipse=TRUE)

#### ####  ####  rarefied, manually hellinger transformed: Constrained ordination (tb-RDA)
# transform the rarified (500 reads) data and Hellinger transform
# convert the data to proportions and then take the square root
# do RDA/PCA  here, since non-distance based ordination

PS.rar.hell.test = transform_sample_counts(PS.rar, function(x) sqrt(x / sum(x)))
ord.hell.RDA <- ordinate(PS.rar.hell.test, method = 'RDA')

Hell.tb.RDA<- plot_ordination(
  physeq=PS.rar.hell.test, 
  ordination = ord.hell.RDA) +
  geom_point(aes(color = organism), size = 2) +    
  scale_color_manual(values=c("coral", "goldenrod2", "darkolivegreen3", "darkslategray4",
                              "dodgerblue", "violet", "wheat3")) +
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  ggtitle("ALL taxa: tb-RDA (Hell-500)") +
  theme_classic() 

#############################

# what samples have low reads
plot_ordination(PS.fin, PCoA.full, shape = "sample_type") +
  geom_point(aes(color = read.sum > 500))

Rare.culling<-plot_ordination(PS.fin, PCoA.full, color = "organism") +
  geom_point(aes(shape = read.sum > 500, size= read.sum > 500)) +
  scale_size_manual(values= c(6,2)) +
  scale_shape_manual(values=c(1, 16))+
  scale_color_manual(values=c("coral", "goldenrod2", "darkolivegreen3", "darkslategray4",
                              "dodgerblue", "violet", "wheat3")) +
  ggtitle("Samples culled by rarefaction") + theme_classic()


# plot the PCoA for non-rarified
Non.rar.PCoA<-plot_ordination(
  physeq = PS.fin,                                                   
  ordination = PCoA.full) +                                                
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral", "goldenrod2", "darkolivegreen3", "darkslategray4", 
                              "dodgerblue", "violet", "wheat3")) +
  ggtitle("ALL taxa-all data") +
  theme_classic() 

###### if rarefied in a distance based approach, PCoA
ord.rar <- ordinate(PS.rar, method = 'PCoA', distance = 'bray')
plot_ordination(PS.rar, ord.rar, color = "sample_type")

Rar.PCoA<- plot_ordination(
  physeq = PS.rar,                                                   
  ordination = ord.rar) +                                                
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral", "goldenrod2","darkolivegreen3", "darkslategray4",
                              "dodgerblue", "violet", "wheat3")) +
  ggtitle("ALL taxa- 500reads") +
  theme_classic() 



########### PCoA - unifrac/weighted
# distance based ordination on rarified data
# UniFrac is a β-diversity measure that uses phylogenetic information to compare environmental samples
# Unweighted UniFrac is more sensitive to differences in low-abundance features
# Weighted UniFrac for abundance and phylogeny, impact of low-abundance features is diminished, is useful for examining differences in community structure

# make phylogenetic tree and add to phyloseq if wanting to do unifrac weight/unweighted
library("ape")
random_tree = rtree(ntaxa(PS.rar), rooted=TRUE, tip.label=taxa_names(PS.rar))

# add in
phy_tree(PS.rar)<-random_tree

# unweighted since we have low abundance taxa
ord.unif <- ordinate(PS.rar, method = 'PCoA', distance ="unifrac")

PCoA.rar.UnW.unif<- plot_ordination(
  physeq=PS.rar, 
  ordination = ord.unif) +
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral", "goldenrod2", "darkolivegreen3", "darkslategray4",
                              "dodgerblue", "violet", "wheat3")) +
  ggtitle("ALL taxa: PCoA 500.UnW.unifrac") +
  theme_classic() 

# weighted for dominant taxa
ord.wunif <- ordinate(PS.rar, method = 'PCoA', distance ="wunifrac")

PCoA.rar.w.unif<- plot_ordination(
  physeq=PS.rar, 
  ordination = ord.wunif) +
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral", "goldenrod2", "darkolivegreen3", "darkslategray4", 
                              "dodgerblue", "violet", "wheat3")) +
  ggtitle("ALL taxa: PCoA 500.W.unifrac") +
  theme_classic() 

## inspect sample size effects for samples with more or less than average # individuals
Rare.number.org<-plot_ordination(PS.fin, PCoA.full, color = "Number.of.Organism") +
  geom_point(aes(color= Number.of.Organism, shape=organism), size=2) +
  scale_colour_gradient(low = "gold", high = "dodgerblue") +
  scale_shape_manual(values=c(16,17,18,1,2,3,4))+
  ggtitle("Organism number rarefaction") + theme_classic() 
Rare.number.org$layers <- Rare.number.org$layers[-1]


####
ordination.tests<-plot_grid(Rare.culling + guides(color="none"), 
                      Non.rar.PCoA +  guides(color="none"), 
                      Rar.PCoA,   
                      Rare.number.org,
                      PCoA.rar.w.unif +  guides(color="none"),
                      Hell.tb.RDA + guides(color="none"), ncol=3, rel_widths=c(5,4,4,5,4,4))

ordination.tests
dev.copy(pdf, "figures/Ordination.tests.pdf", height=10, width=15)
dev.off()
#######

# unload 'MicrobiotaProcess'
detach("package:MicrobiotaProcess", TRUE)
```

After running the tests of rarifying data transformation influence our data (and seeing the low sample size for Bosmina and Ceriodaphnia), we will proceed with a rarified dataset and drop these two species. The variation in DCA shows we can proceed with assumption of heterogeneous distributions and use RDA/CCA, use PCoA/NMDS with Bray-Curits, or proceed with rank-based ordination approached (NMDS) with fewer axises.

We will use rarified data, and demonstrate ordination differences using PCoA with unweighted unifrac distance (accounting for abundance and phylogeny) and NMDS-Bray Crutis for maximal distance between groups
```{r final phyloseq, results='hide'}
################### final phyloseq object based on the above

# give it a redo now that we know we don't want other levels here...

# this PS.500 is used for the richness data -- non-rarefied
# first, use the PS.fin data and remove the samples no longer wanted
PS.rm = subset_samples(PS.fin, !organism=="Bosmina" & !organism=="Ceriodaphnia") # removes 12 samples
PS.rm <- prune_taxa(taxa_sums(PS.rm) >1, PS.rm) # remove those asvs not in > 1 sample (again)

# removes 35 samples and 5467 OTUs
PS.500 = rarefy_even_depth(PS.rm, rngseed=111, sample.size=500, replace=F) 
PS.500 <- prune_taxa(taxa_sums(PS.500) >0, PS.500) # remove those asvs not in > 0 sample, in the final data
# 1651 taxa and 281 samples

table(sample_data(PS.500)$sample_type)
#281 samples: 28 water, 253 zooplankton, 1651 ASVs

# check sample data
table(sample_data(PS.500)$organism) 
# Calanoid    Daphnia Holopedium  Cyclopoid      Water 
#       74         98          8         73         28 

# set order of organims
sample_data(PS.500)$organism<-factor(sample_data(PS.500)$organism, 
                                     levels =c("Daphnia", "Holopedium", 
                                            "Calanoid", "Cyclopoid", "Water")) # back to factor
summarize_phyloseq(PS.500)
##########################

#### export each element
OTU = as(otu_table(PS.500), "matrix")
if(taxa_are_rows(PS.500)){OTU <- t(OTU)}
# Coerce to data.frame
OTUdf.rar = as.data.frame(OTU)

Tax.tab = as(tax_table(PS.500), "matrix")
if(taxa_are_rows(PS.500)){Tax.tab <- t(Tax.tab)}
# Coerce to data.frame
TAXdf.rar = as.data.frame(Tax.tab)

Sam.tab = as(sample_data(PS.500), "matrix")
if(taxa_are_rows(PS.500)){Sam.tab <- t(Sam.tab)}
# Coerce to data.frame
SAMdf.rar = as.data.frame(Sam.tab)

write.csv(OTUdf.rar, "output/phyloseq output_rarified/OTUdf.rar.csv")
write.csv(TAXdf.rar, "output/phyloseq output_rarified/TAXdf.rar.csv")
write.csv(SAMdf.rar, "output/phyloseq output_rarified/SAMdf.rar.csv")

########### ########### ########### ########### 
saveRDS(PS.500, "output/dada proc/PS.500.RDS") # save phyloseq
########### ########### ########### ########### 


# pull the OTU table and sample data from phyloseq so we can run these analyses in vegan
# if needed, this is quite useful....

# convert the sample_data() within a phyloseq object to a vegan compatible data object
pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
psotu2veg <- function(physeq) {
  OTU <- otu_table(physeq)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}
```


Richness (observed and shannon) using PS.500 
- look at richness by sample type (water vs. zooplankton)
- look at how time and site effect richness
- how does richness differ for individual taxa compared to water (it is much lower...)
```{r alpha diversity - plot_richness}
#richness by organism (or sample type i.e, water) -- not transformed, but rarefied

## plots
richness.plot.samples <- plot_richness(PS.500, x="sample_type", measures=c("Observed", "Shannon"), color="sample_type") +
  labs(y= "Alpha Diversity Measure", x = "Sample Type") + theme_bw() + 
  geom_boxplot(aes(fill=sample_type), alpha=0.7) +
  geom_point(aes(color=sample_type), alpha=0.5) +
  scale_color_manual(values = c("lightsteelblue3", "navajowhite3"))+ guides(color="none") +
  scale_fill_manual(values = c("lightsteelblue3", "navajowhite2"))+ guides(fill="none") +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12))

richness.plot.samples$layers <- richness.plot.samples$layers[-1] # remove the first layer of points 
richness.plot.samples

#####
#richness by time_point
alpha.plot.time <- plot_richness(PS.500, x="time_point", measures=c("Shannon")) + 
  labs(y= "Shannon Diversity", x = "Sampling Time Point") + theme_bw() + 
  geom_point(color="gray15", alpha=0.5) +
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(fill= "gray85", alpha=0.5) + 
  theme(strip.text.x = element_text(size = 12))

alpha.plot.time$layers <- alpha.plot.time$layers[-1] # remove the first layer of points 
alpha.plot.time

#Shannon diversity by Lake
alpha.plot.location <- plot_richness(PS.500, x="lake", measures=c("Shannon")) +
  labs(y= "Shannon Diversity", x = "Lake") + theme_bw() + 
  geom_point(color="gray15", alpha=0.5) +
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(fill= "gray15", alpha=0.5)+ 
  theme(strip.text.x = element_text(size = 12)) +
  theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

alpha.plot.location$layers <- alpha.plot.location$layers[-1] # remove the first layer of points 
alpha.plot.location

# group the above two plots
alpha.time.lake<-plot_grid(alpha.plot.time, alpha.plot.location)

#####
#richness by organisms (zoop species and water)
#Shannon
richness.spp.shan <- plot_richness(PS.500, x="organism", color="organism", measures=c("Shannon")) +
  labs(y= "Alpha Diversity", x = "Zooplankton and Water Samples") + 
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(aes(fill=organism), alpha=0.4) + guides(fill="none") + guides(color="none")+
  geom_point(alpha=0.5) +
  scale_fill_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  theme(axis.text.x=element_blank(), strip.text.x = element_text(size = 12)) + theme_bw() +
  theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

richness.spp.shan$layers <- richness.spp.shan$layers[-1] # remove the first layer of points 
richness.spp.shan

#richness by organisms (zoop species and water)
#observed
richness.spp.obs <- plot_richness(PS.500, x="organism", color="organism", measures=c("Observed")) +
  labs(y= "Alpha Diversity", x = "Zooplankton and Water Samples") + 
  coord_cartesian(ylim=c(0, 150))+
  geom_boxplot(aes(fill=organism), alpha=0.4) + guides(fill="none") + guides(color="none")+
  geom_point(alpha=0.5) +
  scale_fill_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  theme(axis.text.x=element_blank(), strip.text.x = element_text(size = 12)) + theme_bw()+
   theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

richness.spp.obs$layers <- richness.spp.obs$layers[-1] # remove the first layer of points 
richness.spp.obs


#richness by organisms (zoop species and water)
#Shannon
richness.spp.time.shan <- plot_richness(PS.500, x="time_point", color="organism", measures=c("Shannon")) +
  labs(y= "Alpha Diversity", x = "Sampling Time Points") + 
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(aes(fill=organism), alpha=0.4, outlier.size=0.8) + guides(fill="none") + guides(color="none")+
  geom_point(alpha=0.5, size=0.8, position=position_dodge(width = 0.75)) +
  scale_fill_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  theme(axis.text.x=element_blank(), strip.text.x = element_text(size = 12)) + theme_bw() +
  theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

richness.spp.time.shan$layers <- richness.spp.time.shan$layers[-1] # remove the first layer of points 
richness.spp.time.shan


#richness by organisms (zoop species and water)
#observed
richness.spp.lake.shan <- plot_richness(PS.500, x="lake", color="organism", measures=c("Shannon")) +
  labs(y= "Alpha Diversity", x = "Lakes") + 
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(aes(fill=organism), alpha=0.4, outlier.size=0.8) + guides(fill="none") + guides(color="none")+
  geom_point(alpha=0.5, size=0.8, position=position_dodge(width = 0.75)) +
  scale_fill_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  theme(axis.text.x=element_blank(), strip.text.x = element_text(size = 12)) + theme_bw()+
   theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

richness.spp.lake.shan$layers <- richness.spp.lake.shan$layers[-1] # remove the first layer of points 
richness.spp.lake.shan

extract.legend.richness <- get_legend(
  # create some space to the left of the legend
  richness.spp.lake.shan + theme(legend.box.margin = margin(0, 0, 0, 10)))


# group the above two plots
alpha.organism.time.lake<-plot_grid(richness.spp.time.shan, 
                                    richness.spp.lake.shan,
                                    ncol=2, labels=c("A", "B"), rel_widths = c(5,5.2))

alpha.organism.time.lake
dev.print(pdf, "figures/Figx.shannon.interaction.pdf", width=10, height=5)
dev.off()
```

Use Shannon Diversity and Alpha richness to test differences across sampling factors
```{r ASV Richness and Diversity stats}
# some stats
Shan.rich<-estimate_richness(PS.500, measures =c("Observed", "Shannon"))
rownames(Shan.rich) <- substr(rownames(Shan.rich),2,nchar(rownames(Shan.rich))) # to remove the "X"
Shan.rich.df<-merge(data.frame(sample_data(PS.500)), Shan.rich, by = "row.names") # merge 

#### SHANNON fully orthogonal
# full 3 way for shannon
rich.shan.all<- lm(Shannon~ time_point*lake*organism, data=Shan.rich.df)
Anova(rich.shan.all) #time*lake significant
plot(allEffects(rich.shan.all))

emmeans(rich.shan.all, ~time_point*lake*organism, calc = c(n = ".wgt."))
with(Shan.rich.df, table(time_point, lake, organism))

rich.test.lake.full<-emmeans(rich.shan.all, pairwise~time_point|organism)
multcomp::cld(rich.test.lake.full, Letters=letters)

#### SHANNON
# straight 3 way no-interaction (bc of missing samples post filtering)
rich.shan.reduced<- lm(Shannon~ time_point+lake+organism, data=Shan.rich.df)
Anova(rich.shan.reduced, type=2) #time*lake significant
plot(allEffects(rich.shan.reduced))

#  posthoc just looking at Lake (p=<0.001)
rich.test.lake<-emmeans(rich.shan.reduced, pairwise~lake)
multcomp::cld(rich.test.lake, Letters=letters)
# lake          emmean    SE  df lower.CL upper.CL .group
# Serene          1.94 0.120 274     1.70     2.17  a    
# Eastern Brook   2.07 0.115 274     1.85     2.30  a   
# Virginia        2.47 0.108 274     2.26     2.69    b  
# Convict         2.57 0.121 274     2.33     2.81    b  
# Cooney          2.68 0.109 274     2.47     2.89    b  
# Blue            2.71 0.106 274     2.50     2.91    b 


rich.test.group<-emmeans(rich.shan.reduced, pairwise~organism)
multcomp::cld(rich.test.group, Letters=letters)

#organism   emmean     SE  df lower.CL upper.CL .group
#Calanoid     2.03 0.0648 270     1.90     2.16  a    
#Holopedium   2.12 0.2062 270     1.72     2.53  ab   
#Cyclopoid    2.19 0.0667 270     2.06     2.33   b   
#Daphnia      2.41 0.0565 270     2.30     2.52   b   
#Water        3.99 0.1021 270     3.79     4.19    c  


#### OBSERVED straight 3 way no-interaction (bc of missing samples post filtering)
rich.obs.all<- lm(Observed~ time_point+lake+organism, data=Shan.rich.df)
Anova(rich.obs.all, type=2) #lake and organism significant
plot(allEffects(rich.obs.all))

#  posthoc just looking at groups
obsrich.test.group<-emmeans(rich.obs.all, pairwise~organism)
multcomp::cld(obsrich.test.group, Letters=letters)

#organism   emmean     SE  df lower.CL upper.CL .group
# Holopedium   31.8 5.34 267     21.3     42.3  ab   
# Daphnia      33.3 1.47 267     30.4     36.2  a    
# Calanoid     37.0 1.68 267     33.7     40.3  ab   
# Cyclopoid    41.3 1.75 267     37.8     44.7   b   
# Water        98.5 2.64 267     93.3    103.7    c

```

Now testing effects of environmental variables on Shannon diversity.
```{r Shannon and environ}
# testing effects of environmental variables on Shannon diversity
# full df = 'Shan.rich.df'

# separate by water or zooplankton
Shan.water.df<-Shan.rich.df[(Shan.rich.df$sample_type=="Water"),]
Shan.zoop.df<-Shan.rich.df[(Shan.rich.df$sample_type=="Zooplankton"),]
 
###### Compute correlation at 2 decimal places
# matrix for environmental conditions
Shan.rich.df.env<-Shan.rich.df %>% dplyr::select(latitude, elevation_m, temp_C, DOC, chla_ug.L, 
                                                 pH, DO_perc, cond, Shannon)
Shan.rich.df.env$elevation_m<-as.numeric(Shan.rich.df.env$elevation_m)

corr_matrix = round(cor(Shan.rich.df.env), 2)

# Compute and show the results
ggcorrplot(corr_matrix, hc.order = TRUE, type = "lower",
          lab = TRUE)
# strong negative correlation between elevation- cond and cond-chla (<0.69-0.73)


####### water first
Shan.water.df$elevation_m<-as.numeric(Shan.water.df$elevation_m)

# model 
a.div.mod.wat <- lm(Shannon ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + pH + DO_perc + cond, data = Shan.water.df)
a.div.AIC.wat <- MASS::stepAIC(a.div.mod.wat, direction = "both")

# best model with AIC -74.9 ====. Shannon ~ elevation_m + cond
shan.mod.wat<-lm(Shannon ~ elevation_m + cond, data = Shan.water.df)
summary(shan.mod.wat)
shan.aov.wat<-Anova(shan.mod.wat, type=2) # near significant for both 0.051 amnd 0.08

# elevation_m, laitude, cond: NS
summary(lm(Shannon ~ elevation_m, data = Shan.water.df)) # NS
summary((lm(Shannon ~ latitude, data = Shan.water.df))) # NS
summary(lm(Shannon ~ DO_perc, data = Shan.water.df)) # NS
summary(lm(Shannon ~ cond, data = Shan.water.df)) # NS




# summary By for median, mean, and standard error of the Shannon diversity by lake
lake_div.wat <- summaryBy(Shannon ~ lake, 
                          data = Shan.water.df, 
                          FUN = c(median, mean, sd))

############ zooplankton alone
# run zooplankton
Shan.zoop.df$elevation_m<-as.numeric(Shan.zoop.df$elevation_m)

# model 
a.div.mod.zoop <- lm(Shannon ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + pH + DO_perc + cond, data = Shan.zoop.df)
a.div.AIC.zoop <- MASS::stepAIC(a.div.mod.zoop, direction = "both")

# best model with AIC -300.46 ==== latitude + log(DOC) + chla_ug.L + pH + DO_perc + cond
shan.mod.zoop<-lm(Shannon ~ latitude + log(DOC) + chla_ug.L + pH + DO_perc + cond, data = Shan.zoop.df)
summary(shan.mod.zoop)
shan.aov.zoop<-Anova(shan.mod.zoop, type=2)
plot(allEffects(shan.mod.zoop)) # shannon increases with elevation, DO, cond, log(DOC)


# latitude signif
summary(lm(Shannon ~ latitude, data = Shan.zoop.df)) #R2=0.245
anova(lm(Shannon ~ latitude, data = Shan.zoop.df)) # p<0.001 alone

#DO 
summary(lm(Shannon ~ DO_perc, data=Shan.zoop.df)) # adjust-R2=0.172
anova(lm(Shannon ~ DO_perc, data = Shan.zoop.df)) #p<0.001

#DOC
summary(lm(Shannon ~ log(DOC), data=Shan.zoop.df)) # adjust-R2=0.004
shan.DOC.zoop<-anova(lm(Shannon ~ log(DOC), data = Shan.zoop.df)) #p=0.148

#cond
summary(lm(Shannon ~ cond, data=Shan.zoop.df)) # adjust-R2=0.160
shan.DOC.zoop<-anova(lm(Shannon ~ cond, data = Shan.zoop.df)) #p=0.148

### plots
lat.shan.plot.zoop<-ggplot(Shan.rich.df, aes(x=latitude, y= Shannon, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylab("Shannon diversity")+
  xlab("Latitude (degrees N)")+
  theme_classic() +
  annotate(geom="text", x=37.7, y=0.1, label="Zoop. Adjusted-R2=0.245, p<0.001", size=3,
           color="black")+
  annotate(geom="text", x=37.7, y=4.2, label="Water NS", size=3,
           color="black")

DO.shan.plot.zoop<-ggplot(Shan.rich.df, aes(x=DO_perc, y= Shannon, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylab("Shannon diversity")+
  xlab("Dissolved Oxygen (%)")+
  theme_classic() +
  annotate(geom="text", x=80, y=0.1, label="Zoop. Adjusted-R2=0.172, p<0.001", size=3,
           color="black") +
annotate(geom="text", x=70, y=4.1, label="Water NS", size=3,
           color="black")

cond.shan.plot.zoop<-ggplot(Shan.rich.df, aes(x=cond, y=Shannon, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylab("Shannon diversity")+
  xlab("Conductivity (uS/cm)") +
  theme_classic() +
  annotate(geom="text", x=125, y=0.1, label="Zoop. Adjusted-R2=0.160, p<0.001", size=3,
           color="black")+
  annotate(geom="text", x=125, y=4.1, label="Water NS", size=3,
           color="black")

DOC.shan.plot.zoop<-ggplot(Shan.rich.df, aes(x=log(DOC), y= Shannon, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylab("Shannon diversity")+
  xlab("log(Dissolved organic carbon) (mg/L)")+
  theme_classic() +
  annotate(geom="text", x=3, y=0.1, label="Water and Zoop NS", size=3,
           color="black")

## get legend
extract.legend.Shan <- get_legend(
  # create some space to the left of the legend
  cond.shan.plot.zoop + theme(legend.box.margin = margin(0, 0, 0, 10)))
 
multregres.plots.shannon.env.wat.zoop<-plot_grid(lat.shan.plot.zoop + guides(color="none", fill="none"),
                                             DO.shan.plot.zoop + guides(color="none", fill="none"),
                                             cond.shan.plot.zoop + guides(color="none", fill="none"),
                                             extract.legend.Shan,
                            ncol=3, nrow=1, labels=c("A", "B", "C", ""), rel_widths=c(5,5,5,2))

# summary By for median, mean, and standard error of the Shannon diversity by lake
lake_div <- summaryBy(Shannon ~ lake, data = Shan.rich.df, FUN = c(median, mean, sd))
```

For ordination use PCoA [rarifed data, db-Bray Curtis], compare with unweighted unifrac [rarified with phylogenetic info]
- perform PCoA on all samples
- perform PCoA on rarified 500 reads 
- perform PCoA on rarified-hellinger transformed
```{r PCoA and W-Unifrac final}

###### check the # of samples in each taxa
table(sample_data(PS.500)$organism)
table(sample_data(PS.500)$sample_type) 
#284 samples: 28 water, 253 zooplankton 

# rarefied non-transformed
set.seed(341)
ord.500<- ordinate(PS.500, method = 'PCoA', distance ="bray")

#########
#PCoA Bray
PCoA.rar.bray<- plot_ordination(
  physeq=PS.500, 
  ordination = ord.500) +
  geom_point(aes(color = organism, shape=sample_type), size = 1.5) +
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral2", "goldenrod2", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_shape_manual(values= c(1,16))+
  ggtitle("Bray-Curtis") +
  theme_classic()
PCoA.rar.bray$layers <- PCoA.rar.bray$layers[-1]

library("colorBlindness")
cvdPlot(PCoA.rar.bray) # see that the pallete is colorblind friendly


####### 
# make phylogenetic tree and add to phyloseq if wanting to do unifrac weight/unweighted
set.seed(341)
random_tree = rtree(ntaxa(PS.500), rooted=TRUE, tip.label=taxa_names(PS.500))

# add in, use unweighted since we have low abundance taxa
phy_tree(PS.500)<-random_tree

set.seed(341)
ord.unif <- ordinate(PS.500, method = 'PCoA', distance ="unifrac") # unweighted

PCoA.rar.unw.unif<- plot_ordination(
  physeq=PS.500, 
  ordination = ord.unif) +
  geom_point(aes(color = organism, shape=sample_type), size = 1.5) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral2", "goldenrod2", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_shape_manual(values= c(1,16))+
  ggtitle("Unweighted Unifrac") +
  theme_classic() 
PCoA.rar.unw.unif$layers <- PCoA.rar.unw.unif$layers[-1]


##### for comparison , see weighted unifrac
set.seed(341)
ord.Wunif <- ordinate(PS.500, method = 'PCoA', distance ="wunifrac")

PCoA.rar.W.unif<- plot_ordination(
  physeq=PS.500, 
  ordination = ord.Wunif) +
  geom_point(aes(color = organism, shape=sample_type), size = 1.5) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral2", "goldenrod2", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_shape_manual(values= c(1,16))+
  ggtitle("Weighted Unifrac") +
  theme_classic() 
PCoA.rar.W.unif$layers <- PCoA.rar.W.unif$layers[-1]


#########
extract.legend.PCoA <- get_legend(
  # create some space to the left of the legend
  PCoA.rar.W.unif + theme(legend.box.margin = margin(0, 0, 0, 10)))

# make combined plot
PCoA.and.Unifracs<-plot_grid(PCoA.rar.bray +  guides(color="none", shape="none"),
                            PCoA.rar.W.unif + guides(color="none",  shape="none"), 
                            PCoA.rar.unw.unif +  guides(color="none",  shape="none"), extract.legend.PCoA, 
                            ncol=4, rel_widths=c(5,5,5,2), labels = c("A","B", "C"))

PCoA.and.Unifracs
dev.copy(pdf, "figures/FigS5.PCoA.Unifrac.pdf", height=4, width=12)
dev.off()
```


Run PERMANOVA on the dissimilarity. The code here is meant to give PERMANOVA tests of main effects, tests of environmental effects in Permanova, and tests of betadispersion for different groups.
- first, we will run analysis for all samples
- then for zooplankton alone
- then for water alone
```{r metaMDS, scores, and PERMANOVAS from BC distance matrices}
####################################
# first let's look at PERMANOVA and NDMS for the full dataset
# rarefied
set.seed(1000)
bc_500 <- phyloseq::distance(PS.500, method = "bray") # rarefied

# run NMDS on Bray distance object (defined above for 'bc_500'), this is
nmds.rar <- metaMDS(bc_500,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)


# get NMDS 1-3, we will use this later in NMDS plots
scores.full <- as.data.frame(scores(nmds.rar, display = "sites"))
scores.1.2<-scores.full[-3] # subset to just give the 2 axis space for NMDs1-2

nmds.rar.df <- merge(sample_data(PS.500), scores.1.2, by = 'row.names', all = TRUE)

# pull out the sample data for PERMANOVA
# all.sam.df <- pssd2veg(PS.500) 

# make the plot to visualize it (rough)
ggplot(nmds.rar.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = organism), size= 2) +
  #coord_fixed(ratio=1.6) + 
  stat_ellipse(level=0.95, linetype = 2, linewidth=0.6, aes(x = NMDS1, y = NMDS2, color=organism)) +
  ggtitle("all samples")


# use this #
### separate by sample_type
####################################
wat.phylo <-subset_samples(PS.500, sample_type=="Water")
wat.phylo <- prune_taxa(taxa_sums(wat.phylo) >0, wat.phylo) # remove those asvs with 0 sum
wat.rar.df <- sample.data.frame(wat.phylo)

set.seed(1000)
bc_wat <- phyloseq::distance(wat.phylo, method = "bray") # rarefied
env.model.wat <- adonis2(bc_wat ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = wat.rar.df, permutations = 999, by="terms")
env.model.wat # latitude is best (0.19), elevation (0.17), cond (0.08)

####################################
zoop.phylo <-subset_samples(PS.500, sample_type=="Zooplankton")
zoop.phylo <- prune_taxa(taxa_sums(zoop.phylo) >0, zoop.phylo) # remove those asvs with 0 sum
zoop.rar.df <- sample.data.frame(zoop.phylo)

set.seed(1000)
bc_zoop <- phyloseq::distance(zoop.phylo, method = "bray") # rarefied
env.model.zoop <- adonis2(bc_zoop ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = zoop.rar.df, permutations = 999, by="terms")
env.model.zoop # latitude is best (0.06), elevation (0.04), DO (0.02)

```

- distance to centroids
```{r beta distance to centroids}
######
# lets look at beta diversity and distance to centroid a bit: (zoop species and water)
# using the rarefied here since beta diversity metric
set.seed(10000)
bc_500.bd <- phyloseq::distance(PS.500, method = "bray") # rarefied 
bc_beta.df <- as.data.frame(sample_data(PS.500))

# one approach
bd<-betadisper(bc_500.bd, bc_beta.df$organism, type="centroid")
permutest(bd, permutations = 99)
anova(bd)
plot(bd)
boxplot(bd)
TukeyHSD(bd)

par(mar=c(5,7,3,1)) 
plot(TukeyHSD(bd), cex.axis=0.7, las=1)

bc_betadisp<-plot(bd, ellipse = TRUE, hull = FALSE, label.cex = 0.4, main="Distance to Centroids")

sqrt(dist(bd$centroids[,bd$eig>0])^2 - dist(bd$centroids[,bd$eig<0]^2))

#
bd$group.distances # this is Average distance to centroid for each group
scores(bd, 1:2, display = "centroids") # this is the centroid for each group
bd.df <- data.frame(Distance_to_centroid = bd$distances, organism=bd$group) # dataframe


######## ######## ######## ######## 
######## let's use another approach for NMDS only
# calculate mean distances to centroid manually (use 'grp.median' if using not using centroid in betadisper)
groups<-nmds.rar.df$organism

df=NULL
for(gr in unique(groups)) {
  organism=print(gr)
  grp.mean = colMeans(scores.1.2[groups == gr,])
  # calculate the euclidean distance between centroid point and all other points
  dist.to.centr = apply(scores.1.2[groups == gr,], 1, function(xx) {sqrt(sum((xx - grp.mean)^2))})
  info = print(paste0('all axes mean dist to centroid: ', round(mean(dist.to.centr),4) )) # get the mean of all distances
  df = rbind(df, data.frame(organism, dist.to.centr))
}

bdisp.df<-df
bdisp.df$organism<-as.factor(bdisp.df$organism)
bdisp.df$organism<-factor(bdisp.df$organism, levels =c("Daphnia", "Holopedium", 
                                   "Calanoid", "Cyclopoid", "Water"))
bdisp.df$dist.to.centr<-as.numeric(bdisp.df$dist.to.centr)

### box plot for distance
beta.dist.box<- ggplot(data=bdisp.df, aes(x=organism, y=dist.to.centr, color=organism)) +
  geom_boxplot(aes(fill=organism), alpha=0.4) + guides(fill="none") +
  geom_point(alpha=0.5) + 
  ylim(0,1) +
  guides(color=guide_legend(title=""))+ xlab("Sample Type")+ ylab("Distance to centroid") +
  ggtitle("Beta diversity") +
  scale_fill_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  theme(axis.text.x=element_blank()) + theme_bw()+
   theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))


# test of distance to centroids
beta.mod<-lm(dist.to.centr ~ organism, data = bdisp.df)
posthoc.betad<-emmeans(beta.mod, pairwise~organism)
multcomp::cld(posthoc.betad, Letters=letters)

#organism   emmean     SE  df lower.CL upper.CL .group
# Water       0.161 0.0603 276   0.0426    0.280  a    
# Daphnia     0.352 0.0322 276   0.2890    0.416   b   
# Cyclopoid   0.474 0.0373 276   0.4008    0.548   b   
# Holopedium  0.475 0.1127 276   0.2529    0.697  abc  
# Calanoid    0.707 0.0371 276   0.6336    0.780    c 
 
# notused, but just in case...
kruskal_test(dist.to.centr ~ organism, data = bdisp.df)
dunn_test(dist.to.centr ~ organism, data = bdisp.df)
```

- combine the plots
- multipanel alpha diversity plot from above with beta diversity distance to centroid
```{r combine alpha and distance to centroids}
# combine with richness
alpha.beta.rich.plot<-plot_grid(alpha.time.lake, richness.spp.shan, 
                           richness.spp.obs + guides(color="none", fill="none"), 
                           beta.dist.box + guides(color="none", fill="none"),
                      ncol=4, rel_widths=c(12,6,6,6))
alpha.beta.rich.plot
dev.copy(pdf, "figures/Fig4.alpha.beta.rich.plot.pdf", height=6, width=16)
dev.off()
```

- PERMANOVA with Time, Lake, Organisms
```{r full Permanova}
###############
# Permanova with 3 methods
# rarefied
bc_500 <- phyloseq::distance(PS.500, method = "bray") # rarefied
unif_500 <- phyloseq::distance(PS.500, method = "unifrac") # rarefied
wunif_500 <- phyloseq::distance(PS.500, method = "wunifrac") # rarefied

# run PERMANOVA with Time, Lake, Organisms
# Bray-Curtis from phyloseq: bc_500
set.seed(781)
broad.model.zoop <- adonis2(bc_500 ~ time_point*lake*organism, 
                            data = nmds.rar.df, method="bray", permutations = 999, by="terms")
broad.model.zoop

#test for beta dispersion
anova(betadisper(bc_500, nmds.rar.df$lake)) #  not-signif (barely)
anova(betadisper(bc_500, nmds.rar.df$time_point)) #  significant *barely
anova(betadisper(bc_500, nmds.rar.df$organism)) # very significant

pairwise.adonis(bc_500, factors=nmds.rar.df$time_point, p.adjust.m="holm")
pairwise.adonis(bc_500, factors=nmds.rar.df$lake, p.adjust.m="holm")
pairwise.adonis(bc_500, factors=nmds.rar.df$organism, p.adjust.m="holm")


##### 
### WEIGHTED Unifrac from phyloseq: wunif_500
set.seed(71)
broad.model.zoop.wunif <- adonis2(wunif_500 ~ time_point*lake*organism, 
                            data = nmds.rar.df, method="wunifrac", permutations = 999, by="terms")
broad.model.zoop.wunif

#test for beta dispersion
anova(betadisper(wunif_500, nmds.rar.df$lake)) #  not-signif (barely)
anova(betadisper(wunif_500, nmds.rar.df$time_point)) #  significant *barely
anova(betadisper(wunif_500, nmds.rar.df$organism)) # very significant

pairwise.adonis(wunif_500, factors=nmds.rar.df$time_point, p.adjust.m="holm")
pairwise.adonis(wunif_500, factors=nmds.rar.df$lake, p.adjust.m="holm")
pairwise.adonis(wunif_500, factors=nmds.rar.df$organism, p.adjust.m="holm")


##### 
### UNWEIGHTED Unifrac from phyloseq: unif_500
set.seed(71)
broad.model.zoop.UNWunif <- adonis2(unif_500 ~ time_point*lake*organism, 
                            data = nmds.rar.df, method="unifrac", permutations = 999, by="terms")
broad.model.zoop.UNWunif

#test for beta dispersion
anova(betadisper(unif_500, nmds.rar.df$lake)) #  not-signif (barely)
anova(betadisper(unif_500, nmds.rar.df$time_point)) #  significant *barely
anova(betadisper(unif_500, nmds.rar.df$organism)) # very significant

pairwise.adonis(unif_500, factors=nmds.rar.df$time_point, p.adjust.m="holm")
pairwise.adonis(unif_500, factors=nmds.rar.df$lake, p.adjust.m="holm")
pairwise.adonis(unif_500, factors=nmds.rar.df$organism, p.adjust.m="holm")

```


#### By-group NMDS and envfit

Run NMDS with envfit for water, daphnia, calanoids, cyclopoids to better understand how environment is impacting these host-microbiomes. We will run NMDS and fit env.fit, perform PERMANOVA, test betadispersion, and test env conditions on BC matrix. 

- First, run  *Daphnia*
- Holopedium too few to be informative here
```{r daph env NMDS}
# run same analysis but subset to only be cladocerans
# sample data and OTU table for vegan
# subset
daph.phylo<-subset_samples(PS.500, organism=="Daphnia")
daph.phylo <- prune_taxa(taxa_sums(daph.phylo) >0, daph.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied
set.seed(1000)
bc_daph <- phyloseq::distance(daph.phylo, method = "bray") # rarefied

# run NMDS on Bray distance object (defined above for 'bc_500.hell'), this is
set.seed(520)
nmds.daph.rar <- metaMDS(bc_daph,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

# get NMDS 1-3
sp.daph.scores <- as.data.frame(scores(nmds.daph.rar, display = "sites"))
nmds.daph.rar.df <- merge(sample_data(daph.phylo), sp.daph.scores, by = 'row.names', all = TRUE)

#### pick envr. data and run the envfit function
env.daph<- nmds.daph.rar.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.daph <- envfit(nmds.daph.rar, env.daph, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

### grab vectors and put into df
spp.daph.scrs <- as.data.frame(scores(vf.daph, display = "vectors")) *ordiArrowMul(vf.daph)
spp.daph.scrs.test <- cbind(spp.daph.scrs, r=vf.daph$vectors[2], pvals=vf.daph$vectors[4], 
                       Envfact = rownames(spp.daph.scrs))
names(spp.daph.scrs.test)[names(spp.daph.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.daph.scrs.signif<-spp.daph.scrs.test[spp.daph.scrs.test$pvals< 0.05, ]

# see color hexadecimal
# Hexadecimal color specification 
brewer.pal(n = 6, name = "GnBu")
#scale_color_brewer(palette="GnBu", n=6, direction=-1) + if want to use the palette and reverse order

# inspect
plot(nmds.daph.rar)
plot(vf.daph)

# make the plot
NMDS.daph <- ggplot(nmds.daph.rar.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake), size= 2) +
  #coord_fixed(ratio=1.6) + 
  xlim(-2.2,1.7)+
  ylim(-1.5,1.5)+
  stat_ellipse(level=0.95, linetype = 2, linewidth=0.6, aes(x = NMDS1, y = NMDS2, color=lake)) +
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  ggtitle("Daphnia")+
  geom_segment(data = spp.daph.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "goldenrod4") +
  geom_text(data = spp.daph.scrs.signif, aes(x = NMDS1, y = NMDS2, label = Envfact), color="goldenrod4",
            size = 2.5)+
  theme_classic() +
  annotate(geom="text", x=1, y=-1.5, label="Stress=0.104", size=3,
           color="black")

### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.daph <- adonis2(bc_daph ~ time_point * lake, 
                            data = nmds.daph.rar.df, permutations = 999, by="terms")
broad.model.daph # signif lake and time effect, driven by differences in time 2

pairwise.adonis(bc_daph, factors=nmds.daph.rar.df$lake, p.adjust.m="holm")

# Env model
env.model.daph <- adonis2(bc_daph ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.daph.rar.df, permutations = 999, by="terms")
env.model.daph # latitude is best (0.12), elevation (0.07), cond (0.07)

# betadispersion
anova(betadisper(bc_daph, nmds.daph.rar.df$time)) # not signf, (p=0.234)
anova(betadisper(bc_daph, nmds.daph.rar.df$lake)) # not significant, (p=0.526)

# alternative means of finding best model
daph.rar.env<- nmds.daph.rar.df %>% dplyr::select(latitude, elevation_m, temp_C, pH, cond, DOC, chla_ug.L, DO_perc)
bioenv(bc_daph, daph.rar.env, scale.var = TRUE)
# best model ph and conductivity = 0.49
```

- Now run NMDS-envfit for *calanoids*
```{r calanoid env nmds}
# subset
calan.phylo<-subset_samples(PS.500, organism=="Calanoid")
calan.phylo <- prune_taxa(taxa_sums(calan.phylo) >0, calan.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied
set.seed(1000)
bc_calan <- phyloseq::distance(calan.phylo, method = "bray") # rarefied

# run NMDS on Bray distance object (defined above for 'bc_500')
set.seed(520)
nmds.calan.rar <- metaMDS(bc_calan,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

# get NMDS 1-3
sp.calan.scores <- as.data.frame(scores(nmds.calan.rar, display = "sites"))
nmds.calan.rar.df <- merge(sample_data(calan.phylo), sp.calan.scores, by = 'row.names', all = TRUE)


#### pick envr. data and run the envfit function
env.calan<- nmds.calan.rar.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.calan <- envfit(nmds.calan.rar, env.calan, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

# inspect
plot(nmds.calan.rar)
plot(vf.calan)

### grab vectors and put into df
spp.calan.scrs <- as.data.frame(scores(vf.calan, display = "vectors")) *ordiArrowMul(vf.calan)
spp.calan.scrs.test <- cbind(spp.calan.scrs, r=vf.calan$vectors[2], pvals=vf.calan$vectors[4], 
                       Envfact = rownames(spp.calan.scrs))
names(spp.calan.scrs.test)[names(spp.calan.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.calan.scrs.signif<-spp.calan.scrs.test[spp.calan.scrs.test$pvals< 0.05, ]

# make the plot
NMDS.calan <- ggplot(nmds.calan.rar.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake), size=2) +
  #coord_fixed(ratio=0.8) +
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  stat_ellipse(level=0.95, linetype = 2, linewidth=0.6, aes(x = NMDS1, y = NMDS2, color=lake)) +
  ggtitle("Calanoids")+
  geom_segment(data = spp.calan.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "goldenrod4") +
  geom_text(data = spp.calan.scrs.signif, 
            aes(x = NMDS1, y = NMDS2, label = Envfact), color="goldenrod4",
            size = 2.5)+
  theme_classic() +
  annotate(geom="text", x=1, y=-4, label="Stress=0.117", size=3,
           color="black")

### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.calan <- adonis2(bc_calan ~ time_point * lake, 
                            data = nmds.calan.rar.df, permutations = 999, by="terms")
broad.model.calan # signif lake and time effect, driven by differences in time 1 and 5 (whcih are similar) 

pairwise.adonis(bc_calan, factors=nmds.calan.rar.df$lake, p.adjust.m="holm")

# Env model
env.model.calan <- adonis2(bc_calan ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.calan.rar.df, permutations = 999)
env.model.calan # elevation (0.22), latitude is best (0.13)

# betadispersion
anova(betadisper(bc_calan, nmds.calan.rar.df$time)) # NS
anova(betadisper(bc_calan, nmds.calan.rar.df$lake)) # significant (p=0.004))

# alternative means of finding best model
calan.rar.env<- nmds.calan.rar.df %>% dplyr::select(latitude, elevation_m, temp_C, pH, cond, DOC, chla_ug.L, DO_perc)
bioenv(bc_calan, calan.rar.env, scale.var = TRUE)
# best model latitude, elevation_m = 0.74
```

- Now run NMDS-envfit for *cyclopoids*
```{r cyclopoid env nmds}
# subset
cyclop.phylo<-subset_samples(PS.500, organism=="Cyclopoid")
cyclop.phylo <- prune_taxa(taxa_sums(cyclop.phylo) >0, cyclop.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied
set.seed(1000)
bc_cyclop <- phyloseq::distance(cyclop.phylo, method = "bray") # rarefied

# run NMDS on Bray distance object
set.seed(520)
nmds.cyclop.rar <- metaMDS(bc_cyclop,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

# get NMDS 1-3
sp.cyclop.scores <- as.data.frame(scores(nmds.cyclop.rar, display = "sites"))
nmds.cyclop.rar.df <- merge(sample_data(cyclop.phylo), sp.cyclop.scores, by = 'row.names', all = TRUE)

#### pick envr. data and run the envfit function
env.cyclop<- nmds.cyclop.rar.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.cyclop <- envfit(nmds.cyclop.rar, env.cyclop, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

### grab vectors and put into df
spp.cyclop.scrs <- as.data.frame(scores(vf.cyclop, display = "vectors")) *ordiArrowMul(vf.cyclop)
spp.cyclop.scrs.test <- cbind(spp.cyclop.scrs, r=vf.cyclop$vectors[2], pvals=vf.cyclop$vectors[4], 
                       Envfact = rownames(spp.cyclop.scrs))
names(spp.cyclop.scrs.test)[names(spp.cyclop.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.cyclop.scrs.signif<-spp.cyclop.scrs.test[spp.cyclop.scrs.test$pvals< 0.05, ]

# inspect
plot(nmds.cyclop.rar)
plot(vf.cyclop)

# make the plot
NMDS.cyclop <- ggplot(nmds.cyclop.rar.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake), size=2) +
  #coord_fixed(ratio=1.2) + 
  stat_ellipse(level=0.95, linetype = 2, linewidth=0.6, aes(x = NMDS1, y = NMDS2, color=lake)) +
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  ggtitle("Cyclopoids")+
  geom_segment(data = spp.cyclop.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "goldenrod4") +
  geom_text(data = spp.cyclop.scrs.signif, aes(x = NMDS1, y = NMDS2, label = Envfact), color="goldenrod4",
            size = 2.5)+
  theme_classic()+
  annotate(geom="text", x=2, y=-1.75, label="Stress=0.129", size=3,
           color="black")


### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.cyclop <- adonis2(bc_cyclop ~ time_point * lake, 
                            data = nmds.cyclop.rar.df, permutations = 999, by="terms")
broad.model.cyclop # signif lake and time effect, driven by time 2

pairwise.adonis(bc_cyclop, factors=nmds.cyclop.rar.df$lake, p.adjust.m="holm")

# Env model
env.model.cyclop <- adonis2(bc_cyclop ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.cyclop.rar.df, permutations = 999)
env.model.cyclop # latitude is best (0.18), elevation (0.12), cond (0.06)

# betadispersion
anova(betadisper(bc_cyclop, nmds.cyclop.rar.df$time)) # NS
anova(betadisper(bc_cyclop, nmds.cyclop.rar.df$lake)) # NS

# alternative means of finding best model
cyclo.rar.env<- nmds.cyclop.rar.df %>% dplyr::select(latitude, elevation_m, temp_C, pH, cond, DOC, chla_ug.L, DO_perc)
bioenv(bc_cyclop, cyclo.rar.env, scale.var = TRUE)
# latitude, pH, cond = 0.71
```

- Lastly, run NMDS-envfit for *Water*
```{r water env nmds}
# subset
water.phylo<-subset_samples(PS.500, sample_type=="Water")
water.phylo <- prune_taxa(taxa_sums(water.phylo) >0, water.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied
set.seed(1000)
bc_water <- phyloseq::distance(water.phylo, method = "bray") # rarefied

# run NMDS on Bray distance object (defined above for 'bc_water'), this is
set.seed(520)
nmds.water.rar <- metaMDS(bc_water,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)


# get NMDS 1-3
sp.water.scores <- as.data.frame(scores(nmds.water.rar, display = "sites"))
nmds.water.rar.df <- merge(sample_data(water.phylo), sp.water.scores, by = 'row.names', all = TRUE)

#### pick envr. data and run the envfit function
env.water<- nmds.water.rar.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.water <- envfit(nmds.water.rar, env.water, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

### grab vectors and put into df
spp.water.scrs <- as.data.frame(scores(vf.water, display = "vectors")) *ordiArrowMul(vf.water)
spp.water.scrs.test <- cbind(spp.water.scrs, r=vf.water$vectors[2], pvals=vf.water$vectors[4], 
                       Envfact = rownames(spp.water.scrs))
names(spp.water.scrs.test)[names(spp.water.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.water.scrs.signif<-spp.water.scrs.test[spp.water.scrs.test$pvals< 0.05, ]

# inspect
plot(nmds.water.rar)
plot(vf.water)

# make the plot
NMDS.water <- ggplot(nmds.water.rar.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake), size=2) +
  #coord_fixed(ratio=0.7) + 
  xlim(-1.3, 1.5)+
  ylim(-2.7,2.3)+
  stat_ellipse(level=0.95, linetype = 2, linewidth=0.6, aes(x = NMDS1, y = NMDS2, color=lake)) +
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  ggtitle("Water")+
  geom_segment(data = spp.water.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "goldenrod4") +
  geom_text(data = spp.water.scrs.signif, aes(x = NMDS1, y = NMDS2, label = Envfact), color="goldenrod4",
            size = 2.5)+
  theme_classic() + 
  annotate(geom="text", x=1, y=-2.7, label="Stress=0.082", size=3,
           color="black")

### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.water <- adonis2(bc_water ~ time_point + lake, 
                            data = nmds.water.rar.df, permutations = 999)
broad.model.water # signif lake effect, time effects minimal and largely random

pairwise.adonis(bc_water, factors=nmds.water.rar.df$lake, p.adjust.m="holm")

# Env model
env.model.water <- adonis2(bc_water ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.water.rar.df, permutations = 999, by="terms")
env.model.water # latitude is best (0.19), elevation (0.18), cond (0.07), temp (0.06)

# betadispersion
anova(betadisper(bc_water, nmds.water.rar.df$time)) # NS
anova(betadisper(bc_water, nmds.water.rar.df$lake)) # NS

# alternative means of finding best model
wat.rar.env<- nmds.water.rar.df %>% dplyr::select(latitude, elevation_m, temp_C, pH, cond, DOC, chla_ug.L, DO_perc)
bioenv(bc_water, wat.rar.env, scale.var = TRUE)
# latitude + cond = 0.81
```

- make an aggregate plot of the 4 NMDS-envs
```{r envfit NMDS fig}
extract.legend <- get_legend(
  # create some space to the left of the legend
  NMDS.daph + theme(legend.box.margin = margin(0, 0, 0, 10)))

NMDS.env.species<-plot_grid(NMDS.daph + guides(color="none"), 
                            NMDS.calan + guides(color="none"),
                            NMDS.cyclop + guides(color="none"),
                            NMDS.water + guides(color="none"),
                            extract.legend,
          ncol=5, rel_widths = c(10,10,10,10,3), labels = c("A","B", "C", "D", ""))

NMDS.env.species
dev.copy(pdf, "figures/Fig8.NMDS.env.specieswater.pdf", height=4, width=13)
dev.off()
```


#### NMDS by taxa-space-time

This code runs NMDS plots for all the data with different binning factors. May need to reduce and/or decide if PCoA or NMDS is best for showing "full community differences", which presently is in both places.
```{r General NMDS time, lake, basin, etc)}

nmds.organism <- ggplot(nmds.rar.df, aes(x=NMDS1, y=NMDS2, color = organism)) +
  geom_point(aes(color = organism, shape=sample_type), size = 1.3) + 
  stat_ellipse(level=0.95, lwd = 0.5, linetype = 2, aes(color=organism)) +
  xlim(-2.4,2.25)+
  ylim(-2.2,2.4)+
  scale_color_manual(values=c("coral2", "gold3", "darkolivegreen3", "darkslategray4", "dodgerblue")) +
  scale_shape_manual(values=c(1,16))+
  ggtitle("Organism") +
  theme_classic()


nmds.phy_group <- ggplot(nmds.rar.df, aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group, shape=sample_type), size = 1.3) + 
  stat_ellipse(level=0.95, lwd = 0.5, linetype = 2, aes(color=phy_group)) +
  theme_classic()+
  xlim(-2.4,2.25)+
  ylim(-2.2,2.4)+
  scale_color_manual(values = c("indianred3","darkolivegreen4", "dodgerblue3")) +
  scale_shape_manual(values=c(1,16))+
  ggtitle("Taxonomic Group")



nmds.rar.df$Date<-as.factor(nmds.rar.df$time_point)
nmds.rar.df$Date<-recode_factor(nmds.rar.df$Date, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")

nmds.time <- ggplot(nmds.rar.df, aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = Date, shape=sample_type), size = 1.3) + 
  stat_ellipse(level=0.95, lwd = 0.5, linetype = 2, aes(color=Date)) +
  scale_shape_manual(values=c(1,16))+
  xlim(-2.4,2.25)+
  ylim(-2.2,2.4)+
  scale_color_manual(values=c("indianred4", "indianred1", "orange", "palevioletred", "thistle")) + 
  ggtitle("Sampling Time Point") +
  theme_classic()
 
# plot by lake and basin and rename the factors
nmds.rar.df$int.basin.type<-interaction(nmds.rar.df$basin,nmds.rar.df$sample_type)
# recode factors
nmds.rar.df$int.basin.type <- recode_factor(nmds.rar.df$int.basin.type,
                                     North.Water = "N.Water",
                                     South.Water = "S.Water",
                                     North.Zooplankton = "N.Zoop",
                                     South.Zooplankton = "S.Zoop")


nmds.lake.basin <- ggplot(nmds.rar.df, aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = lake, shape=int.basin.type), size = 1.3) + 
  stat_ellipse(level=0.95, lwd = 0.5, linetype = 2, aes(color=lake)) +
  scale_shape_manual(values=c(0,2,15,17)) +
  xlim(-2.4,2.25)+
  ylim(-2.2,2.4)+
  scale_color_manual(values =c("darkslategray", "dodgerblue3", "#63A5CA", "lightsteelblue4","lightblue3", "#8BDDC4"))+
  ggtitle("Lake-Basin") +
  theme_classic()


phy_group_plots_combined <- plot_grid(
  nmds.organism, nmds.phy_group, nmds.time, nmds.lake.basin,  rel_widths = c(3,3,3,3,3),
  nrow=2, ncol=2, labels = c("A","B","C","D"))

phy_group_plots_combined
dev.print(pdf, "figures/Fig7.NMDS.groupeffects.pdf", height=8, width=10)
dev.off()
```

#### Stacked bar
Relative abundance stacked bar plot
```{r stacked bar plots}
########### plotting relative abundance, help from Megan Demmel, this is what I will use for analysis of relative abundances/ composition
PS.500 %>%  plot_composition()

# unload("MicrobiotaProcess") if there is an issue with the taxa table, this is the source (conflict)
pm <- psmelt(PS.500)
Abund <- aggregate(Abundance ~ sample_type + organism + phy_group + lake + time_point + sampleNames + Class, data=pm, FUN=sum)
sam <- aggregate(Abundance ~ sample_type + organism + phy_group + lake + time_point +sampleNames, data=pm, FUN=sum) # this should include everything Abund has except for Class

colnames(sam)[ncol(sam)] <- "tot.abund" # renaming the last column 'tot.abund' so that we can differentiate that from rel.abund

Abund.final <- dplyr::full_join(Abund, sam)
Abund.final$rel_abund <- Abund.final$Abundance/Abund.final$tot.abund

# classes that have rel.abund less than 5% will be named as other
Abund.final$Class[Abund.final$rel_abund <= 0.05] <- "Other (<5%)" 

View(Abund.final)

Abund.final$Class<-as.factor(Abund.final$Class)
Abund.final$Class<-factor(Abund.final$Class, 
                          levels=c("Acidimicrobiia", "Actinobacteria", "Alphaproteobacteria",
                              "Bacilli", "Bacteroidia", "Cyanobacteriia", "Gammaproteobacteria", "Saccharimonadia",
                               "Verrucomicrobiae", "Other (<5%)"))

Abund.final$Date<-as.factor(Abund.final$time_point)
Abund.final$Date<-recode_factor(Abund.final$Date, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")

sum.rel<-aggregate(rel_abund~phy_group:Class:lake:Date:sampleNames, data=Abund.final, FUN=sum)

# rel abund plot across time, lake, and sample type
Rel.abund.class <- ggplot(Abund.final, aes(x = Date, y = rel_abund, fill= Class, color = Class)) +
  geom_bar(width = 0.8, aes(fill = Class, color = Class), stat ="identity", position = "fill") +
  facet_grid(cols = vars(lake), rows = vars(phy_group)) + 
  xlab("Sampling Time Point") +  
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text.x=element_text(size=8, angle = 75, vjust = 0.5, hjust=1), axis.title.x=element_text(size=13)) +
  theme(axis.text.y=element_text(size=11), axis.title.y=element_text(size=13)) +
  theme(strip.text.x = element_text(size = 13), strip.text.y = element_text(size = 13)) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightsalmon", "seagreen4", "lightgreen","lightskyblue", "royalblue3", "plum", "tan3")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightsalmon", "seagreen4", "lightgreen","lightskyblue", "royalblue3", "plum", "tan3"))

Rel.abund.class
dev.copy(pdf, "figures/Fig6.Relabund.facet.pdf", height=10, width=12)
dev.off()
```

```{r phyloseq class and family RA}

### being paranoid... show me that phyloseq makes the same figure as above at class level
#total = median(sample_sums(PS.500))
#standf = function(x, t=total) round(t * (x / sum(x)))
#normalized_ps = transform_sample_counts(PS.500, standf)

merged.PS.500 <- merge_samples(PS.500, "sampleNames") # merge by samples (this is how data should be already)
sample_data(merged.PS.500)$phy_group <- factor(sample_data(PS.500)$phy_group)
sample_data(merged.PS.500)$lake <- factor(sample_data(PS.500)$lake)

normalized_ps_class = tax_glom(merged.PS.500, "Class", NArm = TRUE)
normalized_ps_class_relabun = transform_sample_counts(normalized_ps_class, function(OTU) OTU/sum(OTU))

############
ps.abund.class<-psmelt(normalized_ps_class_relabun)
ps.abund.class<-ps.abund.class %>%
   dplyr::rename("rel_abund" = "Abundance") 

# classes that have rel.abund less than 5% will be named as other
ps.abund.class$Class[ps.abund.class$rel_abund <= 0.051] <- "Other (<5%)" 

ps.abund.class$Class<-as.factor(ps.abund.class$Class)
ps.abund.class$Class<-fct_relevel(ps.abund.class$Class, 'Other (<5%)', after=Inf)

ps.abund.class$Date<-as.factor(ps.abund.class$time_point)
ps.abund.class$Date<-recode_factor(ps.abund.class$Date, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")

# by lake, time, phy_group
ggplot(ps.abund.class, aes(x = Date, y = rel_abund, fill= Class, color = Class)) +
  geom_bar(width = 0.8, aes(fill = Class, color = Class), stat ="identity", position = "fill") +
  facet_grid(cols = vars(lake), rows = vars(phy_group)) + 
  xlab("Sampling Time Point") +  
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text.x=element_text(size=8, angle = 75, vjust = 0.5, hjust=1), axis.title.x=element_text(size=13)) +
  theme(axis.text.y=element_text(size=11), axis.title.y=element_text(size=13)) +
  theme(strip.text.x = element_text(size = 13), strip.text.y = element_text(size = 13)) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightsalmon", "seagreen4", "lightgreen","lightskyblue", "royalblue3", "plum", "tan3")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightsalmon", "seagreen4", "lightgreen","lightskyblue", "royalblue3", "plum", "tan3"))                                      


# if merged across all samples at organism level...
merged.PS.500.organism <- merge_samples(PS.500, "organism") # merge by organisms, pooling across all levels
sample_data(merged.PS.500.organism)$organism<-factor(c("Daphnia", "Holopedium", "Calanoid", "Cyclopoid", "Water"))
sample_data(merged.PS.500.organism)$organism<-factor(sample_data(merged.PS.500.organism)$organism,
                                  levels=c("Daphnia", "Holopedium", "Calanoid", "Cyclopoid", "Water"))

normalized_ps_class.org = tax_glom(merged.PS.500.organism, "Class", NArm = TRUE)
norm_ps_class_relabun_org = transform_sample_counts(normalized_ps_class.org, function(OTU) OTU/sum(OTU))

ps.abund.class.org<-psmelt(norm_ps_class_relabun_org)
ps.abund.class.org<-ps.abund.class.org %>%
   dplyr::rename("rel_abund" = "Abundance") 

# classes that have rel.abund less than 5% will be named as other
ps.abund.class.org$Class[ps.abund.class.org$rel_abund <= 0.05] <- "Other (<5%)" 

ps.abund.class.org$Class<-as.factor(ps.abund.class.org$Class)
ps.abund.class.org$Class<-fct_relevel(ps.abund.class.org$Class, 'Other (<5%)', after=Inf)
ps.abund.class.org$organism<-factor(ps.abund.class.org$organism)


ggplot(ps.abund.class.org, aes(x=organism, y=rel_abund, fill=Class, color=Class)) +
  geom_bar(aes(fill=Class, color=Class), stat="identity", position="fill") +
  xlab("Sample Type") +ylab("Relative Abundance") + theme_bw() +
  theme(axis.text.x=element_text(size=10)) + theme(axis.title.x=element_text(size=13)) +
  theme(axis.text.y=element_text(size=11)) + theme(axis.title.y=element_text(size=13)) +
  scale_fill_manual(values = c("orangered", "gold", "seagreen4", "lightskyblue", "plum", "tan3")) +
  scale_color_manual(values = c("orangered", "gold", "seagreen4", "lightskyblue", "plum", "tan3"))


class.org.dom<-ps.abund.class.org[!(ps.abund.class.org$Class=="Other (<5%)"),]
class.org.dom.sum<- class.org.dom %>%
  dplyr::select(organism, rel_abund, Class)





############# by Family
normalized_ps_family = tax_glom(merged.PS.500, "Family", NArm = TRUE)
normalized_ps_family_relabun = transform_sample_counts(normalized_ps_family, function(OTU) OTU/sum(OTU))

ps.abund.family<-psmelt(normalized_ps_family_relabun)
ps.abund.family<-ps.abund.family %>%
   dplyr::rename("rel_abund" = "Abundance") 

# classes that have rel.abund less than 10% will be named as other
ps.abund.family$Family[ps.abund.family$rel_abund <= 0.10] <- "Other (<10%)" 

ps.abund.family$Family<-as.factor(ps.abund.family$Family)
ps.abund.family$Family<-fct_relevel(ps.abund.family$Family, 'Other (<10%)', after=Inf)

ps.abund.family$Date<-as.factor(ps.abund.family$time_point)
ps.abund.family$Date<-recode_factor(ps.abund.family$Date, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")

# family-class

ps.abund.family$Fam.clas.other<-paste(ps.abund.family$Class, ps.abund.family$Family, sep="-")
ps.abund.family$Fam.clas.other<-ifelse(
                  ps.abund.family$Family == "Other (<10%)", "Other (<10%)", ps.abund.family$Fam.clas.other)

ps.abund.family$Fam.clas.other<-fct_relevel(ps.abund.family$Fam.clas.other, 'Other (<10%)', after=Inf)


ggplot(ps.abund.family, aes(x = Date, y = rel_abund, fill= Fam.clas.other, color = Fam.clas.other)) +
  geom_bar(width = 0.8, aes(fill = Fam.clas.other, color = Fam.clas.other),
           stat ="identity", position = "fill") +
  facet_grid(cols = vars(lake), rows = vars(phy_group)) + 
  xlab("Sampling Time Point") +  
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text.x=element_text(size=8, angle = 75, vjust = 0.5, hjust=1), axis.title.x=element_text(size=13)) +
  theme(axis.text.y=element_text(size=11), axis.title.y=element_text(size=13)) +
  theme(strip.text.x = element_text(size = 13), strip.text.y = element_text(size = 13)) +
  scale_fill_manual(values = c("tomato", "coral",
                               "goldenrod2","gold", "yellow2",
                               "navajowhite3",
                               "olivedrab","springgreen4","seagreen3", "yellowgreen",
                               "mediumaquamarine","limegreen","palegreen2","darkseagreen3",
      "steelblue4","blue", "royalblue3", "cornflowerblue", "dodgerblue", "skyblue3","lightskyblue", "lightblue", "paleturquoise", "slategray1", "slategray3",
      "gray50", "gray30",
     "thistle"))+
  scale_color_manual(values = c("tomato", "coral",
                               "goldenrod2","gold", "yellow2",
                               "navajowhite3",
                               "olivedrab","springgreen4","seagreen3", "yellowgreen",
                               "mediumaquamarine","limegreen","palegreen2","darkseagreen3",
      "steelblue4","blue", "royalblue3", "cornflowerblue", "dodgerblue","skyblue3", "lightskyblue", "lightblue", "paleturquoise", "slategray1", "slategray3",
      "gray50", "gray30",
     "thistle"))



# if merged across all samples at organism level...
normalized_ps_family.org = tax_glom(merged.PS.500.organism, "Family", NArm = TRUE)
norma_ps_family_relabun_org = transform_sample_counts(normalized_ps_family.org, function(OTU) OTU/sum(OTU))

ps.abund.fam.org<-psmelt(norma_ps_family_relabun_org)
ps.abund.fam.org<-ps.abund.fam.org %>%
   dplyr::rename("rel_abund" = "Abundance") 

# families that have rel.abund less than 10% will be named as other
ps.abund.fam.org$Family[ps.abund.fam.org$rel_abund <= 0.05] <- "Other (<5%)" 

ps.abund.fam.org$Fam.clas.other<-paste(ps.abund.fam.org$Class, ps.abund.fam.org$Family, sep="-")
ps.abund.fam.org$Fam.clas.other<-ifelse(
                  ps.abund.fam.org$Family == "Other (<5%)", "Other (<5%)", ps.abund.fam.org$Fam.clas.other)

ps.abund.fam.org$Fam.clas.other<-fct_relevel(ps.abund.fam.org$Fam.clas.other, 'Other (<5%)', after=Inf)


ggplot(ps.abund.fam.org, aes(x = organism, y = rel_abund, fill= Fam.clas.other, color = Fam.clas.other)) +
  geom_bar(width = 0.8, aes(fill = Fam.clas.other, color = Fam.clas.other),
           stat ="identity", position = "fill") +
  xlab("Zooplankton and Water Samples") +  
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text.x=element_text(size=8, vjust = 0.5, hjust=1), axis.title.x=element_text(size=13)) +
  theme(axis.text.y=element_text(size=11), axis.title.y=element_text(size=13)) +
  theme(strip.text.x = element_text(size = 13), strip.text.y = element_text(size = 13)) +
   scale_fill_manual(values = c("orangered",
                               "goldenrod1",
                               "springgreen4",  "limegreen","yellowgreen",
      "deepskyblue4", "steelblue", "cornflowerblue", "lightskyblue","turquoise", "tan3")) +
  scale_color_manual(values = c("orangered",
                               "goldenrod2",
                               "springgreen4", "limegreen", "yellowgreen",
      "deepskyblue4",  "steelblue", "cornflowerblue", "lightskyblue", "turquoise","tan3"))

dev.copy(pdf, "figures/FigS4.Relabund.class.fam.pdf", height=8, width=10)
dev.off()


fam.org.dom<-ps.abund.fam.org[!(ps.abund.fam.org$Fam.clas.other=="Other (<5%)"),]
fam.org.dom.sum<- fam.org.dom %>%
  dplyr::select(organism, rel_abund, Fam.clas.other)


```

```{r testing effects of env. variables on rel. abund. and richness of dominant bacteria}

# using Abund.final from the last code chunk
# testing the effects of time, lake, and sample type on the 5 most dominant classes of bacteria

############ GAMMAPROTEOBACTERIA
gammaproteobacteria <- subset(Abund.final, Class == "Gammaproteobacteria")
gammaproteobacteria <- summaryBy(rel_abund ~ sampleNames, data = gammaproteobacteria, FUN = sum)
gammaproteobacteria <- data.frame(gammaproteobacteria, row.names = gammaproteobacteria[,1])
names(gammaproteobacteria)[names(gammaproteobacteria) == 'rel_abund.sum'] <- 'Gamma.rel_abund.sum'
gammaproteobacteria <- merge(gammaproteobacteria, sample_data(PS.500), by = "row.names")
names(gammaproteobacteria)[names(gammaproteobacteria) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Gam.model<-lm(Gamma.rel_abund.sum ~ time_point + lake + phy_group, data = gammaproteobacteria)
Anova(Gam.model, type=2)
plot(allEffects(Gam.model))
# time, lake, phy.group all matter

# separate sample types
gamma.wat<-gammaproteobacteria[(gammaproteobacteria$sample_type=="Water"),]
gamma.zoop<-gammaproteobacteria[(gammaproteobacteria$sample_type=="Zooplankton"),]

#### water gamma
# testing environmental effects on gamaproteobacteria abundances
FitAll.gamma.wat <- lm(Gamma.rel_abund.sum ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond, data = gamma.wat)
stepAIC(FitAll.gamma.wat, direction = "both")

# best model
Best.gamma.wat<-lm(Gamma.rel_abund.sum ~ latitude + temp_C + chla_ug.L + pH + cond, data = gamma.wat)
summary(Best.gamma.wat)
Anova(Best.gamma.wat, type=2) # conductivity is most important, followed by elevation_m, temp

summary(lm(Gamma.rel_abund.sum ~ cond , data = gamma.wat)) #p<0.001, R2=0.436 <<< best
summary(lm(Gamma.rel_abund.sum ~ latitude , data = gamma.wat)) #p=0.007, R2=0.22
summary(lm(Gamma.rel_abund.sum ~ temp_C , data = gamma.wat)) #p=0.056, R2=0.100
anova(lm(Gamma.rel_abund.sum ~ chla_ug.L , data = gamma.wat)) #NS
anova(lm(Gamma.rel_abund.sum ~ pH , data = gamma.wat)) #NS

#### zoop gamma
# testing environmental effects on gamaproteobacteria abundances
FitAll.gamma.zoop <- lm(Gamma.rel_abund.sum ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond, data = gamma.zoop)
stepAIC(FitAll.gamma.zoop, direction = "both")

# best model
Best.gamma.zoop<-lm(Gamma.rel_abund.sum ~ latitude + elevation_m + temp_C + chla_ug.L + pH + DO_perc + cond, data = gamma.zoop)
summary(Best.gamma.zoop)
Anova(Best.gamma.zoop, type=2) # conductivity is most important, p=0.025, R2=0.016

#latitude
summary(lm(Gamma.rel_abund.sum ~ latitude , data = gamma.zoop)) # R2=0.073, p<0.001

#conductivity 
summary(lm(Gamma.rel_abund.sum ~ cond , data = gamma.zoop)) #signif, R2=0.016, p=0.025

#elevation NS
summary(lm(Gamma.rel_abund.sum ~ elevation_m , data = gamma.zoop)) #NS

#chla NS
summary(lm(Gamma.rel_abund.sum ~ chla_ug.L , data = gamma.zoop)) #NS

#DO NS
summary(lm(Gamma.rel_abund.sum ~ DO_perc , data = gamma.zoop)) #NS

### plots for gamma with by sample types
gamma_cond <- ggplot(gammaproteobacteria, aes(x=cond, y=Gamma.rel_abund.sum, 
                                              color=sample_type, fill=sample_type, linetrype=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  scale_linetype_manual(values = c("dashed", "solid"))+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  geom_point(size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("Conductivity(uS/cm)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16))+
  ggtitle("Gammaproteobacteria") + theme_classic()+
  annotate(geom="text", x=60, y=1.0, label="Zoop. Adj-R2=0.025, p=0.025", size=2, color="black") +
  annotate(geom="text", x=60, y=0.1, label="Water Adj-R2=0.436, p<0.001", size=2, color="black")
gamma_cond

# latitude
gamma_lat <- ggplot(gammaproteobacteria, aes(x=latitude, y=Gamma.rel_abund.sum, 
                                        color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylim(c(0,1))+
  xlab("Latitude (o N)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16))+
  ggtitle("Gammaproteobacteria") + theme_classic()+
  annotate(geom="text", x=37.8, y=0.0, label="Water Adj-R2=0.220, p=0.007", size=2, color="black")+ 
  annotate(geom="text", x=37.8, y=0.5, label="Zoop Adj-R2=0.073, p<0.001", size=2, color="black")
gamma_lat


########### ALPHAPROTEOBACTERIA
alphaproteobacteria <- subset(Abund.final, Class == "Alphaproteobacteria")
alphaproteobacteria <- summaryBy(rel_abund ~ sampleNames, data = alphaproteobacteria, FUN = sum)
alphaproteobacteria <- data.frame(alphaproteobacteria, row.names = alphaproteobacteria[,1])
names(alphaproteobacteria)[names(alphaproteobacteria) == 'rel_abund.sum'] <- 'Alphapr.rel_abund.sum'
alphaproteobacteria <- merge(alphaproteobacteria, sample_data(PS.500), by = "row.names")
names(alphaproteobacteria)[names(alphaproteobacteria) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Alpha.model<-lm(Alphapr.rel_abund.sum ~ time_point + lake + phy_group, data = alphaproteobacteria)
Anova(Alpha.model, type=2)
plot(allEffects(Alpha.model))
# time, lake, phy.group all matter

# separate sample types
alph.wat<-alphaproteobacteria[(alphaproteobacteria$sample_type=="Water"),]
alph.zoop<-alphaproteobacteria[(alphaproteobacteria$sample_type=="Zooplankton"),]

# testing environmental effects on alphaproteobacteria abundances
FitAll.alpha.wat<- lm(Alphapr.rel_abund.sum ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond, data = alph.wat)
stepAIC(FitAll.alpha.wat, direction = "both") # best is temp_C + DO_perc + cond 

# best model
Best.alpha.wat<-lm(Alphapr.rel_abund.sum ~  latitude + elevation_m + temp_C + DO_perc, data = alph.wat)
summary(Best.alpha.wat)
Anova(Best.alpha.wat, type=2) # 
plot(allEffects(Best.alpha.wat))

#elevation
summary(lm(Alphapr.rel_abund.sum ~ elevation_m , data = alph.wat)) # R2=0.284, #p=0.005

#cond
summary(lm(Alphapr.rel_abund.sum ~ cond , data = alph.wat)) #R2=0.450, #p=0.009,

#Temp
summary(lm(Alphapr.rel_abund.sum ~ temp_C , data = alph.wat)) # R2=0.145, #p=0.041

#latitude
summary(lm(Alphapr.rel_abund.sum ~ latitude , data = alph.wat)) #NS

#DO 
summary(lm(Alphapr.rel_abund.sum ~ DO_perc , data = alph.wat)) #NS


##### now zoops
# testing environmental effects on alphaproteobacteria abundances
FitAll.alpha.zoop<- lm(Alphapr.rel_abund.sum ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond, data = alph.zoop)
stepAIC(FitAll.alpha.zoop, direction = "both") # DO_perc + cond best

# best model
Best.alpha.zoop<-lm(Alphapr.rel_abund.sum ~  DO_perc + cond, data = alph.zoop)
summary(Best.alpha.zoop)
Anova(Best.alpha.zoop, type=2) # DO is most important, cond less so.
plot(allEffects(Best.alpha.zoop))

summary(lm(Alphapr.rel_abund.sum ~ DO_perc , data = alph.zoop)) #p=0.001, R2=0.111
summary(lm(Alphapr.rel_abund.sum ~ cond , data = alph.zoop)) #p<0.001, R2=0.096
summary(lm(Alphapr.rel_abund.sum ~ elevation_m , data = alph.zoop)) #p=0.007, R2=0.053

# plot it
# DO important for zoops
alpha_DO<-ggplot(alphaproteobacteria, aes(x=DO_perc, y=Alphapr.rel_abund.sum, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  geom_point(size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("Dissolved Oxygen (%)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) +
  ggtitle("Alphaproteobacteria") + theme_classic() +
  annotate(geom="text", x=75, y=0.75, label="Zoop Adj-R2=0.111, p=0.001", size=2,
           color="black") 
alpha_DO

# condutivity
alpha_cond <-ggplot(alphaproteobacteria, aes(x=cond, y=Alphapr.rel_abund.sum, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  geom_point(size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("Conductivity(uS/cm)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) +
  ggtitle("Alphaproteobacteria") + theme_classic() +
  annotate(geom="text", x=75, y=0.75, label="Zoop Adj-R2=0.096, p<0.001", size=2,
           color="black") +
  annotate(geom="text", x=75, y=0, label="Water Adj-R2=0.250, p=0.009", size=2,
           color="black")
alpha_cond

# elevation
alpha_elev <-ggplot(alphaproteobacteria, aes(x=elevation_m, y=Alphapr.rel_abund.sum, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  geom_point(size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("Elevation (m)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) +
  ggtitle("Alphaproteobacteria") + theme_classic() +
  annotate(geom="text", x=2700, y=0.75, label="Zoop Adj-R2=0.053, p=0.007", size=2,
           color="black") +
  annotate(geom="text", x=2700, y=0.5, label="Water Adj-R2=0.284, p=0.005", size=2, 
           color="black")
alpha_elev


##### #ACTINOBACTERIA
actinobacteria <- subset(Abund.final, Class == "Actinobacteria")
actinobacteria <- summaryBy(rel_abund ~ sampleNames, data = actinobacteria, FUN = sum)
actinobacteria <- data.frame(actinobacteria, row.names = actinobacteria[,1])
names(actinobacteria)[names(actinobacteria) == 'rel_abund.sum'] <- 'Actinob.rel_abund.sum'
actinobacteria <- merge(actinobacteria, sample_data(PS.500), by = "row.names")
names(actinobacteria)[names(actinobacteria) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Actin.model<-lm(Actinob.rel_abund.sum ~ time_point + lake + phy_group, data = actinobacteria)
Anova(Actin.model)
plot(allEffects(Actin.model))
# lake, phy.group all matter

# separate sample types
act.wat<-actinobacteria[(actinobacteria$sample_type=="Water"),]
act.zoop<-actinobacteria[(actinobacteria$sample_type=="Zooplankton"),]

# testing environmental effects on alphaproteobacteria abundances
### WATER first
FitAll.actin.wat<- lm(Actinob.rel_abund.sum ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond,  data = act.wat)
stepAIC(FitAll.actin.wat, direction = "both") # cond best model

# best model
Best.actin<-lm(Actinob.rel_abund.sum ~ cond, data = act.wat)
Anova(Best.actin)
summary(Best.actin) # cond is most important (p<0.001, R2=0.429)  
plot(allEffects(Best.actin))

#elevation
summary(lm(Actinob.rel_abund.sum ~ elevation_m, data = act.wat)) #R2=0.216, p=0.010

#cond
summary(lm(Actinob.rel_abund.sum ~ cond, data = act.wat)) #(p<0.001, R2=0.429)  

### now Zoops 
FitAll.actin.zoop<- lm(Actinob.rel_abund.sum ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond, data = act.zoop)
step(FitAll.actin.zoop, direction = "both") # no effects and no relationship
summary(lm(Actinob.rel_abund.sum ~ cond, data = act.zoop)) # for the plot

anova(lm(Actinob.rel_abund.sum ~ 1, data = act.zoop)) #NA intercept fit

# for plotting
summary(lm(Actinob.rel_abund.sum ~ cond, data = act.zoop)) #R2=-0.140, p=0.629
summary(lm(Actinob.rel_abund.sum ~ elevation_m, data = act.zoop)) #R2=-0.139, p=0.627

# plot it
actino_cond <- ggplot(actinobacteria, aes(x=cond, y=Actinob.rel_abund.sum, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  geom_point(size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("Conductivity a (uS/cm)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) +
  ggtitle("Actinobacteria") + theme_classic() +
   annotate(geom="text", x=70, y=0.50, label="Water Adj-R2=0.429, p<0.001", size=2,
           color="black") +
  annotate(geom="text", x=70, y=0.50, label="Zoop. Adj-R2=0.429, p<0.001", size=2,
           color="black")
actino_cond 

#elevation
actino_elev <- ggplot(actinobacteria, aes(x=elevation_m, y=Actinob.rel_abund.sum, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  geom_point(size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("Elevation (m)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) +
  ggtitle("Actinobacteria") + theme_classic() +
   annotate(geom="text", x=2700, y=0.50, label="Water Adj-R2=0.216, p=0.010", size=2, 
           color="black")
actino_elev 


##### Bacteroidia
bacteroidia <- subset(Abund.final, Class == "Bacteroidia")
bacteroidia <- summaryBy(rel_abund ~ sampleNames, data = bacteroidia, FUN = sum)
bacteroidia <- data.frame(bacteroidia, row.names = bacteroidia[,1])
names(bacteroidia)[names(bacteroidia) == 'rel_abund.sum'] <- 'Bactero.rel_abund.sum'
bacteroidia <- merge(bacteroidia, sample_data(PS.500), by = "row.names")
names(bacteroidia)[names(bacteroidia) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Bactero.model<-lm(Bactero.rel_abund.sum ~ time_point + lake + phy_group, data = bacteroidia)
Anova(Bactero.model, type=2)
plot(allEffects(Bactero.model)) # time marginal signif, lake and phy strong

# separate sample types
bact.wat<-bacteroidia[(bacteroidia$sample_type=="Water"),]
bact.zoop<-bacteroidia[(bacteroidia$sample_type=="Zooplankton"),]

### Water first
# testing environmental effects on alphaproteobacteria abundances
FitAll.Bactero.wat<- lm(Bactero.rel_abund.sum ~ latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond, data = bact.wat)
stepAIC(FitAll.Bactero.wat, direction = "both") # DO_perc + elevation_m  best

# best model
Best.bactero.wat<-lm(Bactero.rel_abund.sum ~ latitude + elevation_m + DO_perc + cond, data = bact.wat)
summary(Best.bactero.wat)
Anova(Best.bactero.wat, type=2) # DO signif, increases
plot(allEffects(Best.bactero.wat))

# water DO
summary(lm(Bactero.rel_abund.sum ~ DO_perc, data = bact.wat)) #R2=0.561, p<0.001

# water latitude
summary(lm(Bactero.rel_abund.sum ~ latitude, data = bact.wat)) #R2=0.449, p<0.001

# water conductivity
summary(lm(Bactero.rel_abund.sum ~ cond, data = bact.wat)) #R2=0.112, p=0.046

# water elevation
anova(lm(Bactero.rel_abund.sum ~ elevation_m, data = bact.wat)) #NS

### now zoops
FitAll.Bactero.zoop<- lm(Bactero.rel_abund.sum ~  latitude + elevation_m + temp_C + log(DOC) + chla_ug.L + 
                         pH + DO_perc + cond,  data = bact.zoop)
stepAIC(FitAll.Bactero.zoop, direction = "both") # latitude + elevation_m + temp_C + DO_perc + cond

# best model
Best.bactero.zoop<-lm(Bactero.rel_abund.sum ~ latitude + elevation_m + temp_C + DO_perc + cond, data = bact.zoop)
summary(Best.bactero.zoop)
Anova(Best.bactero.zoop, type=2) # DOC signif
plot(allEffects(Best.bactero.zoop))

# zoop temp
summary(lm(Bactero.rel_abund.sum ~ temp_C, data = bact.zoop)) #0.030, R2=0.019

# zoop DO
summary(lm(Bactero.rel_abund.sum ~ DO_perc, data = bact.zoop)) #NS

# zoop latitude
summary(lm(Bactero.rel_abund.sum ~ latitude, data = bact.zoop)) #NS

# zoop elevation
Anova(lm(Bactero.rel_abund.sum ~ elevation_m, data = bact.zoop)) #NS

# zoop DO
summary(lm(Bactero.rel_abund.sum ~ cond, data = bact.zoop)) #NS

# zoop DOC
summary(lm(Bactero.rel_abund.sum ~ log(DOC), data = bact.zoop)) #p=0.003, R2=0.04

# plot it
bactero_DO <- ggplot(bacteroidia, aes(x=DO_perc, y=Bactero.rel_abund.sum, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.15)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylim(c(0,1))+
  xlab("DO (%)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) + 
  ggtitle("Bacteroidia") + theme_classic() +
  annotate(geom="text", x=70, y=1.0, label="Water Adj-R2=0.561, p<0.001", size=2,
           color="black")
bactero_DO 

# DOC importnat (sorta) for zoops
bactero_DOC <- ggplot(bacteroidia, aes(x=log(DOC), y=Bactero.rel_abund.sum, color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.15)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylim(c(0,1))+
  xlab("log(Dissolved organic carbon) (mg/L)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) + 
  ggtitle("Bacteroidia") + theme_classic() +
  annotate(geom="text", x=2, y=1.0, label="Zoop Adj-R2=0.040, p=0.003", size=2,
           color="black") +
  annotate(geom="text", x=2, y=0.7, label="Water Adj-R2=0.112, p=0.046", size=2,
           color="black")
bactero_DOC

# temp, signif for zoop (but poor R2)
bactero_temp<-ggplot(bacteroidia, aes(x=temp_C, y=Bactero.rel_abund.sum, 
                                      color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylim(c(0,1))+
  xlab("Temperature (C)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) + 
  ggtitle("Bacteroidia") + theme_classic() +
  annotate(geom="text", x=20, y=1.0, label="Zoop Adj-R2=0.019, p=0.030", size=2,
           color="black")

# DO, signif for zoop (but poor R2)
bactero_lat<-ggplot(bacteroidia, aes(x=latitude, y=Bactero.rel_abund.sum, 
                                      color=sample_type, fill=sample_type)) +
  geom_smooth(method = lm, alpha=0.2)+
  geom_point(size=1.5, alpha=0.6)+
  scale_color_manual(values = c("dodgerblue", "coral"))+
  scale_fill_manual(values = c("dodgerblue3", "firebrick"))+
  ylim(c(0,1))+
  xlab("Latitude (o N)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) + 
  ggtitle("Bacteroidia") + theme_classic() +
  annotate(geom="text", x=37.6, y=1.0, label="Water Adj-R2=0.449, p<0.001", size=2,
           color="black")
bactero_lat

#### pool the plots (add in shannon from above)
multregres.plots.shannon.env.wat.zoop<-plot_grid(plot_grid(lat.shan.plot.zoop + guides(color="none", fill="none"),
                                             DO.shan.plot.zoop + guides(color="none", fill="none"),
                                             cond.shan.plot.zoop + guides(color="none", fill="none"),
                                             extract.legend.Shan,
                            ncol=3, nrow=1, labels=c("A", "B", "C", ""), rel_widths=c(5,5,5,0.5)))

Taxa.relabund.env.regr<-plot_grid(gamma_lat + guides(color="none", fill="none"), 
                                  alpha_elev + guides(color="none", fill="none"), 
                                  actino_elev + guides(color="none", fill="none"), 
                                  bactero_lat + guides(color="none", fill="none"), 
                                  
                                  gamma_cond + guides(color="none", fill="none"), 
                                  alpha_cond + guides(color="none", fill="none"), 
                                  actino_cond + guides(color="none", fill="none"), 
                                  bactero_DO + guides(color="none", fill="none"), ncol=4,
                                  labels=c("D", "E", "F", "G", "H", "I", "J", "K"))

######### combine shannon with the taxa-specific
plot_grid(multregres.plots.shannon.env.wat.zoop, Taxa.relabund.env.regr, ncol=1, nrow=2, rel_heights = c(0.8,2))
dev.copy(pdf, "figures/Fig5.Shann.env.Taxa.relabund.regr_rev.pdf", width=12, height=10)
dev.off()
```
