---
title: "Zooplankton Microbiomes Across Time and Space"
author: "MG Perreault, CB Wall"
date: "1/15/2024"
output:
  html_document:
    code_folding: hide
    toc: yes
    toc_depth: 4
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '4'
editor_options: 
  chunk_output_type: console
---
```{r global options, results="hide", warning=FALSE, message=FALSE}
if (!require('knitr')) install.packages('knitr'); library('knitr')
knitr::opts_chunk$set(warning=FALSE, message=FALSE, fig.align='center')

# load packages
if (!require("pacman")) install.packages("pacman") # for rapid install if not in library

devtools::install_github("benjjneb/dada2", ref="v1.20") # update to most recent dada2
devtools::install_github("zdk123/SpiecEasi")
remotes::install_github("microbiome/microbiome")


# use pacman to load CRAN packages missing
pacman::p_load('knitr', 'microbiome', 'phyloseq', 'tidyr', 'tidyverse', 'knitr', 'magrittr', 'effects', 'devtools',
               'stringi', 'dplyr', "ggplot2", "gridExtra", "dada2", "phyloseq", "vegan", "cowplot", 'doBy', 'ecodist',
               'glue', 'geosphere', 'data.table', 'patchwork', 'car', 'ggcorrplot', 'FactoMineR', 'devtools', 'reshape',
               'lattice',  'plyr', 'magrittr', 'factoextra', 'multcompView', 'decontam', 'factoextra', 'car', 'mia', 
               'ade4', 'fossil', 'picante', 'reshape', 'readr', 'corrr', 'scipplot', 'Hmisc', "MASS", "FSA", "sciplot",
               "decontam","BiocManager", 'ggpubr', 'ggmap', "ggordiplots", "fossil", "SpiecEasi", "igraph", "huge",
               "leaflet")
```

## Overview

### Import Phyloseq
Import the data and modify the metadata to meet downstream needs
```{r loading in data and modifying metadata}
##########################################################

# ps.prune is the final phyloseq object from pipeline, can load it in here...
### or load in the raw data and re-assemble

ps.prune.LOCAL<-readRDS("output/reanalysis/ps.prune_local.RDS") #7141 taxa of 328 samples

# idiot check: make sure to remove all mitochondria and chloroplast taxonomic IDs
ps.prune.LOCAL <- subset_taxa(ps.prune.LOCAL, Family!= "Mitochondria" | 
                        is.na(Family) & Class!="Chloroplast" | is.na(Class))

# chloroplasts sneaking in under Order in some cyanobacteria, looks good @ 7141 ASVs
ps.prune.LOCAL <- subset_taxa(ps.prune.LOCAL, Order!="Chloroplast")

# make sure any ASVs in the data that not in > 1 sample, removed
ps.prune.LOCAL <- prune_taxa(taxa_sums(ps.prune.LOCAL) > 1, ps.prune.LOCAL) 
# few sneaked in, @ 7133 ASVs in 328 samples

# order the lakes by North to South,  in phyloseq 
sample_data(ps.prune.LOCAL)$Lake <- factor(sample_data(ps.prune.LOCAL)$Lake, levels = 
                                           c("Cooney", "Blue", "Virginia", "Convict", "Serene", "Eastern Brook"))


# export metadata and play
metaD<-microbiome::meta(ps.prune.LOCAL)

# adjust caps
metaD<-metaD %>% 
  dplyr::rename(
    sequencing_ID = Sequencing_ID,
    sample_type = Sample_Type,
    organism = Organism,
    time_point= Time.Point,
    lake = Lake,
    sample_ID = Original_ID
  )

# correct NAs to be "water"
metaD$organism<-as.character(metaD$organism) # need to do this to change the level
metaD$organism[is.na(metaD$organism)] = "Water"
metaD$organism<-as.factor(metaD$organism)
metaD$organism<-factor(metaD$organism, levels =c("Bosmina", "Ceriodaphnia", "Daphnia", "Holopedium", 
                                             "Cyclopoid", "Calanoid", "Large Calanoid",
                                             "Water")) # back to factor

metaD$organism<-recode_factor(metaD$organism, "Large Calanoid" = "Calanoid")

# order
metaD$organism<-factor(metaD$organism, levels =c("Daphnia", "Holopedium", 
                                   "Calanoid", "Cyclopoid", "Water", "Bosmina", "Ceriodaphnia"))

# format metadata (have to redo this if loading back in)
make.fac<-c("sample_ID", "sequencing_ID", "sample_type", "time_point", "lake", "Plate", "Plate_name", "Well", "sample_control")

metaD[make.fac]<-lapply(metaD[make.fac], factor) # make all these factors

# create phy group (Cladocera vs Copepoda vs water) column
metaD$phy_group <- as.factor(ifelse(metaD$organism == "Calanoid" | 
                            metaD$organism == "Cyclopoid", "copepoda", 
                   ifelse(metaD$organism == "Daphnia" | 
                            metaD$organism == "Ceriodaphnia" | 
                            metaD$organism =="Bosmina" | 
                            metaD$organism == "Holopedium", "cladocera", 
                            "water")))

# create basin column
metaD$basin <- as.factor(ifelse(metaD$lake == "Eastern Brook" | 
                                metaD$lake == "Serene" | 
                                metaD$lake == "Convict", "South", 
                                "North"))

# create latitude column
metaD$latitude <- as.numeric(ifelse(metaD$lake == "Blue", "38.050952", 
                            (ifelse(metaD$lake == "Convict", "37.590094",
                            (ifelse(metaD$lake == "Cooney", "38.04806",
                            (ifelse(metaD$lake == "Eastern Brook", "37.431526", 
                            (ifelse(metaD$lake == "Serene", "37.438392" ,
                            (ifelse(metaD$lake == "Virginia", "38.046975",
                                  "0"))))))))))))

# create longitude column
metaD$longitude <- as.numeric(ifelse(metaD$lake == "Blue", "-119.270331", 
                            (ifelse(metaD$lake == "Convict", "-118.857422",
                            (ifelse(metaD$lake == "Cooney", "-119.27702",
                            (ifelse(metaD$lake == "Eastern Brook", "-118.742615", 
                            (ifelse(metaD$lake == "Serene", "-118.744386" ,
                            (ifelse(metaD$lake == "Virginia", "-119.265076",
                                  "0"))))))))))))
                            

##### Read counts and rarefaction curves
# Make a data frame with a column for the read counts of each sample
sample_sum_df<-as.data.frame(sample_sums(ps.prune.LOCAL))
colnames(sample_sum_df)<-"read.sum"
sample_sum_df$sampleNames<-rownames(sample_sum_df)

# are row names the same? if so, add them in
identical(rownames(sample_sum_df), rownames(metaD))
metaD$read.sum<-sample_sum_df$read.sum

# add in dates, as per metadata
metaD$date<-ifelse(metaD$time_point =="1" | metaD$lake =="Blue", "6/29/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Blue", "7/09/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Blue", "7/21/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Blue", "8/04/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Blue", "8/22/2022",
                    
                     ifelse(metaD$time_point =="1" | metaD$lake =="Convict", "6/25/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Convict", "7/07/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Convict", "7/22/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Convict", "8/05/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Convict", "8/23/2022",
                          
                     ifelse(metaD$time_point =="1" | metaD$lake =="Cooney", "6/29/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Cooney", "7/09/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Cooney", "7/21/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Cooney", "8/04/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Cooney", "8/22/2022", 
                                
                     ifelse(metaD$time_point =="1" | metaD$lake =="Eastern Brook", "6/26/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Eastern Brook", "7/06/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Eastern Brook", "7/22/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Eastern Brook", "8/09/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Eastern Brook", "8/23/2022",
                                   
                    ifelse(metaD$time_point =="1" | metaD$lake =="Serene", "6/26/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Serene", "7/06/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Serene", "7/22/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Serene", "8/09/2022",
                            ifelse(metaD$time_point =="5" | metaD$lake =="Serene", "8/23/2022",
                                  
                    ifelse(metaD$time_point =="1" | metaD$lake =="Virginia", "6/29/2022",
                            ifelse(metaD$time_point =="2" | metaD$lake =="Virginia", "7/09/2022",
                            ifelse(metaD$time_point =="3" | metaD$lake =="Virginia", "7/21/2022",
                            ifelse(metaD$time_point =="4" | metaD$lake =="Virginia", "8/04/2022",
                            "8/22/2022"
                            )))))))))))))))))))))))))))))

# adjust formatting to be Date (from)
metaD$date<-as.Date(mdy(metaD$date))

metaD<- metaD %>%
  dplyr::select(sequencing_ID, sampleNames, sample_ID, read.sum, time_point, date, lake, latitude, longitude, basin, sample_type, organism, Number.of.Organism, phy_group)

   
# the metadata above is now up to date with info necessary for downstream analysis
# now merge in the environmental data 

################################## ##################################
# read in the environmental data
env.metad<-read.csv("data/full.library.env.metaD.csv")

# make a column for 'Samplenames' and use this as rownames
env.metad$sampleNames<-env.metad$sequencing_ID
env.metad$sampleNames <- gsub("^.{0,3}", "", env.metad$sampleNames)
rownames(env.metad)<- env.metad$sampleNames

# remove the rows not found in the meta-meta data -- these are samples lost in dada2/controls
env.metad<-env.metad[(env.metad$sampleNames %in% metaD$sampleNames),]

env.metad<- env.metad %>%
  dplyr::select(sample_ID, sequencing_ID, sampleNames, sample_type, organism, Number.of.Organism, 
                time_point, lake, elevation_m, temp_C, chla_ug.L, pH, DO_perc, spc, cond, DOC, TDN, TDP)

################################## ##################################
# no merge, subset to make it easier to merge...
env.metad.reduce<-env.metad %>%
  dplyr::select(sampleNames, elevation_m, temp_C, chla_ug.L, pH, DO_perc, spc, cond, DOC, TDN, TDP)


# merge the two = final metadata, add in rownames for phyloseq merge
MetaD.SQ.Env<-merge(metaD, env.metad.reduce, by="sampleNames", all.x=TRUE)
rownames(MetaD.SQ.Env)<-MetaD.SQ.Env$sampleNames

# make new column for above or below treeline
MetaD.SQ.Env$elev.cat <- as.factor(ifelse(MetaD.SQ.Env$elevation_m < 2900, "Below", "Above"))

# are row names the same? if so, let's boogie
identical(rownames(sample_data(ps.prune.LOCAL)), rownames(MetaD.SQ.Env))
write.csv(MetaD.SQ.Env, "output/MetaD.SQ.Env.csv")

#updated metadata back into phyloseq
sample_data(ps.prune.LOCAL)<-MetaD.SQ.Env
PS.fin<- ps.prune.LOCAL

table(sample_data(PS.fin)$sample_type) #328 samples: 30 water, 298 zooplankton

# change to this order for later plot tests-- ultimately Bosmina and Ceriodaphnia are dropped nad large calanoids grouped into calanoids...
sample_data(PS.fin)$organism<-factor(sample_data(PS.fin)$organism, levels =c("Daphnia", "Holopedium", 
                                   "Calanoid", "Cyclopoid", "Water", "Bosmina", "Ceriodaphnia")) # back to factor

table(tax_table(PS.fin)[, "Kingdom"], exclude = NULL) # 304 Archaea, 8197 Bacteria
summarize_phyloseq(PS.fin)

########### ########### ########### ########### 
saveRDS(PS.fin, "output/reanalysis/PS.fin.RDS") # save phyloseq
########### ########### ########### ########### 

#### export each element
OTU = as(otu_table(PS.fin), "matrix")
if(taxa_are_rows(PS.fin)){OTU <- t(OTU)}
# Coerce to data.frame
OTUdf = as.data.frame(OTU)

Tax.tab = as(tax_table(PS.fin), "matrix")
if(taxa_are_rows(PS.fin)){Tax.tab <- t(Tax.tab)}
# Coerce to data.frame
TAXdf = as.data.frame(Tax.tab)

Sam.tab = as(sample_data(PS.fin), "matrix")
if(taxa_are_rows(PS.fin)){Sam.tab <- t(Sam.tab)}
# Coerce to data.frame
SAMdf = as.data.frame(Sam.tab)

write.csv(OTUdf, "output/OTUdf.nonrar.csv")
write.csv(TAXdf, "output/TAXdf.nonrar.csv")
write.csv(SAMdf, "output/SAMdf.nonrar.csv")

## with bosmina and ceriodaphnia still in dataset, and pre-rarefaction...
summarize_phyloseq(PS.fin)
```

Inspect the run: what is the # of reads/sample? A bit over inflated due to water being so deeply sequenced, but log-read sum gives us a good picture
```{r total reads by species boxplots raw data}
#### inspect plots

# reads per sample
pdf(file="figures/reads.sample.pdf", height=4, width=7)
ggplot(MetaD.SQ.Env, aes(x=sample_type, y=read.sum, color=organism)) + geom_boxplot() + theme_bw() +
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", 
                              "dodgerblue", "violet", "wheat3"))
dev.off() 

### log reads per sample
pdf(file="figures/log.reads.sample.pdf", height=4, width=12)
ggplot(MetaD.SQ.Env, aes(x=organism, y=log(read.sum), color=organism)) + geom_boxplot() + theme_bw() +
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", 
                              "dodgerblue", "violet", "wheat3"))
dev.off() 

# read depth a bit low for zooplankton vs water, average depth is less than 2000 reads
ggplot(MetaD.SQ.Env[(MetaD.SQ.Env$sample_type=="Zooplankton"),], aes(x=sample_type, y=read.sum, color=organism)) +
  geom_boxplot() +  theme_bw() +
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", 
                              "dodgerblue", "violet", "wheat3"))
```

Now look at read depth and rarefaction curves to determine how to standardize sampling depth
```{r Read depth plots and rarefaction curves}

# Histogram of sample read counts
hist.depth <- ggplot(MetaD.SQ.Env, aes(read.sum)) + 
  geom_histogram(color="gold4", fill= "gold2", binwidth = 10000) + 
  xlab("Read counts") +
  ggtitle("Sequencing Depth") + theme_classic()

hist.depth
dev.copy(pdf, "figures/hist.depth.pdf", height=7, width=9)
dev.off() 

# a different view
read_depth_violin <- ggplot(MetaD.SQ.Env, aes(x=1, y=read.sum)) +
  geom_violin() + ggtitle("Distribution of sample sequencing depth") + scale_y_log10() +
  xlab("Samples") + ylab("Read Depth") + geom_boxplot(width=0.1)

read_depth_violin
dev.copy(pdf, "figures/read_depth_violin.pdf", height=4, width=5)
dev.off()

# Rarefaction curve
count_table_filt <- as.data.frame(otu_table(PS.fin))
rarecurve((count_table_filt), step=50, cex=0.5, xlim=c(0,100000), ylab = "ASVs", label=FALSE)
#abline(v = 500, lty = "dotted", col="red", lwd=2)

dev.copy(pdf, "figures/rare.raw.pdf", height=4, width=8)
dev.off() 
```

Now we will subset the phyloseq objects and compare the influence of rarefaction for the phyloseq data.
- `PS.fin` is the final phyloseq object with all the data.  
- `PS.500` is rarefied to 500 read-depth (with low abundance samples removed)
- `PS.zoop` is rarefied zooplankton data (500 read depth)
- `PS.water` is rarefied water only (500 read depth)
```{r exploring what low read count samples do and what to rarefy to}
###### subset the PS objects for needs...
# final and NOT rarefied data: 
PS.fin

# rarified to 500 reads: PS.rar, then reclassify and reduce some taxa and name "PS.500'
# 44 samples removedbecause they contained fewer reads than `sample.size`
# 5460 OTUs were removed because they are no longer present
PS.rar = rarefy_even_depth(PS.fin, rngseed=111, sample.size=500, replace=F) 
PS.rar <- prune_taxa(taxa_sums(PS.rar) >0, PS.rar) # remove those asvs not in at least 1 sample again

table(sample_data(PS.rar)$sample_type) #284 samples: 28 water, 256 zooplankton, 1031 ASVs
summarize_phyloseq(PS.rar)

######################
# rarify, and hellinger
PS.500.hell.test = transform_sample_counts(PS.rar, function(x) sqrt(x / sum(x)))

###### ###### ###### 
# Compare full data vs. rarefied data beta diversity
####### full data
ord.full <- ordinate(PS.fin, method = 'PCoA', distance = 'bray')
plot_ordination(PS.fin, ord.full, color = "sample_type")

# what samples have low reads
plot_ordination(PS.fin, ord.full, shape = "sample_type") +
  geom_point(aes(color = read.sum > 500))

Rare.culling<-plot_ordination(PS.fin, ord.full, color = "organism") +
  geom_point(aes(shape = read.sum > 500, size= read.sum > 500)) +
  scale_size_manual(values= c(6,2)) +
  scale_shape_manual(values=c(1, 16))+
  scale_color_manual(values=c("coral", "goldenrod2", "mediumaquamarine", "olivedrab3", 
                              "steelblue1", "violet", "wheat3")) +
  ggtitle("Samples culled by rarefaction") + theme_classic()

# plot the PCoA for non-rarified
Non.rar.NMDS<-plot_ordination(
  physeq = PS.fin,                                                   
  ordination = ord.full) +                                                
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral", "goldenrod2", "mediumaquamarine", "olivedrab3", 
                              "steelblue1", "violet", "wheat3")) +
  ggtitle("ALL taxa-all data") +
  theme_classic() 

###### if rarefied
ord.rar <- ordinate(PS.rar, method = 'PCoA', distance = 'bray')
plot_ordination(PS.rar, ord.rar, color = "sample_type")

Rar.NMDS<- plot_ordination(
  physeq = PS.rar,                                                   
  ordination = ord.rar) +                                                
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral", "goldenrod2", "mediumaquamarine", "olivedrab3", 
                              "steelblue1", "violet", "wheat3")) +
  ggtitle("ALL taxa- 500reads") +
  theme_classic() 

#### rarefied, but hellinger transformed
ord.hell <- ordinate(PS.500.hell.test, method = 'PCoA', distance = 'bray')

Hell.NMDS<- plot_ordination(
  physeq=PS.500.hell.test, 
  ordination = ord.hell) +
  geom_point(aes(color = organism), size = 2) +    
  scale_color_manual(values=c("coral", "goldenrod2", "mediumaquamarine", "olivedrab3", 
                              "steelblue1", "violet", "wheat3")) +
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  ggtitle("ALL taxa - hellinger-500") +
  theme_classic() 


NMDS.tests<-plot_grid(Rare.culling + guides(color="none"), 
                      Non.rar.NMDS +  guides(color="none"), 
                      Rar.NMDS +  guides(color="none"), 
                      Hell.NMDS, ncol=4, rel_widths=c(4,4,4,5.5))
NMDS.tests
dev.copy(pdf, "figures/NMDS.tests.pdf", height=6, width=20)
dev.off()
#######

```
After running the tests of PCoA showing how rarefaction and transformation influence our data (and seeing the low sample size for Bosmina and Ceriodaphnia), we will proceed with a rarified dataset and drop these two species.

Now that we are through the weeding out and have decided on the phyloseq object/OTU table to use, we can get into the analyses.
```{r final phyloseq and PCoA all}
################### final phyloseq object based on the above

# give it a redo now that we know we don't want other levels here...


# this PS.500 is used for the richness data -- non-rarefied
# first, use the PS.fin data and remove the samples no longer wanted
PS.rm = subset_samples(PS.fin, !organism=="Bosmina" & !organism=="Ceriodaphnia") # removes 12 samples
PS.rm <- prune_taxa(taxa_sums(PS.rm) >1, PS.rm) # remove those asvs not in > 1 sample (again)

# removes 35 samples and 5467 OTUs
PS.500 = rarefy_even_depth(PS.rm, rngseed=111, sample.size=500, replace=F) 
PS.500 <- prune_taxa(taxa_sums(PS.500) >0, PS.500) # remove those asvs not in > 1 sample, just a check!
# 1008 taxa and 281 samples

table(sample_data(PS.500)$sample_type) #281 samples: 28 water, 253 zooplankton, 1008 ASVs
table(sample_data(PS.500)$organism) 

# classify the large calanoids as calanoids
sample_data(PS.500)$organism<-sample_data(PS.500)$organism %>% 
  dplyr::recode_factor(
    "Large Calanoid" = "Calanoid")

# check sample data
table(sample_data(PS.500)$organism) 
# Calanoid    Daphnia Holopedium  Cyclopoid      Water 
#       74         98          8         73         28 

# set order of organims
sample_data(PS.500)$organism<-factor(sample_data(PS.500)$organism, levels =c("Daphnia", "Holopedium", 
                                            "Calanoid", "Cyclopoid", "Water")) # back to factor

##########################
#### export each element
OTU = as(otu_table(PS.500), "matrix")
if(taxa_are_rows(PS.500)){OTU <- t(OTU)}
# Coerce to data.frame
OTUdf.rar = as.data.frame(OTU)

Tax.tab = as(tax_table(PS.500), "matrix")
if(taxa_are_rows(PS.500)){Tax.tab <- t(Tax.tab)}
# Coerce to data.frame
TAXdf.rar = as.data.frame(Tax.tab)

Sam.tab = as(sample_data(PS.500), "matrix")
if(taxa_are_rows(PS.500)){Sam.tab <- t(Sam.tab)}
# Coerce to data.frame
SAMdf.rar = as.data.frame(Sam.tab)

write.csv(OTUdf.rar, "output/OTUdf.rar.csv")
write.csv(TAXdf.rar, "output/TAXdf.rar.csv")
write.csv(SAMdf.rar, "output/SAMdf.rar.csv")


# pull the OTU table and sample data from phyloseq so we can run these analyses in vegan
# if needed, this is quite useful....

# convert the sample_data() within a phyloseq object to a vegan compatible data object
pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

# convert the otu_table() within a phyloseq object to a vegan compatible data object
psotu2veg <- function(physeq) {
  OTU <- otu_table(physeq)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}
```

Inspect effects of rarifaction, or not and hellinger transformation
- perform PCoA on all samples
- perform PCoA on rarified 500 reads 
- perform PCoA on rarified-hellinger transformed
```{r PCoA hell500 all samples}
##############################
######### hellinger transform the cleaned up data -- this is for beta diversity and NMDS etc
# rarify, and hellinger
PS.500.hell = transform_sample_counts(PS.500, function(x) sqrt(x / sum(x))) #1912 taxa in 284 samples

###### check the # of samples in each taxa
table(sample_data(PS.500.hell)$organism)
table(sample_data(PS.500.hell)$sample_type) 
#284 samples: 28 water, 256 zooplankton 
#4 samples removed for low-abundance

# rarefied, and hellinger transformed
ord.hell<- ordinate(PS.500.hell, method = 'PCoA', distance ="bray")

#########
#PCoA Bray
PCoA.hell.rar.bray<- plot_ordination(
  physeq=PS.500.hell, 
  ordination = ord.hell) +
  geom_point(aes(color = organism), size = 2) + 
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  ggtitle("ALL taxa - hellinger-500 Bray") +
  theme_classic()

#########
PCoA.hell.rar.bray
dev.copy(pdf, "figures/PCoA.hell.rar.fin.pdf", height=6, width=7)
dev.off()
```

Here is the map of sampling locations
```{r map}
gps <- as.data.frame(cbind(unique(sample_data(PS.500.hell)$longitude), unique(sample_data(PS.500.hell)$latitude)))
rownames(gps) <- unique(sample_data(PS.500.hell)$lake)

map <- leaflet() %>%
  setView(lng = -119, lat = 38.25, zoom = 10) %>%
  addTiles() %>%
  fitBounds(-118.75, 37.42, -119, 38.06) %>%
  addMarkers(data = gps, lng = ~V1, lat = ~V2, popup = ~row.names(gps)) %>%
  addMarkers(
    lng = -118.7426, lat = 37.43153,
    label = "Eastern Brook (3,155 m)",
    labelOptions = labelOptions(noHide = T, direction = "left",
      style = list(
        "color" = "hotpink",
        "font-family" = "serif",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -118.7444, lat = 37.43839,
    label = "Serene (3,124 m)",
    labelOptions = labelOptions(noHide = T, direction = "right",
      style = list(
        "color" = "green",
        "font-family" = "sanserif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -118.8574, lat = 37.59009,
    label = "Convict (2,315 m)",
    labelOptions = labelOptions(noHide = T, direction = "right",
      style = list(
        "color" = "orange",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
   addMarkers(
    lng = -119.2651, lat = 38.04698,
    label = "Virginia (2,983 m)",
    labelOptions = labelOptions(noHide = T, direction = "bottom",
      style = list(
        "color" = "gold",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -119.2703, lat = 38.05095,
    label = "Blue (3,005 m)",
    labelOptions = labelOptions(noHide = T, direction = "right",
      style = list(
        "color" = "blue",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -119.2770, lat = 38.04806,
    label = "Cooney (3,116 m)",
    labelOptions = labelOptions(noHide = T, direction = "left",
      style = list(
        "color" = "purple",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)")))
map
```

Richness (observed and shannon) using PS.500 
- look at richness by sample type (water vs. zooplankton)
- look at how time and site effect richness
- how does richness differ for individual taxa compared to water (it is much lower...)
```{r alpha diversity - plot_richness}
#richness by organism (or sample type i.e, water) -- not transformed, but rarefied

## plots
richness.plot.samples <- plot_richness(PS.500, x="sample_type", measures=c("Observed", "Shannon"), color="sample_type") +
  labs(y= "Alpha Diversity Measure", x = "Sample Type") + theme_bw() + 
  geom_boxplot(aes(fill=sample_type), alpha=0.7) +
  geom_point(aes(color=sample_type), alpha=0.5) +
  scale_color_manual(values = c("lightsteelblue3", "navajowhite3"))+ guides(color="none") +
  scale_fill_manual(values = c("lightsteelblue2", "navajowhite2"))+ guides(fill="none") +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12))

richness.plot.samples$layers <- richness.plot.samples$layers[-1] # remove the first layer of points 
richness.plot.samples

#####
#richness by time_point
alpha.plot.time <- plot_richness(PS.500, x="time_point", measures=c("Shannon")) + 
  labs(y= "Shannon Diversity", x = "Sampling Time Point") + theme_bw() + 
  geom_point(color="gray15", alpha=0.5) +
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(fill= "gray85", alpha=0.5) + 
  theme(strip.text.x = element_text(size = 12))

alpha.plot.time$layers <- alpha.plot.time$layers[-1] # remove the first layer of points 
alpha.plot.time

#Shannon diversity by Lake
alpha.plot.location <- plot_richness(PS.500, x="lake", measures=c("Shannon")) +
  labs(y= "Shannon Diversity", x = "Lake") + theme_bw() + 
  geom_point(color="gray15", alpha=0.5) +
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(fill= "gray15", alpha=0.5)+ 
  theme(strip.text.x = element_text(size = 12)) +
  theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

alpha.plot.location$layers <- alpha.plot.location$layers[-1] # remove the first layer of points 
alpha.plot.location

# group the above two plots
alpha.time.lake<-plot_grid(alpha.plot.time, alpha.plot.location)

#####
#richness by organisms (zoop species and water)
#Shannon
richness.spp.shan <- plot_richness(PS.500, x="organism", color="organism", measures=c("Shannon")) +
  labs(y= "Alpha Diversity", x = "Zooplankton and Water Samples") + 
  coord_cartesian(ylim=c(0, 5))+
  geom_boxplot(aes(fill=organism), alpha=0.2) + guides(fill="none") + guides(color="none")+
  geom_point(alpha=0.5) +
  scale_fill_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  theme(axis.text.x=element_blank(), strip.text.x = element_text(size = 12)) + theme_bw() +
  theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

richness.spp.shan$layers <- richness.spp.shan$layers[-1] # remove the first layer of points 
richness.spp.shan

#richness by organisms (zoop species and water)
#observed
richness.spp.obs <- plot_richness(PS.500, x="organism", color="organism", measures=c("Observed")) +
  labs(y= "Alpha Diversity", x = "Zooplankton and Water Samples") + 
  coord_cartesian(ylim=c(0, 150))+
  geom_boxplot(aes(fill=organism), alpha=0.2) + guides(fill="none") + guides(color="none")+
  geom_point(alpha=0.5) +
  scale_fill_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  theme(axis.text.x=element_blank(), strip.text.x = element_text(size = 12)) + theme_bw()+
   theme(axis.text.x = element_text(angle = 20, vjust = 0.5, hjust=1))

richness.spp.obs$layers <- richness.spp.obs$layers[-1] # remove the first layer of points 
richness.spp.obs


# combine all richness
Fig.4.rich.plot<-plot_grid(alpha.time.lake, richness.spp.shan, richness.spp.obs,
                      ncol=3, rel_widths=c(12,6,6))
Fig.4.rich.plot
dev.copy(pdf, "figures/Fig.4.rich.plot.pdf", height=6, width=14)
dev.off()


######
# lets look at beta diversity and distance to centroid a bit: (zoop species and water)
# using the rarefied, hellinger data here since beta diversity metric
set.seed(10000)
bc_500.hell.bd <- phyloseq::distance(PS.500.hell, method = "bray") # rarefied and hellinger transformed
bc_beta.df <- as.data.frame(sample_data(PS.500.hell))

# one approach
bd<-betadisper(bc_500.hell.bd, bc_beta.df$organism, type="centroid")
permutest(bd, permutations = 99)
anova(bd)
plot(bd)
boxplot(bd)
TukeyHSD(bd)
plot(TukeyHSD(bd))
plot(bd, axes = c(1,3), seg.col = "black", seg.lty = "dashed")


plot(bd, ellipse = TRUE, hull = FALSE, label.cex = 0.4, main="Distance to Centroids")
dev.copy(pdf, "figures/bc_beta.df.pdf", height=6, width=6)
dev.off()

sqrt(dist(bd$centroids[,bd$eig>0])^2 - dist(bd$centroids[,bd$eig<0]^2))

#
bd$group.distances # this is Average distance to centroid for each group
scores(bd, 1:2, display = "centroids") # this is the centroid for each group
bd.df <- data.frame(Distance_to_centroid = bd$distances, organism=bd$group) # dataframe

betadisp.org.plot<- ggplot(data=bd.df, aes(x=organism, y=Distance_to_centroid, color=organism)) +
  geom_boxplot(aes(fill=organism), alpha=0.2) + guides(fill="none") +
  geom_point(alpha=0.5) + 
  ylim(0,0.8) +
  guides(color=guide_legend(title=""))+ xlab("Sample Type")+ ylab("Distance to centroid") +
  ggtitle("Betadispersion") +
  scale_fill_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  theme(axis.text.x=element_blank()) + theme_bw() 
```

Use Shannon Diversity and Alpha richness to test differences across sampling factors
```{r ASV Richness and Diversity stats}
# some stats
Shan.rich<-estimate_richness(PS.500, measures =c("Observed", "Shannon"))
rownames(Shan.rich) <- substr(rownames(Shan.rich),2,nchar(rownames(Shan.rich))) # to remove the "X"
Shan.rich.df<-merge(data.frame(sample_data(PS.500)), Shan.rich, by = "row.names") # merge 

#### SHANNON straight 3 way no-interaction (bc of missing samples post filtering)
rich.shan.all<- lm(Shannon~ time_point+lake+organism, data=Shan.rich.df)
Anova(rich.shan.all, type=2) #time*lake significant
plot(allEffects(rich.shan.all))

#  posthoc just looking at Lake (p=<0.001)
rich.test.lake<-emmeans(rich.shan.time.lake, pairwise~lake)
multcomp::cld(rich.test.lake, Letters=letters)
# lake          emmean    SE  df lower.CL upper.CL .group
#Serene          1.94 0.120 274     1.70     2.17  a    lowest
#Eastern Brook   2.07 0.115 274     1.85     2.30  a   
#Virginia        2.47 0.108 274     2.26     2.69    b  
#Convict         2.57 0.121 274     2.33     2.81    b  
#Cooney          2.68 0.109 274     2.47     2.89    b  
#Blue            2.71 0.106 274     2.50     2.91    b  


rich.test.group<-emmeans(rich.shan.all, pairwise~organism)
multcomp::cld(rich.test.group, Letters=letters)

#organism   emmean     SE  df lower.CL upper.CL .group
#Calanoid     2.03 0.0648 270     1.90     2.16  a    
#Holopedium   2.12 0.2062 270     1.72     2.53  ab   
#Cyclopoid    2.19 0.0667 270     2.06     2.33   b   
#Daphnia      2.41 0.0565 270     2.30     2.52   b   
#Water        3.99 0.1021 270     3.79     4.19    c  


#### OBSERVED straight 3 way no-interaction (bc of missing samples post filtering)
rich.obs.all<- lm(Observed~ time_point+lake+organism, data=Shan.rich.df)
Anova(rich.obs.all, type=2) #lake and organism significant
plot(allEffects(rich.obs.all))

#  posthoc just looking at groups
obsrich.test.group<-emmeans(rich.obs.all, pairwise~organism)
multcomp::cld(obsrich.test.group, Letters=letters)

#organism   emmean     SE  df lower.CL upper.CL .group
#Holopedium   33.2 5.53 270     22.3     44.1  ab    
#Daphnia      36.7 1.51 270     33.7     39.6  a    
#Calanoid     40.0 1.74 270     36.6     43.5  a    
#Cyclopoid    40.2 1.79 270     36.6     43.7   b    
#Water       112.7 2.74 270    107.3    118.1   c 

```

Now testing effects of environmental variables on Shannon diversity.
```{r Shannon and environ}
# testing effects of environmental variables on Shannon diversity

# Compute correlation at 2 decimal places
Shan.rich.df.env<-Shan.rich.df %>% dplyr::select(temp_C, DOC, elevation_m, chla_ug.L, pH, DO_perc, cond)
corr_matrix = round(cor(Shan.rich.df.env), 2)

# Compute and show the  result
ggcorrplot(corr_matrix, hc.order = TRUE, type = "lower",
          lab = TRUE)

# model 
a.div.mod <- lm(Shannon ~ temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = Shan.rich.df)
a.div.AIC <- MASS::stepAIC(a.div.mod, direction = "both")

# best model with AIC -149.5 ====. Shannon ~ elevation_m + pH + DO_perc + cond
summary(lm(Shannon ~ elevation_m + pH + DO_perc + cond, data = Shan.rich.df))
Anova(lm(Shannon ~ elevation_m + pH + DO_perc + cond, data = Shan.rich.df), type=2)

# elevation
summary(lm(Shannon ~ elevation_m, data=Shan.rich.df)) # adjust-R2=0.008
elevation.shan.plot<-ggplot(Shan.rich.df, aes(x=elevation_m, y=Shannon)) +
  geom_smooth(method = lm, color="gray30", alpha=0.2, fill="gray30")+
  geom_point(color="gray20", size=1.5, alpha=0.6)+
  ylab("Shannon diversity")+
  xlab("Elevation (m)")+
  theme_classic()

# pH
summary(lm(Shannon ~ pH, data=Shan.rich.df)) # adjust-R2=0.006
pH.shan.plot<-ggplot(Shan.rich.df, aes(x=pH, y=Shannon)) +
  geom_smooth(method = lm, color="gray30", alpha=0.2, fill="gray30")+
  geom_point(color="gray20", size=1.5, alpha=0.6)+
  ylab("Shannon diversity")+
  xlab("pH")+
  theme_classic()

#DO 
summary(lm(Shannon ~ DO_perc, data=Shan.rich.df)) # adjust-R2=0.096
DO.shan.plot<-ggplot(Shan.rich.df, aes(x=DO_perc, y=Shannon)) +
  geom_smooth(method = lm, color="gray30", alpha=0.2, fill="gray30")+
  geom_point(color="gray20", size=1.5, alpha=0.6)+
  ylab("Shannon diversity")+
  xlab("Dissolved Oxygen (%)")+
  theme_classic()

#cond
summary(lm(Shannon ~ cond, data=Shan.rich.df)) # adjust-R2=0.076
cond.shan.plot<-ggplot(Shan.rich.df, aes(x=cond, y=Shannon)) +
  geom_smooth(method = lm, color="gray30", alpha=0.2, fill="gray30")+
  geom_point(color="gray20", size=1.5, alpha=0.6)+
  ylab("Shannon diversity")+
  xlab("Conductivity (uS/cm)")+
  theme_classic()
 
multregres.plots.shannon.env<-plot_grid(elevation.shan.plot, DO.shan.plot, cond.shan.plot,
                            ncol=3, nrow=1)
multregres.plots.shannon.env
dev.copy(pdf, "figures/Figx.Multregrss_BW.pdf", height=3.5, width=11)
dev.off()


# summary By for median, mean, and standard error of the Shannon diversity by lake
lake_div <- summaryBy(Shannon ~ lake, data = Shan.rich.df, FUN = c(median, mean, se))
```


Run PERMANOVA on the dissimilarity. The code here is meant to give PERMANOVA tests of main effects, tests of environmental effects in Permanova, and tests of betadispersion for different groups.
- first, we will run analysis for all samples
- then for zooplankton alone
- then for water alone
```{r metaMDS, scores, and PERMANOVAS from BC distance matrices}
####################################
# first let's look at PERMANOVA and NDMS for the full dataset
# rarefied, hellinger
set.seed(1000)
bc_500.hell <- phyloseq::distance(PS.500.hell, method = "bray") # rarefied and hellinger transformed

# run NMDS on Bray distance object (defined above for 'bc_500.hell'), this is
set.seed(520)
nmds.rarhell <- metaMDS(bc_500.hell,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

# get NMDS 1-3, we will use this later in NMDS plots
scores.hell <- as.data.frame(scores(nmds.rarhell, display = "sites"))
nmds.rarhell.df <- merge(sample_data(PS.500.hell), scores.hell, by = 'row.names', all = TRUE)

# pull out the sample data for PERMANOVA
all.sam.df <- pssd2veg(PS.500.hell) 

###############
# run PERMANOVA with Time, Lake, Organisms
# use the bray-curtis dist from phyloseq: bc_500.hell
set.seed(781)
broad.model.zoop <- adonis2(bc_500.hell ~ time_point*lake*organism, 
                            data = all.sam.df, method="bray", permutations = 999)
broad.model.zoop

#test for beta dispersion
anova(betadisper(bc_500.hell, all.sam.df$lake)) #  significant (barely)
anova(betadisper(bc_500.hell, all.sam.df$time_point)) #  significant
anova(betadisper(bc_500.hell, all.sam.df$organism)) # very significant

######################################################
##################
# zoops only
# just zoops, from rarefied data, hellinger transformed
PS.zoops = subset_samples(PS.500.hell, sample_type=="Zooplankton")
PS.zoops <- prune_taxa(taxa_sums(PS.zoops) >0, PS.zoops) # remove those asvs that sum to 0 , just a check!

# both of the below are using rarefied and hellinger transformed data: PS.zoops and PS.water
set.seed(1000)
bc_zoops <- phyloseq::distance(PS.zoops, method = "bray")
zoop.df <- pssd2veg(PS.zoops) # pull out the sample data for PERMANOVA

# run PERMANOVA
set.seed(781)
broad.model.zoop <- adonis2(bc_zoops ~ time_point*lake*organism, data = zoop.df, permutations = 999)
broad.model.zoop

# Env model
env.model.zoop <- adonis2(bc_zoops ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = zoop.df, permutations = 999)
env.model.zoop # latitude is best (0.08),  elevation (0.04), DO (0.02), cond (0.02)

# betadispersion
anova(betadisper(bc_zoops, zoop.df$time_point)) # significant (second)
anova(betadisper(bc_zoops, zoop.df$lake)) # significant (barely)
anova(betadisper(bc_zoops, zoop.df$organism)) # significant (most)

##############################################
#######################
# water only
# just water, from rarefied data, hellinger transformed
PS.water = subset_samples(PS.500.hell, sample_type=="Water")
PS.water <- prune_taxa(taxa_sums(PS.water) >0, PS.water) # remove those asvs that sum to 0 , just a check!

set.seed(1000)
bc_water <- phyloseq::distance(PS.water, method = "bray")
water.df <- pssd2veg(PS.water) # pull out the sample data for PERMANOVA

# run PERMANOVA
set.seed(781)
broad.model.water <- adonis2(bc_water ~ time_point + lake, data = water.df, permutations = 999)
broad.model.water

env.model.water <- adonis2(bc_water ~ latitude + elevation_m + temp_C + pH + 
                             cond + DOC + chla_ug.L + DO_perc, data = water.df, permutations = 999)
env.model.water #latitude (0.21), elevation (0.17), cond (0.07), temp (0.05)

#betadispersion
anova(betadisper(bc_water, water.df$time_point)) #not significant
anova(betadisper(bc_water, water.df$lake)) #not significant
```

Run NMDS with envfit for water, daphnia, calanoids, cyclopoids to better understand how environment is impacting these host-microbiomes. We will run NMDS and fit env.fit, perform PERMANOVA, test betadispersion, and test env conditions on BC matrix. 

- First, run  *Daphnia*
- Holopedium too few to be informative here
```{r daph env NMDS}
# run same analysis but subset to only be cladocerans
# sample data and OTU table for vegan
# subset
daph.phylo<-subset_samples(PS.500.hell, organism=="Daphnia")
daph.phylo <- prune_taxa(taxa_sums(daph.phylo) >0, daph.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied, hellinger
set.seed(1000)
bc_daph.hell <- phyloseq::distance(daph.phylo, method = "bray") # rarefied and hellinger transformed

# run NMDS on Bray distance object (defined above for 'bc_500.hell'), this is
set.seed(520)
nmds.daph.rarhell <- metaMDS(bc_daph.hell,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

# get NMDS 1-3
sp.daph.scores.hell <- as.data.frame(scores(nmds.daph.rarhell, display = "sites"))
nmds.daph.rarhell.df <- merge(sample_data(daph.phylo), sp.daph.scores.hell, by = 'row.names', all = TRUE)

#### pick envr. data and run the envfit function
env.daph.hell<- nmds.daph.rarhell.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.daph <- envfit(nmds.daph.rarhell, env.daph.hell, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

### grab vectors and put into df
spp.daph.scrs <- as.data.frame(scores(vf.daph, display = "vectors")) #*ordiArrowMul(vf.daph)
spp.daph.scrs.test <- cbind(spp.daph.scrs, r=vf.daph$vectors[2], pvals=vf.daph$vectors[4], 
                       Envfact = rownames(spp.daph.scrs))
names(spp.daph.scrs.test)[names(spp.daph.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.daph.scrs.signif<-spp.daph.scrs.test[spp.daph.scrs.test$pvals< 0.05, ]

# make the plot
NMDS.daph <- ggplot(nmds.daph.rarhell.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake)) +
  coord_fixed(ratio=1.6) + 
  stat_ellipse(level=0.9, linetype = 2, aes(x = NMDS1, y = NMDS2, color=lake)) +
  scale_color_manual(values=c("skyblue4", "dodgerblue", "skyblue", "indianred","coral", "thistle3"))+
  ggtitle("Daphnia")+
  geom_segment(data = spp.daph.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "grey") +
  geom_text(data = spp.daph.scrs.signif, aes(x = NMDS1, y = NMDS2, label = Envfact),
            size = 3)+
  theme_classic()

### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.daph <- adonis2(bc_daph.hell ~ time_point + lake, 
                            data = nmds.daph.rarhell.df, permutations = 999)
broad.model.daph # signif lake and time effect, driven by differences in time 2

# Env model
env.model.daph <- adonis2(bc_daph.hell ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.daph.rarhell.df, permutations = 999)
env.model.daph # latitude is best (0.16), elevation (0.09), cond (0.06), other factors ~ 0.03, DOC (0.01)

# betadispersion
anova(betadisper(bc_daph.hell, nmds.daph.rarhell.df$lake)) # not significant, (p=0.317)
```

- Now run NMDS-envfit for *calanoids*
```{r calanoid env nmds}
# subset
calan.phylo<-subset_samples(PS.500.hell, organism=="Calanoid")
calan.phylo <- prune_taxa(taxa_sums(calan.phylo) >0, calan.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied, hellinger
set.seed(1000)
bc_calan.hell <- phyloseq::distance(calan.phylo, method = "bray") # rarefied and hellinger transformed

# run NMDS on Bray distance object (defined above for 'bc_500.hell'), this is
set.seed(520)
nmds.calan.rarhell <- metaMDS(bc_calan.hell,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

# get NMDS 1-3
sp.calan.scores.hell <- as.data.frame(scores(nmds.calan.rarhell, display = "sites"))
nmds.calan.rarhell.df <- merge(sample_data(calan.phylo), sp.calan.scores.hell, by = 'row.names', all = TRUE)

#### pick envr. data and run the envfit function
env.calan.hell<- nmds.calan.rarhell.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.calan <- envfit(nmds.calan.rarhell, env.calan.hell, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

### grab vectors and put into df
spp.calan.scrs <- as.data.frame(scores(vf.calan, display = "vectors")) #*ordiArrowMul(vf.calan)
spp.calan.scrs.test <- cbind(spp.calan.scrs, r=vf.calan$vectors[2], pvals=vf.calan$vectors[4], 
                       Envfact = rownames(spp.calan.scrs))
names(spp.calan.scrs.test)[names(spp.calan.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.calan.scrs.signif<-spp.calan.scrs.test[spp.calan.scrs.test$pvals< 0.05, ]

# make the plot
NMDS.calan <- ggplot(nmds.calan.rarhell.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake)) +
  coord_fixed(ratio=1) +
  scale_color_manual(values=c("skyblue4", "dodgerblue", "skyblue", "indianred","coral", "thistle3"))+
  stat_ellipse(level=0.9, linetype = 2, aes(x = NMDS1, y = NMDS2, color=lake)) +
  ggtitle("Calanoids")+
  geom_segment(data = spp.calan.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "grey") +
  geom_text(data = spp.calan.scrs.signif, aes(x = NMDS1, y = NMDS2, label = Envfact),
            size = 3)+
  theme_classic()

### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.calan <- adonis2(bc_calan.hell ~ time_point + lake, 
                            data = nmds.calan.rarhell.df, permutations = 999)
broad.model.calan # signif lake and time effect, driven by differences in time 1 and 5 (whcih are similar) 

# Env model
env.model.calan <- adonis2(bc_calan.hell ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.calan.rarhell.df, permutations = 999)
env.model.calan # elevation (0.18), latitude is best (0.13), cond/DO/temp (0.03)

# betadispersion
anova(betadisper(bc_calan.hell, nmds.calan.rarhell.df$lake)) # not significant (p=0.128))
```

- Now run NMDS-envfit for *cyclopoids*
```{r cyclopoid env nmds}
# subset
cyclop.phylo<-subset_samples(PS.500.hell, organism=="Cyclopoid")
cyclop.phylo <- prune_taxa(taxa_sums(cyclop.phylo) >0, cyclop.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied, hellinger
set.seed(1000)
bc_cyclop.hell <- phyloseq::distance(cyclop.phylo, method = "bray") # rarefied and hellinger transformed

# run NMDS on Bray distance object (defined above for 'bc_500.hell'), this is
set.seed(520)
nmds.cyclop.rarhell <- metaMDS(bc_cyclop.hell,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

# get NMDS 1-3
sp.cyclop.scores.hell <- as.data.frame(scores(nmds.cyclop.rarhell, display = "sites"))
nmds.cyclop.rarhell.df <- merge(sample_data(cyclop.phylo), sp.cyclop.scores.hell, by = 'row.names', all = TRUE)

#### pick envr. data and run the envfit function
env.cyclop.hell<- nmds.cyclop.rarhell.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.cyclop <- envfit(nmds.cyclop.rarhell, env.cyclop.hell, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

### grab vectors and put into df
spp.cyclop.scrs <- as.data.frame(scores(vf.cyclop, display = "vectors")) #*ordiArrowMul(vf.cyclop)
spp.cyclop.scrs.test <- cbind(spp.cyclop.scrs, r=vf.cyclop$vectors[2], pvals=vf.cyclop$vectors[4], 
                       Envfact = rownames(spp.cyclop.scrs))
names(spp.cyclop.scrs.test)[names(spp.cyclop.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.cyclop.scrs.signif<-spp.cyclop.scrs.test[spp.cyclop.scrs.test$pvals< 0.05, ]

# make the plot
NMDS.cyclop <- ggplot(nmds.cyclop.rarhell.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake)) +
  coord_fixed(ratio=1.2) + 
  stat_ellipse(level=0.9, linetype = 2, aes(x = NMDS1, y = NMDS2, color=lake)) +
  scale_color_manual(values=c("skyblue4", "dodgerblue", "skyblue", "indianred","coral", "thistle3"))+
  ggtitle("Cyclopoids")+
  geom_segment(data = spp.cyclop.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "grey") +
  geom_text(data = spp.cyclop.scrs.signif, aes(x = NMDS1, y = NMDS2, label = Envfact),
            size = 3)+
  theme_classic()


### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.cyclop <- adonis2(bc_cyclop.hell ~ time_point + lake, 
                            data = nmds.cyclop.rarhell.df, permutations = 999)
broad.model.cyclop # signif lake and time effect, driven by time 2

# Env model
env.model.cyclop <- adonis2(bc_cyclop.hell ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.cyclop.rarhell.df, permutations = 999)
env.model.cyclop # latitude is best (0.17), elevation (0.11), cond (0.06), pH (0.05)

# betadispersion
anova(betadisper(bc_cyclop.hell, nmds.cyclop.rarhell.df$lake)) # not significant (p=0.616)

```

- Lastly, run NMDS-envfit for *Water*
```{r water env nmds}
# subset
water.phylo<-subset_samples(PS.500.hell, sample_type=="Water")
water.phylo <- prune_taxa(taxa_sums(water.phylo) >0, water.phylo) # remove those asvs with 0 sum

####################################
# first let's look at PERMANOVA for the full dataset
# rarefied, hellinger
set.seed(1000)
bc_water.hell <- phyloseq::distance(water.phylo, method = "bray") # rarefied and hellinger transformed

# run NMDS on Bray distance object (defined above for 'bc_water.hell'), this is
set.seed(520)
nmds.water.rarhell <- metaMDS(bc_water.hell,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)


# get NMDS 1-3
sp.water.scores.hell <- as.data.frame(scores(nmds.water.rarhell, display = "sites"))
nmds.water.rarhell.df <- merge(sample_data(water.phylo), sp.water.scores.hell, by = 'row.names', all = TRUE)

#### pick envr. data and run the envfit function
env.water.hell<- nmds.water.rarhell.df %>%
  dplyr::select(latitude, longitude, elevation_m, temp_C, chla_ug.L, pH, DO_perc, cond, DOC, TDN, TDP)

set.seed(123)
vf.water <- envfit(nmds.water.rarhell, env.water.hell, perm = 999, na.rm = TRUE) 
# lots of significance, except elevation, pH, DOC, TDN,    highest significance lat/long, spc, cond

### grab vectors and put into df
spp.water.scrs <- as.data.frame(scores(vf.water, display = "vectors")) #*ordiArrowMul(vf.water)
spp.water.scrs.test <- cbind(spp.water.scrs, r=vf.water$vectors[2], pvals=vf.water$vectors[4], 
                       Envfact = rownames(spp.water.scrs))
names(spp.water.scrs.test)[names(spp.water.scrs.test) == 'r'] <- 'r2' # rename bc vector has naming issues

# subset to keep only those Envfactors that are significant 
spp.water.scrs.signif<-spp.water.scrs.test[spp.water.scrs.test$pvals< 0.05, ]

# make the plot
NMDS.water <- ggplot(nmds.water.rarhell.df) +
  geom_point(mapping = aes(x = NMDS1, y = NMDS2, colour = lake)) +
  coord_fixed(ratio=1.6) + 
  stat_ellipse(level=0.9, linetype = 2, aes(x = NMDS1, y = NMDS2, color=lake)) +
  scale_color_manual(values=c("skyblue4", "dodgerblue", "skyblue", "indianred","coral", "thistle3"))+
  ggtitle("Water")+
  geom_segment(data = spp.water.scrs.signif,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "grey") +
  geom_text(data = spp.water.scrs.signif, aes(x = NMDS1, y = NMDS2, label = Envfact),
            size = 3)+
  theme_classic()

### permanova and betadisp
# run PERMANOVA
set.seed(781)
broad.model.water <- adonis2(bc_water.hell ~ time_point + lake, 
                            data = nmds.water.rarhell.df, permutations = 999)
broad.model.water # signif lake effect, lake effects minimal and largely random

# Env model
env.model.water <- adonis2(bc_water.hell ~ latitude + elevation_m + temp_C + pH + cond + 
                            DOC + chla_ug.L + DO_perc, data = nmds.water.rarhell.df, permutations = 999)
env.model.water # latitude is best (0.21), elevation (0.18), cond (0.07), temp (0.05)

# betadispersion
anova(betadisper(bc_water.hell, nmds.water.rarhell.df$lake)) # not significant (p=0.257)

```

- make an aggregate plot of the 4 NMDS-envs
```{r envfit NMDS fig}
extract.legend <- get_legend(
  # create some space to the left of the legend
  NMDS.daph + theme(legend.box.margin = margin(0, 0, 0, 10)))

NMDS.env.species<-plot_grid(NMDS.daph + guides(color="none"), 
                            NMDS.calan + guides(color="none"),
                            NMDS.cyclop + guides(color="none"),
                            NMDS.water + guides(color="none"),
                            extract.legend,
          ncol=5, rel_widths = c(10,10,10,10,3), labels = c("A","B", "C", "D", ""))

NMDS.env.species
dev.copy(pdf, "figures/Figx.NMDS.env.specieswater.pdf", height=5, width=18)
dev.off()
```

This code runs NMDS plots for all the data with different binning factors. May need to reduce and/or decide if PCoA or NMDS is best for showing "full community differences", which presently is in both places.
```{r General NMDS time, lake, basin, etc)}

nmds.rarhell.df$Date<-as.factor(nmds.rarhell.df$time_point)
nmds.rarhell.df$Date<-recode_factor(nmds.rarhell.df$Date, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")

nmds.time <- ggplot(nmds.rarhell.df, aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = Date, shape=sample_type), size = 2) + 
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color=Date)) +
  scale_shape_manual(values=c(1,16))+
  scale_color_manual(values = c("lightblue3","lightcoral", "plum3", "mediumaquamarine", "orange1")) + 
  ggtitle("by Time, pooled") +
  theme_bw()
 
# plot by lake and basin and rename the factors
nmds.rarhell.df$int.basin.type<-interaction(nmds.rarhell.df$basin,nmds.rarhell.df$sample_type)
# recode factors
nmds.rarhell.df$int.basin.type <- recode_factor(nmds.rarhell.df$int.basin.type,
                                     North.Water = "N.Water",
                                     South.Water = "S.Water",
                                     North.Zooplankton = "N.Zoop",
                                     South.Zooplankton = "S.Zoop")

nmds.lake.basin <- ggplot(nmds.rarhell.df, aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = lake, shape=int.basin.type), size = 2) + 
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color=lake)) +
  scale_shape_manual(values=c(0,2,15,17)) +
  scale_color_manual(values=c("skyblue4", "dodgerblue", "skyblue", "indianred","coral", "thistle3"))+
  scale_fill_manual(values=c("skyblue4", "dodgerblue", "skyblue", "indianred","coral", "thistle3"))+
  ggtitle("by Lake-Basin, pooled") +
  theme_bw()

nmds.organism <- ggplot(nmds.rarhell.df, aes(x=NMDS1, y=NMDS2, color = organism)) +
  geom_point(aes(color = organism, shape=sample_type), size = 2) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=organism)) +
  scale_color_manual(values=c("coral2", "goldenrod2", "mediumaquamarine", "springgreen4", "dodgerblue")) +
  scale_shape_manual(values=c(1,16))+
  ggtitle("by Taxa, pooled") +
  theme_bw()

nmds.phy_group <- ggplot(nmds.rarhell.df, aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group, shape=sample_type), size = 2) + 
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color=phy_group)) +
  theme_bw() +
  scale_color_manual(values = c("maroon3","orange2", "royalblue1")) +
  scale_shape_manual(values=c(1,16))+
  ggtitle("by Phygroup, pooled")

phy_group_plots_combined <- plot_grid(
  nmds.organism, nmds.phy_group, nmds.time, nmds.lake.basin,  rel_widths = c(3,3,3,3),
  nrow=2, ncol=2, labels = c("A","B", "C", "D"))

phy_group_plots_combined
dev.print(pdf, "figures/Figx.NMDS.groupeffects.pdf", height=10, width=10)
dev.off()

```

Relative abundance stacked bar plot
```{r stacked bar plots}
########### plotting relative abundance, help from Megan Demmel, this is what I will use for analysis of relative abundances/ composition

#ASV_rel_abund <- transform_sample_counts(PS.fin, function(x) x/sum(x))
#pm <- psmelt(ASV_rel_abund)
pm <- psmelt(PS.500)
Abund <- aggregate(Abundance ~ sample_type + organism + phy_group + lake + time_point + sampleNames + Class, data=pm, FUN=sum)
sam <- aggregate(Abundance ~ sample_type + phy_group + lake + time_point +sampleNames, data=pm, FUN=sum) # this should include everything Abund has except for Class

colnames(sam)[ncol(sam)] <- "tot.abund" # renaming the last column 'tot.abund' so that we can differentiate that from rel.abund

Abund.final <- dplyr::full_join(Abund, sam)
Abund.final$rel_abund <- Abund.final$Abundance/Abund.final$tot.abund

# classes that have rel.abund less than 5% will be named as other
Abund.final$Class[Abund.final$rel_abund <= 0.05] <- "Other (<5%)" 

View(Abund.final)

Abund.final$Class<-as.factor(Abund.final$Class)
Abund.final$Class<-factor(Abund.final$Class, 
                          levels=c("Acidimicrobiia", "Actinobacteria", "Alphaproteobacteria", "Armatimonadia",
                                   "Bacilli", "Bacteroidia", "Cyanobacteriia", "Gammaproteobacteria",
                                   "Gracilibacteria", "Verrucomicrobiae", "Other (<5%)"))

Abund.final$Date<-as.factor(Abund.final$time_point)
Abund.final$Date<-recode_factor(Abund.final$Date, 
                                "1" = "25-Jun",
                                "2" = "6-Jul",
                                "3" = "21-Jul",
                                "4" = "4-Aug",
                                "5" = "22-Aug")

# plot by sample type
overall.comp.class <- ggplot(Abund.final, aes(x=organism, y=rel_abund, fill=Class, color=Class)) +
  geom_bar(aes(fill=Class, color=Class), stat="identity", position="fill") +
  xlab("Sample Type") +ylab("Relative Abundance (%)") + theme_bw() +
  theme(axis.text.x=element_text(size=10)) + theme(axis.title.x=element_text(size=13)) +
  theme(axis.text.y=element_text(size=11)) + theme(axis.title.y=element_text(size=13)) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "plum", "turquoise3", "seagreen4", "lightgreen", "lightskyblue", "royalblue3", "tan3", "lightsalmon")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "plum", "turquoise3", "seagreen4", "lightgreen", "lightskyblue", "royalblue3", "tan3", "lightsalmon"))
overall.comp.class
dev.copy(pdf, "figures/overall.comp.class.pdf", height=6, width=10)
dev.off()

# rel abund plot across time, lake, and sample type
Rel.abund.phy <- ggplot(Abund.final, aes(x = Date, y = rel_abund, fill= Class, color = Class)) +
  geom_bar(width = 0.8, aes(fill = Class, color = Class), stat ="identity", position = "fill") +
  facet_grid(cols = vars(lake), rows = vars(phy_group)) + 
  xlab("Sampling Time Point") +  
  ylab("Relative Abundance (%)") +
  theme_bw() +
  theme(axis.text.x=element_text(size=8, angle = 75, vjust = 0.5, hjust=1), axis.title.x=element_text(size=13)) +
  theme(axis.text.y=element_text(size=11), axis.title.y=element_text(size=13)) +
  theme(strip.text.x = element_text(size = 13), strip.text.y = element_text(size = 13)) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "plum", "turquoise3", "seagreen4", "lightgreen", "lightskyblue", "royalblue3", "tan3", "lightsalmon")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "plum", "turquoise3", "seagreen4", "lightgreen", "lightskyblue", "royalblue3", "tan3", "lightsalmon"))

Rel.abund.phy
dev.copy(pdf, "figures/Relabund.facet.pdf", height=10, width=12)
dev.off()
```

```{r testing effects of env. variables on rel. abund. and richness of dominant bacteria}

# using Abund.final from the last code chunk
# testing the effects of time, lake, and sample type on the 5 most dominant classes of bacteria

############ GAMMAPROTEOBACTERIA
gammaproteobacteria <- subset(Abund.final, Class == "Gammaproteobacteria")
gammaproteobacteria <- summaryBy(rel_abund ~ sampleNames, data = gammaproteobacteria, FUN = sum)
gammaproteobacteria <- data.frame(gammaproteobacteria, row.names = gammaproteobacteria[,1])
names(gammaproteobacteria)[names(gammaproteobacteria) == 'rel_abund.sum'] <- 'Gamma.rel_abund.sum'
gammaproteobacteria <- merge(gammaproteobacteria, sample_data(PS.500), by = "row.names")
names(gammaproteobacteria)[names(gammaproteobacteria) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Gam.model<-lm(Gamma.rel_abund.sum ~ time_point + lake + phy_group, data = gammaproteobacteria)
Anova(Gam.model, type=2)
plot(allEffects(Gam.model))
# time, lake, phy.group all matter

# testing environmental effects on gamaproteobacteria abundances
FitAll.gamma <- lm(Gamma.rel_abund.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = gammaproteobacteria)
step(FitAll.gamma, direction = "both")

# best model
Best.gamma<-lm(Gamma.rel_abund.sum ~ DOC + cond, data = gammaproteobacteria)
summary(Best.gamma)
Anova(Best.gamma) # conductivity is most important, Gamma decline with conduct. overall

gamma_cond <- ggplot(gammaproteobacteria, aes(x=cond, y=Gamma.rel_abund.sum)) +
  geom_smooth(method = lm, color="lightskyblue", alpha=0.2, fill="lightskyblue")+
  geom_point(color="lightskyblue3", size=1.5, alpha=0.6)+
  xlab("Conductivity(uS/cm)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16))+
  ggtitle("Gammaproteobacteria") + theme_classic()
gamma_cond


##### Bacteroidia
bacteroidia <- subset(Abund.final, Class == "Bacteroidia")
bacteroidia <- summaryBy(rel_abund ~ sampleNames, data = bacteroidia, FUN = sum)
bacteroidia <- data.frame(bacteroidia, row.names = bacteroidia[,1])
names(bacteroidia)[names(bacteroidia) == 'rel_abund.sum'] <- 'Bactero.rel_abund.sum'
bacteroidia <- merge(bacteroidia, sample_data(PS.500), by = "row.names")
names(bacteroidia)[names(bacteroidia) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Bactero.model<-lm(Bactero.rel_abund.sum ~ time_point + lake + phy_group, data = bacteroidia)
Anova(Bactero.model, type=2)
plot(allEffects(Bactero.model)) # no differnce

# testing environmental effects on alphaproteobacteria abundances
FitAll.Bactero<- lm(Bactero.rel_abund.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = bacteroidia)
step(FitAll.Bactero, direction = "both") # DO temp best

# best model
Best.bactero<-lm(Bactero.rel_abund.sum ~ temp_C + DOC, data = bacteroidia)
summary(Best.bactero)
Anova(Best.bactero) # DOC signif, temp almost signif, decline with DOC and temp
plot(allEffects(Best.bactero))

# plot it
bactero_DOC <- ggplot(bacteroidia, aes(x=DOC, y=Bactero.rel_abund.sum)) +
  geom_smooth(method = lm, color="seagreen3", alpha=0.2, fill="seagreen3")+
  geom_point(color="seagreen4", size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("DOC (mg/L)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) + 
  ggtitle("Bacteroidia") + theme_classic()
bactero_DOC 


########### ALPHAPROTEOBACTERIA
alphaproteobacteria <- subset(Abund.final, Class == "Alphaproteobacteria")
alphaproteobacteria <- summaryBy(rel_abund ~ sampleNames, data = alphaproteobacteria, FUN = sum)
alphaproteobacteria <- data.frame(alphaproteobacteria, row.names = alphaproteobacteria[,1])
names(alphaproteobacteria)[names(alphaproteobacteria) == 'rel_abund.sum'] <- 'Alphapr.rel_abund.sum'
alphaproteobacteria <- merge(alphaproteobacteria, sample_data(PS.500), by = "row.names")
names(alphaproteobacteria)[names(alphaproteobacteria) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Alpha.model<-lm(Alphapr.rel_abund.sum ~ time_point + lake + phy_group, data = alphaproteobacteria)
Anova(Alpha.model, type=2)
plot(allEffects(Alpha.model))
# time, lake, phy.group all matter

# testing environmental effects on alphaproteobacteria abundances
FitAll.alpha<- lm(Alphapr.rel_abund.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = alphaproteobacteria)
step(FitAll.alpha, direction = "both")

# best model
Best.alpha<-lm(Alphapr.rel_abund.sum ~ DO_perc, data = alphaproteobacteria)
summary(Best.alpha)
Anova(Best.alpha) # DO is most important, Alpha decline with DO increasing
plot(allEffects(Best.alpha))

# plot it
alpha_DO <- ggplot(alphaproteobacteria, aes(x=DO_perc, y=Alphapr.rel_abund.sum)) +
  geom_smooth(method = lm, color="gold", alpha=0.2, fill="gold")+
  geom_point(color="gold3", size=1.5, alpha=0.6)+
  xlab("Dissolved Oxygen (%)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) +
  ggtitle("Alphaproteobacteria") + theme_classic()
alpha_DO


##### #ACTINOBACTERIA
actinobacteria <- subset(Abund.final, Class == "Actinobacteria")
actinobacteria <- summaryBy(rel_abund ~ sampleNames, data = actinobacteria, FUN = sum)
actinobacteria <- data.frame(actinobacteria, row.names = actinobacteria[,1])
names(actinobacteria)[names(actinobacteria) == 'rel_abund.sum'] <- 'Actinob.rel_abund.sum'
actinobacteria <- merge(actinobacteria, sample_data(PS.500), by = "row.names")
names(actinobacteria)[names(actinobacteria) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Actin.model<-lm(Actinob.rel_abund.sum ~ time_point + lake + phy_group, data = actinobacteria)
Anova(Actin.model, type=2)
plot(allEffects(Actin.model))
# lake, phy.group all matter

# testing environmental effects on alphaproteobacteria abundances
FitAll.actin<- lm(Actinob.rel_abund.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = actinobacteria)
step(FitAll.actin, direction = "both") # chla best model

# best model
Best.actin<-lm(Actinob.rel_abund.sum ~ chla_ug.L, data = actinobacteria)
summary(Best.actin)
Anova(Best.actin) # chla is most important, Actino increase with chla overall
plot(allEffects(Best.actin))

# plot it
actino_chla <- ggplot(actinobacteria, aes(x=chla_ug.L, y=Actinob.rel_abund.sum)) +
  geom_smooth(method = lm, color="orangered2", alpha=0.2, fill="orangered2")+
  geom_point(color="coral", size=1.5, alpha=0.6)+
  ylim(c(0,1))+
  xlab("Chlorophyll a (ug/L)") + ylab("Relative Abundance") +
  theme(axis.text.x=element_text(size=14)) + theme(axis.title.x=element_text(size=16)) +
  theme(axis.text.y=element_text(size=14)) + theme(axis.title.y=element_text(size=16)) +
  ggtitle("Actinobacteria") + theme_classic()
actino_chla 


##### CYANOBACTERIA
cyanobacteriia <- subset(Abund.final, Class == "Cyanobacteriia")
cyanobacteriia <- summaryBy(rel_abund ~ sampleNames, data = cyanobacteriia, FUN = sum)
cyanobacteriia <- data.frame(cyanobacteriia, row.names = cyanobacteriia[,1])
names(cyanobacteriia)[names(cyanobacteriia) == 'rel_abund.sum'] <- 'Cyano.rel_abund.sum'
cyanobacteriia <- merge(cyanobacteriia, sample_data(PS.500), by = "row.names")
names(cyanobacteriia)[names(cyanobacteriia) == 'sampleNames.x'] <- 'sampleNames'

#run a model
Cyano.model<-lm(Cyano.rel_abund.sum ~ time_point + phy_group, data = cyanobacteriia)
Anova(Cyano.model, type=2)
plot(allEffects(Cyano.model)) # no differnce

# testing environmental effects on alphaproteobacteria abundances
FitAll.Cyano<- lm(Cyano.rel_abund.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = cyanobacteriia)
step(FitAll.Cyano, direction = "both") # DO only best

# best model
Best.cyano<-lm(Actinob.rel_abund.sum ~ DO_perc, data = actinobacteria)
summary(Best.cyano)
Anova(Best.cyano) # DO best, but not significant


#### pool the plots
Taxa.relabund.env.regr<-plot_grid(gamma_cond, alpha_DO, actino_chla, bactero_DOC, ncol=4)

######### combine shannon with the taxa-specific
plot_grid(multregres.plots.shannon.env, Taxa.relabund.env.regr, ncol=1, nrow=2, labels=c("A", "B"), rel_heights = c(0.8,1))
dev.copy(pdf, "figures/Shann.env.Taxa.relabund.regr.pdf", width=12, height=7)
dev.off()
```


```{r venn diagrams}
ASV.tab<-as.data.frame(otu_table(PS.500))
ASV.samp<-sample_data(PS.500)

ASV.data.all<-as.data.frame(cbind(ASV.samp,ASV.tab))

remotes::install_github("Russel88/MicEco")
library("MicEco")

# venns of all samples by time
All.venn.time<-ps_venn(
  PS.500,
  "time_point",
  fraction = 0,
  weight = FALSE,
  relative = TRUE,
  plot = TRUE,
  main="All times"
)

# venns of all phy
All.venn.phy<-ps_venn(
  PS.500,
  "phy_group",
  fraction = 0,
  weight = FALSE,
  relative = TRUE,
  plot = TRUE,
  main="All times"
)

# venn for all water
All.venn.water.time<-ps_venn(
  PS.water,
  "time_point",
  plot = TRUE,
  #fill= c("lightblue", "orange"),
  main="Water by time"
)


zoops<-ps_venn(
  PS.zoops,
  "organism",
  plot = TRUE,
  main="Zoops"
)

# venns of all phy
All.venn.org<-ps_venn(
  PS.500,
  "organism",
  plot = TRUE,
  quantities = list(type = "counts", cex = 1),
  main="All times"
)


# venn for all daphnia
PS.daph = subset_samples(PS.500, organism=="Daphnia")
PS.daph <- prune_taxa(taxa_sums(PS.daph) > 0, PS.daph)

PS.daph.south = subset_samples(PS.daph, basin=="South")
PS.daph.south <- prune_taxa(taxa_sums(PS.daph.south) > 0, PS.daph.south)

PS.daph.north = subset_samples(PS.daph, basin=="North")
PS.daph.north <- prune_taxa(taxa_sums(PS.daph.north) > 0, PS.daph.north)

#see the basin effects
richness.lake <- plot_richness(PS.daph, x="lake", measures=c("Observed", "Shannon"), color="lake") +
  labs(y= "Alpha Diversity Measure", x = "lake") + theme_bw() + 
  geom_boxplot(aes(fill=lake), alpha=0.7) +
  geom_point(aes(color=lake), alpha=0.5) +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12))

daph.S<-ps_venn(
  PS.daph.south,
  "lake",
  fraction = 0,
  weight = FALSE,
  relative = TRUE,
  plot = TRUE,
  main="Daphnia by South-lake"
)

daph.N<-ps_venn(
  PS.daph.north,
  "lake",
  fraction = 0,
  weight = FALSE,
  relative = TRUE,
  plot = TRUE,
  main="Daphnia by North-lake"
)

daph.basin<-ps_venn(
  PS.daph,
  "basin",
  fraction = 0,
  weight = FALSE,
  relative = FALSE,
  plot = TRUE,
  main="Daphnia by Basin"
)

g<-plot_grid(All.venn.time, All.venn.water.time, All.venn.phy,
             daph.N, daph.S, daph.basin, ncol=3)
g
dev.copy(pdf, "figures/Venns.daph.pdf", height=8, width=12)
dev.off()

#########
Euler.phy<-ps_euler(
  PS.500,
  "phy_group",
  shape = "circle",
  quantities = list(type=c("percent","counts"), font = 1, cex = 0.7), labels = list(cex = 0.9),
  plot = TRUE
)

Venn.phy<-ps_venn(
  PS.500,
  "phy_group",
  quantities = list(type=c("percent","counts"), font = 1, cex = 0.7), labels = list(cex = 0.9),
  plot = TRUE
)

Venn.zoop<-ps_venn(
  PS.zoops,
  "organism",
  quantities = list(type=c("percent","counts"), font = 1, cex = 0.7), labels = list(cex = 0.9),
  plot = TRUE
)

g2<-plot_grid(Euler.phy, Venn.zoop,
            ncol=2)
g2
dev.copy(pdf, "figures/Euler.Venn.phy.zoop.pdf", height=7, width=10)
dev.off()


```

look for core taxa for Daphnia
```{r core taxa daphnia, eval=FALSE}
# https://microbiome.github.io/tutorials/Core.html

#Zooplankton core taxa
daph.phylo.rar<-subset_samples(PS.500.hell, organism=="Daphnia")
#keep only taxa with positive sums
daph.phylo.pos <- prune_taxa(taxa_sums(daph.phylo.rar) > 0, daph.phylo.rar)
# make compositional, relative abundance
pseq.rel <- microbiome::transform(daph.phylo.pos, "compositional")

taxa_names(pseq.rel) # shows ASVs

#If you only need the names of the core taxa, do as follows. This returns the taxa that exceed the given prevalence and detection thresholds.
core.taxa.standard <- core_members(pseq.rel, detection = 0.0001, prevalence = 50/100)

colnames(tax_table(pseq.rel))[1] = "Domain" #rename Kingdom to Domain for the microbiome package needs
pseq.rel.f <- microbiome::add_besthit(pseq.rel) # adds taxonomy to ASV names

core.taxa.standard <- core_members(pseq.rel.f, detection = 0.0001, prevalence = 50/100)

#full phyloseq
pseq.core <- core(pseq.rel.f, detection = 0.0001, prevalence = .5)
core.taxa <- taxa(pseq.core)
class(core.taxa)

# get the taxonomy data
tax.mat <- tax_table(pseq.core)
tax.df <- as.data.frame(tax.mat)

# add the OTus to last column
tax.df$OTU <- rownames(tax.df)

# select taxonomy of only 
# those OTUs that are core memebers based on the thresholds that were used.
core.taxa.class <- dplyr::filter(tax.df, rownames(tax.df) %in% core.taxa)
knitr::kable(head(core.taxa.class))


# With compositional (relative) abundances
det <- c(0, 0.1, 0.5, 2, 5, 20)/100
prevalences <- seq(.05, 1, .05)

plot_core(pseq.rel.f, prevalences = prevalences, 
          detections = det, plot.type = "lineplot") + 
  xlab("Relative Abundance (%)") + 
  theme_bw()


# Core with compositionals:
prevalences <- seq(.05, 1, .05)
detections <- round(10^seq(log10(1e-2), log10(.5), length = 10), 3)

#Deletes "ASV" from taxa_names, e.g. ASV1 --> 1
#taxa_names(ps.m3.rel) = taxa_names(ps.m3.rel) %>% str_replace("ASV", "")
# Also define gray color palette
gray <- gray(seq(0,1,length=5))

daph.core <- plot_core(pseq.rel.f,
  plot.type = "heatmap",
  colours = gray,
  prevalences = prevalences,
  detections = detections, min.prevalence = .5) +
  xlab("Detection Threshold (Relative Abundance (%))") + theme_bw() + ylab("ASVs")
daph.core

## make in color
library(viridis)
library(RColorBrewer)

print(daph.core + scale_fill_viridis())
dev.print(pdf, "figures/Daph.core.pdf", height=6, width=6)
dev.off()
```

look for core taxa for Calanoids
```{r core calan, eval=FALSE}
#Zooplankton core taxa
calan.phylo.rar<-subset_samples(PS.500.hell, organism=="Calanoid")
#keep only taxa with positive sums
calan.phylo.pos <- prune_taxa(taxa_sums(calan.phylo.rar) > 0, calan.phylo.rar)
# make compositional, relative abundance
pseq.rel <- microbiome::transform(calan.phylo.pos, "compositional")

taxa_names(pseq.rel) # shows ASVs

#If you only need the names of the core taxa, do as follows. This returns the taxa that exceed the given prevalence and detection thresholds.
core.taxa.standard <- core_members(pseq.rel, detection = 0.0001, prevalence = 50/100)

colnames(tax_table(pseq.rel))[1] = "Domain" #rename Kingdom to Domain for the microbiome package needs
pseq.rel.f <- microbiome::add_besthit(pseq.rel) # adds taxonomy to ASV names

core.taxa.standard <- core_members(pseq.rel.f, detection = 0.0001, prevalence = 50/100)

#full phyloseq
pseq.core <- core(pseq.rel.f, detection = 0.0001, prevalence = .5)
core.taxa <- taxa(pseq.core)
class(core.taxa)

# get the taxonomy data
tax.mat <- tax_table(pseq.core)
tax.df <- as.data.frame(tax.mat)

# add the OTus to last column
tax.df$OTU <- rownames(tax.df)

# select taxonomy of only 
# those OTUs that are core memebers based on the thresholds that were used.
core.taxa.class <- dplyr::filter(tax.df, rownames(tax.df) %in% core.taxa)
knitr::kable(head(core.taxa.class))


# With compositional (relative) abundances
det <- c(0, 0.1, 0.5, 2, 5, 20)/100
prevalences <- seq(.05, 1, .05)

plot_core(pseq.rel.f, prevalences = prevalences, 
          detections = det, plot.type = "lineplot") + 
  xlab("Relative Abundance (%)") + 
  theme_bw()


# Core with compositionals:
prevalences <- seq(.05, 1, .05)
detections <- round(10^seq(log10(1e-2), log10(.4), length = 10), 3)

#Deletes "ASV" from taxa_names, e.g. ASV1 --> 1
#taxa_names(ps.m3.rel) = taxa_names(ps.m3.rel) %>% str_replace("ASV", "")
# Also define gray color palette
gray <- gray(seq(0,1,length=5))

calan.core <- plot_core(pseq.rel.f,
  plot.type = "heatmap",
  colours = gray,
  prevalences = prevalences,
  detections = detections, min.prevalence = .5) +
  xlab("Detection Threshold (Relative Abundance (%))")

calan.core <- calan.core + theme_bw() + ylab("ASVs")
calan.core

## make in color
library(viridis)
library(RColorBrewer)

print(calan.core + scale_fill_viridis())
dev.print(pdf, "figures/Calan.core.pdf", height=6, width=6)
dev.off()
```

look for core taxa for Cyclopoids
```{r cyclo core,  eval=FALSE}
#Zooplankton core taxa
cyclop.phylo.rar<-subset_samples(PS.500.hell, organism=="Cyclopoid")
#keep only taxa with positive sums
cyclop.phylo.pos <- prune_taxa(taxa_sums(cyclop.phylo.rar) > 0, cyclop.phylo.rar)
# make compositional, relative abundance
pseq.rel <- microbiome::transform(cyclop.phylo.pos, "compositional")

taxa_names(pseq.rel) # shows ASVs

#If you only need the names of the core taxa, do as follows. This returns the taxa that exceed the given prevalence and detection thresholds.
core.taxa.standard <- core_members(pseq.rel, detection = 0.0001, prevalence = 50/100)

colnames(tax_table(pseq.rel))[1] = "Domain" #rename Kingdom to Domain for the microbiome package needs
pseq.rel.f <- microbiome::add_besthit(pseq.rel) # adds taxonomy to ASV names

core.taxa.standard <- core_members(pseq.rel.f, detection = 0.0001, prevalence = 50/100)

#full phyloseq
pseq.core <- core(pseq.rel.f, detection = 0.0001, prevalence = .5)
core.taxa <- taxa(pseq.core)
class(core.taxa)

# get the taxonomy data
tax.mat <- tax_table(pseq.core)
tax.df <- as.data.frame(tax.mat)

# add the OTus to last column
tax.df$OTU <- rownames(tax.df)

# select taxonomy of only 
# those OTUs that are core memebers based on the thresholds that were used.
core.taxa.class <- dplyr::filter(tax.df, rownames(tax.df) %in% core.taxa)
knitr::kable(head(core.taxa.class))


# With compositional (relative) abundances
det <- c(0, 0.1, 0.5, 2, 5, 20)/100
prevalences <- seq(.05, 1, .05)

plot_core(pseq.rel.f, prevalences = prevalences, 
          detections = det, plot.type = "lineplot") + 
  xlab("Relative Abundance (%)") + 
  theme_bw()


# Core with compositionals:
prevalences <- seq(.05, 1, .05)
detections <- round(10^seq(log10(1e-2), log10(.3), length = 10), 3)

#Deletes "ASV" from taxa_names, e.g. ASV1 --> 1
#taxa_names(ps.m3.rel) = taxa_names(ps.m3.rel) %>% str_replace("ASV", "")
# Also define gray color palette
gray <- gray(seq(0,1,length=5))

cyclop.core <- plot_core(pseq.rel.f,
  plot.type = "heatmap",
  colours = gray,
  prevalences = prevalences,
  detections = detections, min.prevalence = .5) +
  xlab("Detection Threshold (Relative Abundance (%))")

cyclop.core <- cyclop.core + theme_bw() + ylab("ASVs")
cyclop.core

## make in color
library(viridis)
library(RColorBrewer)

print(cyclop.core + scale_fill_viridis())
dev.print(pdf, "figures/Cyclop.core.pdf", height=6, width=6)
dev.off()
```

look for core taxa for Daphnia by genus
```{r core taxa daphnia by genus,  eval=FALSE}
###################### by Genus
###########
ps.gen <- aggregate_taxa(daph.phylo.pos, "Genus")

# Check if any taxa with no genus classification. aggregate_taxa will merge all unclassified to Unknown
any(taxa_names(ps.gen) == "Unknown")
# Remove Unknown
ps.gen <- subset_taxa(ps.gen, Genus!="Unknown")

# make compositional, relative abundance
pseq.rel.gen <- microbiome::transform(ps.gen, "compositional")

#keep only taxa with positive sums
pseq.rel.gen <- prune_taxa(taxa_sums(pseq.rel.gen) > 0, pseq.rel.gen)

#### setup for plot
prevalences <- seq(.05, 1, .05) # heatmap scale
detections <- round(10^seq(log10(1e-3), log10(.3), length = 10), 3)

daph.genera.core <- plot_core(pseq.rel.gen, 
                plot.type = "heatmap", 
                colours = rev(brewer.pal(5, "RdBu")),
                prevalences = prevalences,
                detections = detections, min.prevalence = 0.5) +
    xlab("Detection Threshold (Relative Abundance (%))") + theme_bw() + ylab("ASVs")
daph.genera.core
dev.print(pdf, "figures/daph.genera.core.pdf", height=5, width=7)
dev.off()

```

Don't need this, but useful if wanting to look at unfrac-weighted/unweighted
```{r PCoA testing unifrac, eval=FALSE}
# PCoA - unifrac/weighted

# make phylogenetic tree and add to phyloseq if wanting to do unifrac weight/unweighted
library("ape")
random_tree = rtree(ntaxa(PS.500), rooted=TRUE, tip.label=taxa_names(ps.prune.LOCAL))

# add in
phy_tree(PS.500)<-random_tree


ord.hell.unif <- ordinate(PS.500.hell, method = 'PCoA', distance ="unifrac")
ord.hell.wunif <- ordinate(PS.500.hell, method = 'PCoA', distance ="wunifrac")

PCoA.hell.rar.wunif<- plot_ordination(
  physeq=PS.500.hell, 
  ordination = ord.hell.wunif) +
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  ggtitle("ALL taxa - hellinger-500.weight unifrac") +
  theme_classic() 

#PCoA - unifrac/unweighted
PCoA.hell.rar.unif<- plot_ordination(
  physeq=PS.500.hell, 
  ordination = ord.hell.unif) +
  geom_point(aes(color = organism), size = 2) +    
  stat_ellipse(level=0.95, linetype = 2, aes(color=organism)) +
  ggtitle("ALL taxa - hellinger-500.unifrac") +
  theme_classic() 

# plot of all 3 PCoAs
plot_grid(PCoA.hell.rar.bray + guides(color="none"), 
          PCoA.hell.rar.wunif + guides(color="none") , 
          PCoA.hell.rar.unif, ncol=3, rel_widths = c(3,3,4))
dev.print(pdf, "figures/Unifract.ord.pdf", height=4.5, width=15)
dev.off()
```

```{r nestedness}
#nestedness

#Merge samples by sample type
agg=merge_samples(PS.500, "organism")

#convert to dataframe
aggtab=as.data.frame(otu_table(agg))

#calculate nested temperature
nested=nestedtemp(aggtab) 
#nestedness temperature: 30.96868 
#with matrix fill 0.3309509 

#compare nestedtemp to randomized null distribution
# note memory intensive
agg.mod<-oecosimu(round(aggtab), nsimul=1000, nestedtemp, "r2dtable", parallel=4)



# separate by zoops and water
#Merge samples by sample type
agg.zo.wa=merge_samples(PS.500, "phy_group")

#convert to dataframe
aggtab.zo.wa=as.data.frame(otu_table(agg.zo.wa))

#calculate nested temperature
nested.zo.wa=nestedtemp(aggtab.zo.wa) 
#nestedness temperature: 34.40586 
#with matrix fill 0.4784979 

#compare nestedtemp to randomized null distribution
# note memory intensive
agg.zo.wa.mod<-oecosimu(round(aggtab.zo.wa), nsimul=1000, nestedtemp, "r2dtable", parallel=4)



#make a two row composite figure, reguce spacing between inner and outer margins
pdf(file= "figures/nestedness.pdf", height=8, width=6)

par(mfrow=c(2,1), mai=c(.1,1.,1,1), oma=c(.1,1,.1,1))

#plot the Organism nestedness nested figure, suppress taxon names
plot(nested, kind="incid", names=c(TRUE,FALSE), col=c("white", "gray48"), lwd=2, main="Zooplankton and Water", cex.main=0.8, cex.axis=0.8)
#add stats are margin text
mtext(expression(paste("Nested Temp=30.97, ", italic("p") ,"=0.001")), cex=0.7,)
mtext(expression(bold('a')), side=3, line=0, at=0, cex=0.8,)


#plot the Water and zoops nested figure, suppress taxon names
plot(nested.zo.wa, kind="incid", names=c(TRUE,FALSE), col=c("white", "gray48"), lwd=2, main="Grouped Zooplankton Taxa and Water", cex.main=0.8, cex.axis=0.8)
#add stats are margin text
mtext(expression(paste("Nested Temp=30.17, ", italic("p") ,"=0.001")), cex=0.7,)
mtext(expression(bold('a')), side=3, line=0, at=0, cex=0.8,)

dev.copy(pdf, "figures/nestedness.pdf", height=8, width=6)
dev.off()

```

